<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />




<title>Module 5 Project</title>

<script src="site_libs/header-attrs-2.28/header-attrs.js"></script>
<script src="site_libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/flatly.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/jqueryui-1.13.2/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/default.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>

<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>





<style type="text/css">
/* for pandoc --citeproc since 2.11 */
div.csl-bib-body { }
div.csl-entry {
  clear: both;
}
.hanging div.csl-entry {
  margin-left:2em;
  text-indent:-2em;
}
div.csl-left-margin {
  min-width:2em;
  float:left;
}
div.csl-right-inline {
  margin-left:2em;
  padding-left:1em;
}
div.csl-indent {
  margin-left: 2em;
}
</style>

<link rel="stylesheet" href="style.css" type="text/css" />



<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
details > summary > p:only-child {
  display: inline;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark the anchor link active (and if it's in a dropdown, also mark that active)
  var dropdown = menuAnchor.closest('li.dropdown');
  if (window.bootstrap) { // Bootstrap 4+
    menuAnchor.addClass('active');
    dropdown.find('> .dropdown-toggle').addClass('active');
  } else { // Bootstrap 3
    menuAnchor.parent().addClass('active');
    dropdown.addClass('active');
  }

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before, .tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "\e259";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "\e258";
  font-family: 'Glyphicons Halflings';
  border: none;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-bs-toggle="collapse" data-target="#navbar" data-bs-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">BIOL 3140</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="schedule.html">Schedule</a>
</li>
<li>
  <a href="at_a_glance.html">Calendar</a>
</li>
<li>
  <a href="WCRs.html">WCRs</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    Projects
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li class="dropdown-header">Phase I</li>
    <li>
      <a href="phaseI_rubric.html">Phase I report guidlines</a>
    </li>
    <li>
      <a href="mod1_proj.html">Module 1 Project</a>
    </li>
    <li>
      <a href="mod2_proj.html">Module 2 Project</a>
    </li>
    <li>
      <a href="mod3_proj.html">Module 3 Project</a>
    </li>
    <li>
      <a href="mod4_proj.html">Module 4 Project</a>
    </li>
    <li class="dropdown-header">Phase II</li>
    <li>
      <a href="phaseII_rubric.html">Phase II report guidlines</a>
    </li>
    <li>
      <a href="mod5_proj.html">Module 5 Project</a>
    </li>
    <li>
      <a href="mod6_proj.html">Module 6 Project</a>
    </li>
    <li>
      <a href="mod7_proj.html">Module 7 Project</a>
    </li>
    <li class="dropdown-header">Phase III</li>
    <li>
      <a href="phaseIII_rubric.html">Phase III report guidlines</a>
    </li>
    <li>
      <a href="final_proj.html">Final Project</a>
    </li>
  </ul>
</li>
<li>
  <a href="https://github.com/orgs/bcorgbio/discussions">BIOL 3140 Discussion</a>
</li>
<li>
  <a href="https://bostoncollege.instructure.com/courses/1659476/gradebook">Grades on Canvas</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    Peer Contribution Evaluations
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="peer_contributions.html">Peer Contribution Evaluation Guidelines</a>
    </li>
    <li>
      <a href="https://forms.gle/ATeFa9wsV1t4g27g6">Author Contribution Evaluation Form</a>
    </li>
  </ul>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">



<h1 class="title toc-ignore">Module 5 Project</h1>

</div>


<style>
div.green { background-color:#c7d1c0; border-radius: 5px; padding: 20px;}
</style>
<!-- Add icon library -->
<p><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.14.0/css/all.min.css"></p>
<div id="introduction" class="section level1">
<h1>Introduction</h1>
<p>Sharks are an ancient group of cartilaginous fishes in the class
Chondrichthyes, subclass Elasmobranchii, and subdivision Selachimorpha.
The selachians include over 500 species that are further placed in in
two superorders: Squalomorphii and Galeomorphii. Modern sharks
originated nearly 200 million years ago (<span class="citation">Grogan,
Lund, and Greenfest-Allen (2012)</span>) and, since then, have
diversified into nearly every major marine ecosystem. This evolutionary
success has been attributed to the diversity of their locomotor modes
and associated body shape (<span class="citation">Lauder and Di Santo
(2015)</span>,<span class="citation">Thomson and Simanek (1977)</span>).
A modest body of work devoted to the <a
href="https://en.wikipedia.org/wiki/Ecomorphology">ecomorphology</a> of
sharks has focused largely on caudal find shape as it relates to ecology
and locomotor modes (<span class="citation">Thomson (1976)</span>,<span
class="citation">Wilga and Lauder (2004)</span>). Sharks have a
heterocercal tail in which the caudal part of notochord flexes dorsally
and supports a large dorsal lobe of the fin, while, with more than a few
exceptions, the ventral lobe is small.</p>
<p><br> <br></p>
<center>
<p><img src="https://ars.els-cdn.com/content/image/1-s2.0-S0944200620300581-gr1.jpg" alt="bull shark" width="400"/></p>
<p><br> <br></p>
(<b>A</b>) The phylogenetic relationships and systematic placement of
major groups of shark and (<b>B</b>) <span class="citation">Thomson and
Simanek (1977)</span>’s ecomorphological groups. Figure from <span
class="citation">Sternes and Shimada (2020)</span>.
</center>
<p><br> <br></p>
<p>In there seminal work on shark ecomorphology, <span
class="citation">Thomson and Simanek (1977)</span> put forth that there
were four shark morphotypes:</p>
<ol style="list-style-type: decimal">
<li>A group characterized by having a deep body, large pectoral fins, a
symmetrical caudal fin with a narrow peduncle (ie., base to the fin) and
high-aspect ratio. A white shark represents this group.</li>
<li>A group characterized by a body less deep then Group 1 and cadual
fin that is swept (ie., large heterocercal angle–the angle between the
upper and lower lobes of the fin). This group includes most carcharhinid
sharks such as the bull shark.</li>
<li>A group with a large head, blunt snout, a body that has anteriorly
placed pelvic fins and posteriorly placed dorsal and pelvic, and a
caudal fin with a low heterocercal angle and small ventral lobe.
Catsharks (family Scyliorhinidae) and dogfish (order Squaliformes)
represent this group.</li>
<li>A group characterized by a caudal fin with a higher aspect angle
similar to that of Group 2 but lacking an anal fin.</li>
</ol>
<center>
<table>
<tr>
<td>
<center>
<img src="https://earthlife.net/wp-content/uploads/What-Is-A-Elasmobranch-1280x718-1.webp" alt="bull shark" height="200"/><br />
<br> A white shark, <i>Carcharias carchardon</i>, typical of Group 1.
<br> <br>
</center>
</td>
<td>
<img style="height: 20px; width: 20px"/>
</td>
<td>
<center>
<img src="https://biogeodb.stri.si.edu/sftep/resources/img/images/species/95_8407.jpg" alt="bull shark" height="200"/><br />
<br> A bull shark, <i>Carcharhinus leucas</i>, typical of Group 2. <br>
<br>
</center>
</td>
</tr>
<br> <br>
<tr>
<td>
<center>
<img src="https://www.vichighmarine.ca/wp-content/uploads/2017/03/Spiny-Dogfishy-600x321.jpg" alt="bull shark" height="200"/><br />
<br> A dogfish of the genus <i>Squalus</i> typical of Group 3. <br>
</center>
</td>
<td>
<img style="height: 20px; width: 20px"/>
</td>
<td>
<center>
<img src="https://biogeodb.stri.si.edu/caribbean/resources/img/images/species/4437_2257.jpg" alt="bull shark" height="200"/><br />
<br> A kitefin shark, <i>Dalatias licha</i>, typical of Group 4. <br>
</center>
</td>
</tr>
</table>
</center>
<p><br> <br></p>
<p><span class="citation">Sternes and Shimada (2020)</span> used <a
href="https://en.wikipedia.org/wiki/Morphometrics#Landmark-based_geometric_morphometrics">geometric
morphometrics</a> and a <a
href="https://en.wikipedia.org/wiki/Principal_component_analysis">multivariate
analysis</a> to evaluate the groups proposed by <span
class="citation">Thomson and Simanek (1977)</span> and found that sharks
fell into one of two morphotypes (A and B). Their Group A was
characterized by an elongate body, posteriorly placed dorsal fin, and a
swept caudal while Group B was characterized by a deeper body,
anteriorly place dorsal fins, and a less swept caudal fin. <span
class="citation">Thomson and Simanek (1977)</span> suggested that the
shape of species in these two broad ecomorphs evolved due to differences
in ecological niche and <a
href="https://en.wikipedia.org/wiki/Fish_locomotion">locomotor
behavior</a>. Specifically, they posited that the species of elongate
Group A are typically <a
href="https://en.wikipedia.org/wiki/Demersal_fish">benthic</a> and swim
like eels whereas the deeper bodied species of Group B are often <a
href="https://en.wikipedia.org/wiki/Pelagic_fish">pelagic</a> and swim
like salmon or tunas.</p>
<center>
<p><img src="https://ars.els-cdn.com/content/image/1-s2.0-S0944200620300581-ga1_lrg.jpg" alt="bull shark" width="400"/></p>
<p><br> <br></p>
From <span class="citation">Sternes and Shimada (2020)</span> who found
sharks could be assigned two morphotype groups: A, elongate body,
posteriorly placed dorsal fin, and a swept caudal fin; B, deeper body,
anteriorly place dorsal fins, and a less swept caudal fin.
</center>
<p><br> <br></p>
<p>Although the work of <span class="citation">Thomson and Simanek
(1977)</span> and later <span class="citation">Sternes and Shimada
(2020)</span> points to interesting patterns in shark body-shape,
neither study explicitly studied the role habitat—that is, pelagic or
benthic—had in influencing the evolution body-shape evolution. In
particular, <span class="citation">Sternes and Shimada (2020)</span>
merely observed that their two groups with different body shapes often
varied in their habitat preference. In addition, <span
class="citation">Sternes and Shimada (2020)</span> made no explicit,
quantitative phylogenetic test of their patterns. This leaves open an
important question: <strong>Do pelagic and benthic sharks have
significantly different body shape?</strong> Subsequently, we might also
ask, <strong>Has the morphology of pelagic and benthic sharks evolved at
the same rate?</strong> These are the questions we will take on in this
project.</p>
<p>To answer such a question requires that we essentially recapitulate
the <span class="citation">Sternes and Shimada (2020)</span> body-shape
data set using landmark-based <a
href="https://en.wikipedia.org/wiki/Morphometrics#Landmark-based_geometric_morphometrics">geometric
morphometrics</a>. This is a relatively straightforward, if tedious,
method of capturing shape variation in a group of interest. For this,
you’ll digitize 14 homologous landmarks on images of some 270 species of
sharks. Using the best practices in the field of geometric morphometrics
to align our landlandmakrs and extract shape information, we’ll assess
morphological disparity in a phylogenetic frame work between pelagic and
benthic sharks.</p>
</div>
<div id="methods" class="section level1">
<h1>Methods</h1>
<div id="image-acquisition" class="section level2">
<h2>Image acquisition</h2>
<p>A few decades ago, a morphometric project like this would require
visiting museums collections from around the world to take images of the
specimens and species of interest. In fact, this was a major part of
Prof. Kenaley’s PhD thesis, which included trips to natural history
collections in France, Denmark, Great Britain, Taiwan, Japan, Monaco,
New Zealand, and Australia. Ahhhhhh, the good ol’ days. Fortunately (or
unfortunately), travel to such wonderful places may be avoided several
resources exist that contain a trove of images of fish-like
vertebrates.</p>
<p>We’ll make use of shark images in the <a
href="https://www.fishbase.us/">FishBase</a> catalog: a clearing house
of all things fish that has amassed tons of data from the scientific
literature. Prof. Kenaley used the <code>rfishbase</code> package to rip
images of scientific drawings of sharks. This resulted in a cache of
images for the ~220 species included in a recent phylogeny of the
elasmobranchs (<span class="citation">Stein et al. (2018)</span>).</p>
</div>
<div id="species-list-and-phylogeny" class="section level2">
<h2>Species list and phylogeny</h2>
<p>The 200+ species we’ll analyze in the study were chosen because they
had decent images on FishBase and cover the nooks and crannies of the
<span class="citation">Stein et al. (2018)</span> phylogeny, giving a
broad sample of the group. As we move on, we’ll use <a
href="data/shark_data.csv">this table</a> to keep track of images we
need to landmark, the species the images belongs to, and the habitat for
each. Let’s load these data now so we have it handy.</p>
<pre class="r"><code>library(tidyverse)</code></pre>
<pre><code>## ── Attaching core tidyverse packages ──────────────────────── tidyverse 2.0.0 ──
## ✔ dplyr     1.1.1     ✔ readr     2.1.4
## ✔ forcats   1.0.0     ✔ stringr   1.5.0
## ✔ ggplot2   3.4.1     ✔ tibble    3.2.1
## ✔ lubridate 1.9.2     ✔ tidyr     1.3.0
## ✔ purrr     1.0.1     
## ── Conflicts ────────────────────────────────────────── tidyverse_conflicts() ──
## ✖ dplyr::filter() masks stats::filter()
## ✖ dplyr::lag()    masks stats::lag()
## ℹ Use the conflicted package (&lt;http://conflicted.r-lib.org/&gt;) to force all conflicts to become errors</code></pre>
<pre class="r"><code>shark_dat &lt;- read_csv(&quot;shark_data.csv&quot;)</code></pre>
<pre><code>## Rows: 91 Columns: 3
## ── Column specification ────────────────────────────────────────────────────────
## Delimiter: &quot;,&quot;
## chr (3): sp, file_name, habitat
## 
## ℹ Use `spec()` to retrieve the full column specification for this data.
## ℹ Specify the column types or set `show_col_types = FALSE` to quiet this message.</code></pre>
<p>Because we’ll be comparing body shape in sharks, this project is by
definition a comparative one and we’ll need a phylogeny for the group.
Fortunately a relatviely recent phylogenomic analysis of over 500
species of elasmobranchs (sharks + batoids) has been published by <span
class="citation">Stein et al. (2018)</span>. Unfortunately, there is
some phylogenetic discordance in this study. Specifically, when
researchers analyze phylogenetic relationships, they typically repeat
their analysis in a process known as <a
href="https://en.wikipedia.org/wiki/Computational_phylogenetics#Evaluating_tree_support">bootstrapping</a>.
In phylogenetics, bootstrapping is conducted using the columns of the
character matrix. Each pseudoreplicate contains the same number of
species (rows) and characters (columns) randomly sampled from the
original matrix, with replacement. A phylogeny is reconstructed from
each pseudoreplicate, with the same methods used to reconstruct the
phylogeny from the original data. This often results in a data set of
trees that have conflicting relationships—that is, phylogenetic
uncertainty—and this was the case for the <span class="citation">Stein
et al. (2018)</span> study. We’ll have to account for this in our
analysis. But, more later on . . .</p>
<p>In any case, we’ve used the taxa studied in this paper to find images
in FishBase and we’ll use <a href="data/shark_trees.nex">a tree file</a>
representing phylogeny for our comparative analysis.</p>
<center>
<p><br> <img
src="https://media.springernature.com/lw685/springer-static/image/art%3A10.1038%2Fs41559-017-0448-4/MediaObjects/41559_2017_448_Fig1_HTML.jpg?as=webp"
style="width:50.0%" /> <br></p>
The phylogenetic relationships of the elasmobranchs according the <span
class="citation">Stein et al. (2018)</span>.
</center>
<p><br></p>
</div>
<div id="geometric-morphometrics" class="section level2">
<h2>Geometric morphometrics</h2>
<p>As we learned in class and what you’ve been reading, researchers
interested in comparing shape change between species often use
landmark-based geometric morphometrics. A geometric morphometric study
such as ours generally follow these steps:</p>
<ol style="list-style-type: decimal">
<li>Data collection: select landmarks of interest, usually through
digitization of specimen images.</li>
<li>Data standardization: Make landmarks comparable across all
specimens, usually through superimposition.</li>
<li>Analysis: choose a statistical approach appropriate to the
questions.</li>
<li>Interpretation: use the outcome of the statistical analysis to
assess the original questions.</li>
</ol>
<div id="digitization-and-landmarking" class="section level3">
<h3>Digitization and landmarking</h3>
<p>Our shape analysis will be based on 14 landmarks placed on images of
shark bodies of our ~220 species. To landmarks on the body, we’ll follow
this generalize workflow below in the image analysis program FIJI (FIJI
is just imageJ).</p>
<br>
<center>
<img src="shark_points.png" style="width:70.0%" /> <br>
</center>
<p><br></p>
<p>You and your team should will be assigned about 20 of the ~220
species to outline. Images of these species can be accessed through <a
href="https://drive.google.com/drive/folders/1TYAzcMcH0u64OEqsuc61W97zIL7qLGWU?usp=sharing">this
directory</a>. First choose one of your species to work on. For each
species, start with what you think is the best image, that is, it has a
scale bar, the wings are not damaged, and the hindwing is most exposed.
Open that best image in Fiji.</p>
<p>After opening the photo, follow the steps below</p>
<ol style="list-style-type: decimal">
<li><p>Open a image.</p></li>
<li><p>Select the multipoint tool.</p></li>
<li><p>Digitize the 14 landmarks you see above.</p></li>
<li><p>Measure the XY positions of these points
(<code>cmd/ctrl+m</code>).</p></li>
<li><p>Select the macro tab and run the macro
(<code>cmd/ctrl+r</code>).</p></li>
<li><p>Save the results to the appropriate directory.</p></li>
<li><p>Close the results and image tab (don’t save the image).</p></li>
</ol>
This process is demonstrated here. <br> <br> <br>
<center>
<iframe width="460" height="315" src="https://www.youtube.com/embed/-yZu3FkJLzU" frameborder="0" allowfullscreen>
</iframe>
</center>
<p><br></p>
<p><strong>Note:</strong> For the first image you outline, you will have
to open the macro editor (“Plugins” <span
class="math inline">\(\rightarrow\)</span> “New” <span
class="math inline">\(\Macro\)</span>). Paste the script command from
below into the editor and save this new macro to an appropriate
directory, perhaps your cloned repo. You’ll want to access this later
(“Plugins” <span class="math inline">\(\rightarrow\)</span> “Edit”) so
that you automate this process.</p>
<pre><code>f = getTitle();
dir=getDirectory(&quot;Choose where to save data &quot;);
selectWindow(&quot;Results&quot;);
saveAs(&quot;Results&quot;, dir+f+&quot;_.csv&quot;);</code></pre>
<p>This script saves the results as a <em>tab-separated</em> text file
with filname as the image file name. <strong>This is super
important!!</strong> and how we’ll link the landmarks to a species.</p>
<p>After your team has completed landmarking the assigned species,
upload your text files to <a
href="https://drive.google.com/drive/u/1/folders/1-vbmHBqKywxq3Wxd8iA8Awf38xz4Glk5">this
directory</a>. Once all the data appear here, Prof. Kenaley will
evaluate the quality of the landmarks and do some post-processing if
required (i.e., file-name checks, quality, etc.). You can then download
<a href="data/class_out_data_f23.zip">these evaluated data</a> for shape
analysis.</p>
</div>
<div id="superimposition" class="section level3">
<h3>Superimposition</h3>
<p>After digitizing landmarks, we must remove the distracting variations
of size, orientation, and position that can vary between specimens and
thus add noisey variables in analysis of shape. This can be mitigated by
using a superimposition method. Generalized procrustes analysis (GPA) is
the superimposition method of choice nowadays. GPA removes the variation
of size, orientation, and position by superimposing the landmarks in a
common coordinate system. The landmarks for all specimens are optimally
translated, rotated, and scaled based on a least-squared estimation. The
first step is translation and rotation to minimize the squared and
summed differences (squared Procrustes distance) between landmarks on
each specimen. Then the landmarks are individually scaled to the same
units of centroid size. Centroid size is the square root of the sum of
squared distances of the landmarks in configuration to their mean
location. The translation, rotation, and scaling bring the landmark
configurations for all specimens into a common coordinate system so that
the only differing variables are based on shape alone. The new
superimposed landmarks can now be analyzed in multivariate statistical
analyses</p>
<p><img src="gpa.png" /></p>
<p>To perform a Procrustes superimposition on our landmark data—along
with many other analyses—we will turn the the <code>geomorph</code>
package. Please install and load <code>geomorph</code>. But, before we
do, let’s download and read in about 90 or so landmark files that
Prof. Kenaley produceds ahead of this project. You can find these
example files <a href="data/sharks/shark_points.zip">here</a>. You’ll
have to download this compressed directory and unzip it to your project
directory. You’ll also have to download and install the
<code>abind</code> package as well.</p>
<p>To get started with superimposition, let’s load the packages
<code>geomorph</code> and <code>abind</code>. When you unzip the
<code>shark_points.zip</code> directory to you project directory, you
have a folder named <code>shark_points</code>. Our first operation will
be to list the files in that directory and then read them in, all at
once.</p>
<pre class="r"><code>library(geomorph)</code></pre>
<pre><code>## Loading required package: RRPP</code></pre>
<pre><code>## Loading required package: rgl</code></pre>
<pre><code>## Loading required package: Matrix</code></pre>
<pre><code>## 
## Attaching package: &#39;Matrix&#39;</code></pre>
<pre><code>## The following objects are masked from &#39;package:tidyr&#39;:
## 
##     expand, pack, unpack</code></pre>
<pre class="r"><code>library(abind)

f &lt;- list.files(&quot;shark_points&quot;,full.names = T,pattern=&quot;.csv&quot;)

xy &lt;- read_csv(f,id=&quot;file&quot;) %&gt;% 
  select(file,X,Y) %&gt;% 
  mutate(file_name=gsub(&quot;*_.csv&quot;,&quot;&quot;,basename(file))) %&gt;% 
  left_join(shark_dat) </code></pre>
<pre><code>## New names:
## • `` -&gt; `...1`</code></pre>
<pre><code>## Rows: 1274 Columns: 15
## ── Column specification ────────────────────────────────────────────────────────
## Delimiter: &quot;,&quot;
## dbl (14): ...1, Area, Mean, Mode, X, Y, XM, YM, Perim., Circ., Slice, AR, Ro...
## 
## ℹ Use `spec()` to retrieve the full column specification for this data.
## ℹ Specify the column types or set `show_col_types = FALSE` to quiet this message.
## Joining with `by = join_by(file_name)`</code></pre>
<p>Here we used <code>list.files()</code> with
<code>pattern=pattern=".csv"</code> to find all the files in the
<code>shark_points</code> directory. We also set
<code>full.names = T</code> so that the full file path is saved to the
object <code>f</code>.</p>
<pre class="r"><code>head(f)</code></pre>
<pre><code>## [1] &quot;shark_points/Acnig_u0.gif_.csv&quot; &quot;shark_points/Alpel_u0.gif_.csv&quot;
## [3] &quot;shark_points/Alsup_u0.gif_.csv&quot; &quot;shark_points/Alvul_u0.gif_.csv&quot;
## [5] &quot;shark_points/Apbru_u1.gif_.csv&quot; &quot;shark_points/Aplau_u0.gif_.csv&quot;</code></pre>
<p>This made reading all the files and adding them to one tibble in a
single operation possible with <code>read_csv()</code>. Notice we added
a <code>file</code> column to the tibble with <code>id="file"</code> and
then selected only the columns <code>file</code>, <code>X</code>,
<code>Y</code>. This operation is saved to an object logically named
<code>xy</code>.</p>
<p>For landmarks to used in <code>geomorph</code>, they must be stored
as a 3-dimensional array, that is, essentially a list of 2D data.frames,
where the position the <span class="math inline">\(i^{th}\)</span>
matrix is in the <span class="math inline">\(i^{th}\)</span> index of
the third dimension, i.e., <code>array[,,i]</code>. To massage our large
tibble into such an array, we have to split the tibble into a bunch of
lists for this, we can simply run a <code>for</code> loop, breaking the
data up into a list of data frames according to species. Notice how we
use a <a
href="https://www.nbshare.io/notebook/429754141/How-To-Use-Grep-In-R/"><code>gsub()</code>
operation</a> to create a new column <code>file_name</code> that removes
“.csv” from the file name data. Splitting by this, will leave used with
a list of lists as long as all our files. We then use
<code>abind</code>’s <code>abind()</code> function to bind the list of
data.frames in <code>ldk_l</code> into an array, with each data.frame
taking a position in the third dimension. This array is saved in the
object <code>ldk</code>. Lastly, we’ll specify that the names of this
third dimension should be the <code>file_name</code> minus the “.csv”
extension. <strong>Note: this is how we’ll retrieve species
information.</strong></p>
<pre class="r"><code>ldk_l &lt;- list()

shark_sp &lt;- xy$sp %&gt;% unique

for(i in shark_sp){
  ldk_l[[i]] &lt;- xy %&gt;% 
    filter(sp==i) %&gt;% 
    select(X,Y)
    data.frame
}


ldk &lt;-  abind(ldk_l, along=3)

dimnames(ldk)[[3]] &lt;-names(ldk_l)

ldk[,,1:3] #the first few </code></pre>
<pre><code>## , , Aculeola nigra
## 
##           X     Y
##  [1,]  39.5 185.5
##  [2,] 260.0 141.5
##  [3,] 292.0 136.5
##  [4,] 293.5 160.0
##  [5,] 464.0 172.5
##  [6,] 596.0 162.5
##  [7,] 522.0 222.0
##  [8,] 454.5 194.5
##  [9,] 380.0 198.5
## [10,] 341.0 214.5
## [11,] 209.0 202.5
## [12,] 180.0 194.5
## [13,] 100.5 202.5
## [14,]  74.5 209.0
## 
## , , Alopias pelagicus
## 
##           X       Y
##  [1,]  38.0 166.000
##  [2,] 162.5 141.500
##  [3,] 186.5 101.500
##  [4,] 192.5 138.500
##  [5,] 295.5 152.500
##  [6,] 606.5  94.500
##  [7,] 321.5 206.500
##  [8,] 300.5 181.500
##  [9,] 249.5 183.500
## [10,] 228.5 187.500
## [11,] 135.5 185.500
## [12,] 103.5 188.500
## [13,]  69.0 179.833
## [14,]  59.5 180.333
## 
## , , Alopias superciliosus
## 
##           X     Y
##  [1,]  38.5 173.5
##  [2,] 221.5 135.0
##  [3,] 257.0  89.5
##  [4,] 261.5 136.0
##  [5,] 364.0 152.0
##  [6,] 604.5  71.5
##  [7,] 389.5 205.5
##  [8,] 363.0 177.5
##  [9,] 310.0 184.0
## [10,] 282.5 197.0
## [11,] 168.5 194.0
## [12,] 138.5 196.5
## [13,]  82.5 190.5
## [14,]  67.5 189.0</code></pre>
<p>With a proper array in place as <code>ldk</code>, we’re now free to
move on the superimposition with the <code>geomorph</code> function
<code>gpagen()</code>. We’ll do this saving the output to
<code>ldk_al</code> (for landmark alignment) and plot the the 14 aligned
and superimposed landmarks for all the example specimens.</p>
<pre class="r"><code>ldk_al &lt;- gpagen(ldk)</code></pre>
<pre><code>## 
## Performing GPA
##   |                                                                              |                                                                      |   0%  |                                                                              |==================                                                    |  25%  |                                                                              |===================================                                   |  50%  |                                                                              |====================================================                  |  75%  |                                                                              |======================================================================| 100%
## 
## Making projections... Finished!</code></pre>
<pre class="r"><code>plot(ldk_al)</code></pre>
<p><img src="mod5_proj_files/figure-html/gpa-1.png" width="672" /></p>
</div>
<div id="analysis-part-i-a-morphospace" class="section level3">
<h3>Analysis Part I: A Morphospace</h3>
<p>In geometric morphometric analysis, researchers first turn to
describing the axes of shape variation.Indeed, this is what <span
class="citation">Sternes and Shimada (2020)</span> did to produce the
figure above. This and what we’re after is often termed a <a
href="https://en.wiktionary.org/wiki/morphospace">morphospace</a>, a
representation of shape the study group does and does not have.</p>
<p>To develop such a thing, biologist usually turn to Principal
Components Analysis (PCA) whereby the very complicated suite of
variables is distilled into a set of fewer variables that describe the
important variance among the original variables. The coordinates
contained in our <code>ldk_al</code> object describe dozens of
variables: scaled distances between all the landmarks. So we have many
variable to consider and this becomes, very quickly, a serious <a
href="https://en.wikipedia.org/wiki/Multivariate_statistics">multivariate
problem</a>. PCA is suited to multivariate analysis because it reduces
the complexity of the data, taking many variables and distilling the
into just a few. At its core, PCA produces a series of regressions, or
vectors, that pass through the values of the original variables. This
operation is performed iteratively, with the first vector (the first
principal component, PC1), drawn through the variable space that
accounts for the most variance in the data set. After the combination of
variables forming PC1 is removed, a subsequent vector (PC2) is drawn
through another unique variable space that describes the second most
amount of variance. This continues until most of the variance is
captured, resulting in scores of PCs. The first two or three, however,
are usually enough to describe a health amount of variance, say 90% or
more.</p>
<p>To perform PCA, we’ll use <code>geomorph</code>s
<code>gm.prcomp()</code> function like so. Plotting reveals our first
result.</p>
<pre class="r"><code>pca &lt;- gm.prcomp(ldk_al$coords)
plot(pca)</code></pre>
<p><img src="mod5_proj_files/figure-html/pca-1.png" width="672" /></p>
<p>Wow, now we’re getting close to the <span class="citation">Sternes
and Shimada (2020)</span> results and answering, at least in part, one
of our questions: <strong>Does habitat result in different body
shapes</strong>. But let’s break down what this plot means first. By
default, when we plot a PCA object, the first two PCs are plotted, PC1
on the x axis, PC2 on the y. Each point represents the PC scores for a
specimen in the data set. Notice that the amount of variance explained
by our PCs is given and, as expected, PC1 contains most of the variance
(54%) and PC2 considerably less less (19%). If we plot again, but
specify a different combination of PCs, say 1 and 3, we get a different
plot, of course. Notice the variance explained by PC3 is predictably
less than that of PC2.</p>
<pre class="r"><code>plot(pca,1,3)</code></pre>
<p><img src="mod5_proj_files/figure-html/pc2-1.png" width="672" /></p>
<p>OK, so based on our PC plots, it looks as if we have two groupings,
just like <span class="citation">Sternes and Shimada (2020)</span>. But,
do these groups correspond to largely pelagic vs. benthic species? It’s
hard to tell. We don’t really know because we don’t know which species
belong to each point in our plots, nor their habitat.</p>
<p>Notice, however, that the names of the coordinates in
<code>ldk</code> were pegged in a previous operation to the species
names.</p>
<pre class="r"><code>head(dimnames(ldk)[[3]])</code></pre>
<pre><code>## [1] &quot;Aculeola nigra&quot;         &quot;Alopias pelagicus&quot;      &quot;Alopias superciliosus&quot; 
## [4] &quot;Alopias vulpinus&quot;       &quot;Apristurus brunneus&quot;    &quot;Apristurus laurussonii&quot;</code></pre>
<p>These values were maintained throughout the superimpositon step and
in the PCA operation. The coordinates plotted in the in PCA plots are
contained in a matrix named <code>x</code> in our <code>pca</code>
object. Notice the row names:</p>
<pre class="r"><code>pca$x %&gt;% data.frame %&gt;% select(Comp1,Comp2) %&gt;% head</code></pre>
<pre><code>##                              Comp1       Comp2
## Aculeola nigra          0.06258856  0.03252442
## Alopias pelagicus       0.10109723 -0.31064490
## Alopias superciliosus   0.03327460 -0.17242372
## Alopias vulpinus        0.10639978 -0.30795668
## Apristurus brunneus    -0.13126591 -0.01146547
## Apristurus laurussonii -0.12314775 -0.06309869</code></pre>
<p>So, we have the species names from whence the data, both land mark
and PCA, came from. Fortunately, Professor Kenaley assembled a table
containing of all the the species for landmarking, and the species and
habitat for each. You can download <a href="shark_data.csv">this
file</a> and read in the data.</p>
<p>What’s left to do in exploring if benthic and pelagic sharks occupy
different morphospace is to combine this information with the PC scores
in <code>pca$x</code>. This can be done thusly:</p>
<pre class="r"><code>PCA &lt;- pca$x %&gt;% 
  data.frame %&gt;% 
  select(Comp1:Comp4) %&gt;% 
  mutate(sp=rownames(pca$x)) %&gt;% 
  left_join(shark_dat)</code></pre>
<pre><code>## Joining with `by = join_by(sp)`</code></pre>
<pre class="r"><code>head(PCA)</code></pre>
<pre><code>##         Comp1       Comp2      Comp3        Comp4                     sp
## 1  0.06258856  0.03252442 0.03356723 -0.019825631         Aculeola nigra
## 2  0.10109723 -0.31064490 0.11745778 -0.016213252      Alopias pelagicus
## 3  0.03327460 -0.17242372 0.13482941  0.020733446  Alopias superciliosus
## 4  0.10639978 -0.30795668 0.08076650  0.003565676       Alopias vulpinus
## 5 -0.13126591 -0.01146547 0.21487943 -0.048498099    Apristurus brunneus
## 6 -0.12314775 -0.06309869 0.01389941  0.003318482 Apristurus laurussonii
##      file_name habitat
## 1 Acnig_u0.gif benthic
## 2 Alpel_u0.gif pelagic
## 3 Alsup_u0.gif pelagic
## 4 Alvul_u0.gif pelagic
## 5 Apbru_u1.gif benthic
## 6 Aplau_u0.gif benthic</code></pre>
<p>With that over with, we can dispence with the ugly PCA plots above
and use <code>ggplot</code> to visualize our results with repsect to
habitat.</p>
<pre class="r"><code>PCA %&gt;% 
  ggplot(aes(Comp1,Comp2,col=habitat))+geom_point()</code></pre>
<p><img src="mod5_proj_files/figure-html/ggpca-1.png" width="672" /></p>
<pre class="r"><code>PCA %&gt;% 
  ggplot(aes(Comp1,Comp3,col=habitat))+geom_point()</code></pre>
<p><img src="mod5_proj_files/figure-html/ggpca-2.png" width="672" /></p>
<p>Voila! For morphospace described by PCs 1, 2, and 3, it appeears that
pelagic species inhabit a high PC1 score and never a low PC1 score.</p>
<p>This is wonderful, but you may be asking, “What is PC1 (or any other)
describing?” Our PCs points are abstractions of shape, but they don’t
represent explicitly what shape that is. To get a sense of what our PCs
represent in terms of shape values, we can turn to thin-plate spline
analysis.</p>
<pre class="r"><code>max_PC1 &lt;- which.max(pca$x[,1])
min_PC1 &lt;- which.min(pca$x[,1])
max_PC2 &lt;- which.max(pca$x[,2])
min_PC2 &lt;- which.min(pca$x[,2])
M &lt;- ldk_al$consensus

par(mfrow=c(2,2))
plotRefToTarget(M, ldk_al$coords[,,min_PC1], mag = 1,method=&quot;vector&quot;,label = T)
title(&quot;min PC1 warp&quot;,cex=0.6)

plotRefToTarget(M, ldk_al$coords[,,max_PC1], mag = 1,method=&quot;vector&quot;,label = T)
title(&quot;max PC1 warp&quot;,cex=0.6)

plotRefToTarget(M, ldk_al$coords[,,min_PC2], mag = 1,method=&quot;vector&quot;,label = T)
title(&quot;min PC2 warp&quot;,cex=0.6)

plotRefToTarget(M, ldk_al$coords[,,max_PC2], mag = 1,method=&quot;vector&quot;,label = T)
title(&quot;max PC3 warp&quot;,cex=0.6)</code></pre>
<p><img src="mod5_proj_files/figure-html/rw-1.png" width="672" /></p>
</div>
<div id="analysis-part-ii" class="section level3">
<h3>Analysis: Part II</h3>
<p>Our PCA analysis reveals an interesting pattern. This, however, is
not quantitative assessment <em>per se</em>, merely a visual and
qualitative result and not dissimilar to the qualitative results
reported by <span class="citation">Sternes and Shimada (2020)</span>.
Several quantitative frameworks for assessing differences in shape
exist. The <code>geomorph</code> package contains several, but we’ll use
just one: morphological disparity. The function
<code>morphol.disparity()</code> will do this for us by making pairwise
comparisons of residuals of a linear fit to identify differences among
groups. It’s essentially an ANOVA for shape. The statistical
significance of the calculated Procrustes variances between the
different growth stages is assessed using a randomized residual
permutation, that is random set residuals from all groups are chosen in
each iteration.</p>
<p><code>morphol.disparity()</code> requires a special data structure, a
<code>geomorph</code> <code>data.frame</code> and this can be
constructed using the <code>geomorph.data.frame()</code> function. It
takes the landmark data and other variables of interest (i.e., in our
case, habitat). First, we’ll establish the special data frame using our
aligned landmarks and habitat data from the <code>PCA</code> tibble.</p>
<pre class="r"><code>gdf &lt;- geomorph.data.frame(ldk_al,
                             habitat=PCA$habitat
)</code></pre>
<p>To you use the <code>morphol.disparity()</code> function, we then
need to establish a formula describing the model to use and tell the
function which groups the model should include. This is encoded with
<code>coords ~ 1</code> which indicates we’ll look at the differences in
means and <code>groups= ~ habitat</code>, indicating that we’ll evaluate
means between the habitat groups. <code>data = gdf</code> indicates
we’ll use the <code>gdf</code> data frame we established above, while
<code>iter = 10000</code> indicates we’ll be perform 10,000 radom
permutations. Lastly, we just summarize the <code>md</code> object to
return the results</p>
<pre class="r"><code>md &lt;- morphol.disparity(coords ~ 1, groups= ~ habitat, data = gdf, iter = 10000, print.progress = FALSE)

summary(md)</code></pre>
<pre><code>## 
## Call:
## morphol.disparity(f1 = coords ~ 1, groups = ~habitat, iter = 10000,  
##     data = gdf, print.progress = FALSE) 
## 
## 
## 
## Randomized Residual Permutation Procedure Used
## 10001 Permutations
## 
## Procrustes variances for defined groups
##    benthic    pelagic 
## 0.03218948 0.02885998 
## 
## 
## Pairwise absolute differences between variances
##             benthic     pelagic
## benthic 0.000000000 0.003329492
## pelagic 0.003329492 0.000000000
## 
## 
## P-Values
##           benthic   pelagic
## benthic 1.0000000 0.5694431
## pelagic 0.5694431 1.0000000</code></pre>
<p>Notice the the pairwise differences are returned both for the
Procrustes shape variance and the absolute differences along with the p
values. As it turns out, at least for this limited data set of ~90
species, there’s no significant morphological disparity between
groups.</p>
<p>We have a preliminary answer to our first question, “Does habitat use
predict different body shape?”. But this of course could change with a
larger sample of species.</p>
<pre class="r"><code>library(phytools)</code></pre>
<pre><code>## Warning: package &#39;ape&#39; was built under R version 4.2.3</code></pre>
<pre class="r"><code>phy &lt;- readNexus(&quot;data/shark_trees.nex&quot;)

phy2 &lt;- keep.tip(phy,PCA$sp)

  gdf &lt;- geomorph.data.frame(ldk_al,
                             species = PCA$sp,
                             habitat=PCA$habitat,
                             phy=phy2[[1]]
  )
  

library(parallel)  
shark_pgls &lt;- procD.pgls(coords ~ habitat, phy = phy, data = gdf, 
iter = 10000,print.progress = F)


summary(shark_pgls)</code></pre>
<pre><code>## 
## Analysis of Variance, using Residual Randomization
## Permutation procedure: Randomization of null model residuals 
## Number of permutations: 10001 
## Estimation method: Generalized Least-Squares (via OLS projection) 
## Sums of Squares and Cross-products: Type I 
## Effect sizes (Z) based on F distributions
## 
##           Df       SS         MS     Rsq      F        Z Pr(&gt;F)
## habitat    1 0.000129 0.00012941 0.00405 0.3621 -0.83632  0.806
## Residuals 89 0.031805 0.00035736 0.99595                       
## Total     90 0.031935                                          
## 
## Call: procD.lm(f1 = coords ~ habitat, iter = iter, seed = seed, RRPP = TRUE,  
##     SS.type = SS.type, effect.type = effect.type, int.first = int.first,  
##     Cov = Cov, data = data, print.progress = print.progress)</code></pre>
</div>
<div id="analysis-part-iii-evolution-in-parallel"
class="section level3">
<h3>Analysis Part III: Evolution in Parallel</h3>
<p>Our secondary question was, if you remember, was, “<strong>Has the
body shape of pelagic and benthic sharks evolved at the same
rate?</strong>”. This is a much different question than the first. We
want to know, overall, if the shape of lineages that invaded a
particular habitat has evolved faster or slower than those invading
another habitat. For this we’ll turn to the <code>geomorph</code>
package once again, in particular, its <code>compare.evol.rates()</code>
function. This function calculates net rates of shape evolution for
groups of species on a phylogeny from a set of aligned landmarks. The
approach is based between-species differences in morphospace after
phylogenetic transformation. In other words, after correcting for
phylogenetic relatedness, it compares relative rates of shape
evolution.</p>
<p>Because we’re correcting for phylogeny in this analysis, we need the
<a href="data/shark_trees.nex">tree from <span class="citation">Stein et
al. (2018)</span> mentioned above</a>. Once this tree file is
downloaded, let’s load read it in and store it to an object named
<code>phy</code>. Notice you’ll need the phytools package and the
function <code>readNexus()</code>.</p>
<pre class="r"><code>library(phytools)
phy &lt;- readNexus(&quot;data/shark_trees.nex&quot;)
length(phy)</code></pre>
<pre><code>## [1] 500</code></pre>
<p>Next let’s, restrict our tree file to those species in our data set.
For this, we’ll used <code>ape</code>’s <code>keep.tip()</code> which
keeps only the tips in a vector, in this case, the species in our PCA
and landmakr analysis.</p>
<pre class="r"><code>phy2 &lt;- keep.tip(phy,PCA$sp)</code></pre>
<p>To undertake the analysis with <code>compare.evol.rates()</code>, we
must specify <code>A</code>, the aligned coordinates in a lankmark array
(<code>ldk_al$coord</code>), a tree (more in just a bit), and a named
vector of groups descriping to which group each species in the array
belongs.</p>
<p>Let’s establish the grouping vector by passing the
<code>habitat</code> values in our <code>PCA</code> tibble to an object
named <code>gp_hab</code> and then give the names of this vector the
<code>species</code> values. Then we run the function.</p>
<pre class="r"><code>gp_hab &lt;- PCA$habitat
names(gp_hab) &lt;- PCA$sp

er &lt;- compare.evol.rates(A=ldk_al$coords, phy=phy2[[1]], method=&quot;simulation&quot;,gp=gp_hab,print.progress = F)

print(er)</code></pre>
<pre><code>## 
## Call:
## 
## 
## Observed Rate Ratio: 1.5654
## 
## P-value: 0.002
## 
## Effect Size: 2.45
## 
## Based on 1000 random permutations
## 
## The rate for group benthic is 1.40377103447055e-05  
## 
## The rate for group pelagic is 8.96721584612147e-06</code></pre>
<p>So it appears the benthic group body shape has evolved 1.6 times
faster than the pelagic group and this is significant
(<em>p=0.002</em>). But, we have a problem. You may have noticed that
when we loaded our tree file, the length of this list of trees is 500.
These are 500 trees produced by <span class="citation">Stein et al.
(2018)</span> using the bootstrapping method mentioned earlier. Many of
these trees are incongruent, that is, they represent different
relationships for the sharks and this, in turn, represents some serious
phylogenetic uncertainty.</p>
<p>Also notice that in our rate analysis above, we used only one tree in
this list (<code>phy=phy2[[1]]</code>). There are 499 other trees we
could have used, perhaps each representing a different phylogenetic
hypothesis. The results of a comparative analysis like this could, and
likely will, vary according to the phylogenetic hypothesis used in it.
So what to do?</p>
<p>This is a common problem faced by comparative biologists and,
fortunately, there seems to be a consensus among us to repeat our
comparative analysis using a random sample of trees in that represent
the phylogenetic uncertainty. By repeating an analysis, we produce a
series of results based on new parameters to our model. In this case,
that new parameter would be a randomly sampled tree. This establishes
what is akin to a <a
href="https://en.wikipedia.org/wiki/Posterior_probability">posterior
sample</a> and from this we can extract a posterior distribution of
estimates, in our case, a distribution of evolutionary rates.</p>
<p>Therefore, we have before us the proposition of repeating out
analysis, let’s say 10 times. To do this, we could randomly sample 10 of
trees and build a posterior distribution of evolutionary rates using a
<code>for</code> loop.</p>
<pre class="r"><code>set.seed(1234)
phy_sample &lt;- sample(phy2,10)

er_l &lt;- list()


t &lt;- system.time(for(i in 1:length(phy_sample)){
 er_l[[i]] &lt;-compare.evol.rates(A=ldk_al$coords, phy=phy_sample[[i]], method=&quot;simulation&quot;,gp=gp_hab,print.progress = F) %&gt;% summary
}
)</code></pre>
<pre><code>## 
## Call:
## 
## 
## Observed Rate Ratio: 1.3766
## 
## P-value: 0.013
## 
## Effect Size: 2.087
## 
## Based on 1000 random permutations
## 
## The rate for group benthic is 1.70947144489729e-05  
## 
## The rate for group pelagic is 1.24184241879944e-05 
## Call:
## 
## 
## Observed Rate Ratio: 1.4821
## 
## P-value: 0.004
## 
## Effect Size: 2.3
## 
## Based on 1000 random permutations
## 
## The rate for group benthic is 1.22496627992883e-05  
## 
## The rate for group pelagic is 8.26504653836789e-06 
## Call:
## 
## 
## Observed Rate Ratio: 1.4711
## 
## P-value: 0.006
## 
## Effect Size: 2.2689
## 
## Based on 1000 random permutations
## 
## The rate for group benthic is 1.146005595824e-05  
## 
## The rate for group pelagic is 7.78996757098907e-06 
## Call:
## 
## 
## Observed Rate Ratio: 1.5114
## 
## P-value: 0.002
## 
## Effect Size: 2.3659
## 
## Based on 1000 random permutations
## 
## The rate for group benthic is 1.29573966236599e-05  
## 
## The rate for group pelagic is 8.57288744816572e-06 
## Call:
## 
## 
## Observed Rate Ratio: 1.498
## 
## P-value: 0.002
## 
## Effect Size: 2.321
## 
## Based on 1000 random permutations
## 
## The rate for group benthic is 1.00978388979575e-05  
## 
## The rate for group pelagic is 6.74097338255856e-06 
## Call:
## 
## 
## Observed Rate Ratio: 1.4521
## 
## P-value: 0.008
## 
## Effect Size: 2.226
## 
## Based on 1000 random permutations
## 
## The rate for group benthic is 1.24563414926804e-05  
## 
## The rate for group pelagic is 8.57837163470097e-06 
## Call:
## 
## 
## Observed Rate Ratio: 1.5742
## 
## P-value: 0.002
## 
## Effect Size: 2.4468
## 
## Based on 1000 random permutations
## 
## The rate for group benthic is 1.15435009156868e-05  
## 
## The rate for group pelagic is 7.33290098688007e-06 
## Call:
## 
## 
## Observed Rate Ratio: 1.4472
## 
## P-value: 0.008
## 
## Effect Size: 2.2407
## 
## Based on 1000 random permutations
## 
## The rate for group benthic is 1.19807638629684e-05  
## 
## The rate for group pelagic is 8.27830517141801e-06 
## Call:
## 
## 
## Observed Rate Ratio: 1.5248
## 
## P-value: 0.002
## 
## Effect Size: 2.3684
## 
## Based on 1000 random permutations
## 
## The rate for group benthic is 1.16162453729596e-05  
## 
## The rate for group pelagic is 7.61825539324746e-06 
## Call:
## 
## 
## Observed Rate Ratio: 1.5845
## 
## P-value: 0.002
## 
## Effect Size: 2.4614
## 
## Based on 1000 random permutations
## 
## The rate for group benthic is 1.20582596828107e-05  
## 
## The rate for group pelagic is 7.60999305350617e-06</code></pre>
<p>Here we used <code>seet.seed()</code> to make the
<code>sample()</code> function randomly sample the same set of 10 trees.
This is important so that we can all perform the same exact analyses,
that is, our analyses are repeatable. You’ll notice that the loop uses
each random tree to repeat the analysis and store the summary of the
model (i.e., the rate comparisons) to the list <code>er_l</code>. Notice
also that the entire loop is wrapped in <code>system.time()</code>. This
is done so that, in this preliminary analysis, we can record how much
time the loop took. Printing <code>t</code> reveals that it took 7
seconds. Because our results are stored in a list, we can use
<code>lapply()</code> extract the rate comparisons.
<code>lapply()</code> takes a list and applies a function to. In our
case, we need to encoded a custom function that extracts
<code>sigma.d.gp</code> and <code>sigma.d.ratio</code>, the rate for
each group and rate ratio between them, respectively. For this, we ask
the function to work on every value of <code>er_l</code>, read in as
<code>x</code>, then combine <code>sigma.d.gp</code> and
<code>sigma.d.ratio</code> and store it in <code>er_</code> temporatily.
We then apply a name to the third value “ratio”, and ruturn the vector.
<code>lapply()</code> returns a list, so we combine all elements of it
using <code>do.call()</code>. We apply the row-binding function
<code>rbind()</code> to the list using the pipe and <code>.</code> and
lastly, make the output a data frame.</p>
<pre class="r"><code>er &lt;- lapply(er_l,function(x) {
  er_ &lt;- c(x$sigma.d.gp,x$sigma.d.ratio)
  names(er_)[3] &lt;-&quot;ratio&quot;
  return(er_)
}) %&gt;% do.call(rbind,.) %&gt;% data.frame

head(er)</code></pre>
<pre><code>##        benthic      pelagic    ratio
## 1 1.709471e-05 1.241842e-05 1.376561
## 2 1.224966e-05 8.265047e-06 1.482105
## 3 1.146006e-05 7.789968e-06 1.471130
## 4 1.295740e-05 8.572887e-06 1.511439
## 5 1.009784e-05 6.740973e-06 1.497979
## 6 1.245634e-05 8.578372e-06 1.452064</code></pre>
<p>With a little massaging with <code>pivot_longer()</code>, we can plot
the posterior distribution of rates for each habitat like so.</p>
<pre class="r"><code>er %&gt;% 
  pivot_longer(benthic:pelagic) %&gt;% 
  ggplot(aes(name,value)) + geom_boxplot()</code></pre>
<p><img src="mod5_proj_files/figure-html/ploter-1.png" width="672" /></p>
<p>Indeed it appears that when we estimate evolutionary rates for 10
random trees, accounting for uncertainty in species relationships,
pelagic species evolve slower than benthic ones.</p>
<p>Now, this is all well and good. But, most comparative biologists
would argue 10 random trees does not a posterior make. We should
probably sample many more or, perhaps use the entire set of trees.</p>
<p>What’s to stop us? Well, if we shoot for the stars and use all 500
trees in 500 separate analyses, we’d be looking running a for loop 50
times longer than 10 trees took (roughly 350 s or 6 minutes, see figure
below). In the world of comparative methods, this isn’t such a long
time, but remember, this is a preliminary analysis of only 90 or so of
the 220 we hope to include. The figure below indicates that including
all 220 species and using all 500 trees would take ~2500 s or about 40
minutes!!!</p>
<p><br></p>
<img src="mod5_proj_files/figure-html/timescale-1.png" width="672" />
<br>
<center>
The polynomial relationship between time and number of taxa in
comparative analysis, in this case computing evolutionary rates of shape
change with <code>geomorph</code>’s <code>compare.evol.rates()</code>.
</center
<br>
<p><br></p>
<p>We should, therefore, be prepared for the long haul. But, hold on. I
don’t know about you, but on most days, I don’t have 40 minutes to
spare, even for very interesting comparative analyses. Fortunately, to
avoid 100s of iterations of the same operations on different data, we
can make use of a very common work-around in scientific computing: <a
href="https://en.wikipedia.org/wiki/Parallel_computing">parallelization</a>,
the simultaneous execution of different pieces of a larger computation
across multiple computing processors or cores. If fact, this sort of
problem is considered <a
href="https://en.wikipedia.org/wiki/Embarrassingly_parallel">embarrassingly
parallel</a>, that is, it is so very easy to split this task in the many
parallel subtasks. Specifically, we have 500 identical tasks, we need
only to break these up into subtasks that run in parallel. That’s what
we’ll do.</p>
<p><code>R</code> has several packages for parallel computing. Perhaps
the most used is <code>parallel</code>, so cleverly named. Let’s work on
an example of how a task could be parallelized using
<code>parallel</code>’s functions.</p>
<pre class="r"><code>x &lt;- rnorm(1e6)</code></pre>
<!-- hab_names <- c(PCA$habit[which(gp_hab==0)] %>% unique, -->
<!--                PCA$habit[which(gp_hab==1)]%>% unique) -->
<!-- er_fit <- list() -->
<!-- for(i in 1:length(phy2)){ -->
<!--  er_fit[[i]] <-compare.evol.rates(A=ldk_al$coords, phy=phy2[[i]], method="simulation",gp=gp_hab,print.progress = F) -->
<!-- } -->
<!-- plot(er_l[[1]]) -->
<!-- er <- lapply(er_fit,summary) -->
<!-- er <- lapply(er,function(x) { -->
<!--   er <- c(x$sigma.d.gp,x$sigma.d.ratio) -->
<!--   names(er) <- c(hab_names,"ratio") -->
<!--   return(er) -->
<!-- }) %>% do.call(rbind,.) %>% data.frame -->
<!-- er %>%  -->
<!--   pivot_longer(benthic:pelagic) %>%  -->
<!--   ggplot(aes(value,fill=name)) + geom_histogram() -->
<!-- library(parallel) -->
<!-- er_fun <- function(x){compare.evol.rates(A=ldk_al$coords, phy=x, method="simulation",gp=gp_hab,print.progress = F)} -->
<!-- er_l <- mclapply(phy2,er_fun,mc.cores = detectCores()-2) -->
<!-- er <- lapply(er_fit,summary) -->
<!-- er <- lapply(er,function(x) { -->
<!--   er <- c(x$sigma.d.gp,x$sigma.d.ratio) -->
<!--   names(er) <- c(hab_names,"ratio") -->
<!--   return(er) -->
<!-- }) %>% do.call(rbind,.) %>% data.frame -->
<!-- er %>%  -->
<!--   pivot_longer(benthic:pelagic) %>%  -->
<!--   ggplot(aes(value,fill=name)) + geom_histogram() -->
<!-- t.test(value~name, -->
<!-- er %>%  -->
<!--   pivot_longer(benthic:pelagic)  -->
<!-- ) -->
</div>
</div>
</div>
<div id="project-report" class="section level1">
<h1>Project Report</h1>
<p>t-test to posterior</p>
<!-- # Project Report -->
<!-- Please submit your report to your team GitHub repository as an .Rmd document with HTML output that addresses the following: -->
<!--   * Are the rates of wing-shape evolution different between the hind- and forewing? By how much?  -->
<!--   <!-- * Have major lineages of the Lepidoptera undergone significant shifts in evolutionary rate?  -->
<p>–&gt; <!--   * Are hind- and forewing shapes correlated? -->
<!--   * What inferences can you make based on shape correlations and flight control? --></p>
<!-- Please be sure to answer these questions based on both PC1 and PC2 shape scores. -->
<!-- In answering your question, your .Rmd should include the following components: -->
<!--    * A YAML header that specifies HTML output, the authors, and a bibliograph named "BIOL3140.bib". **Submit this bibliography as well!** -->
<!--    * Sections including an introduction, methods, results, discussion, author contributions, and references. Make sure that each, aside from the references, includes one to two short paragraphs. Specifially: -->
<!--       + Introduction: Frame the questions, indicating why they are important, what background work has been done in this realm, and how you will answer them. **Please include at least one reference to support the summary of previous work.** Note: this can be done easily by refiguring the introduction to this project report. -->
<!--       + Methods: Explicitly state how you answered the questions, including a narrative of all the anlyses both qualitative and quantitative. -->
<!--       + Results: Include any appropriate figures or tables and a narrative of the main results that are important to answering the questions. -->
<!--       + Discussion: Succinctly declare how the results relate to the question and how they compare to previous work devoted to the topic. In addition, be sure to comment on the importance of your findings to the broader topic at hand. **Please include at least one reference to another relevant study.** Note: circling back to the introductions, both to this project description and yours, will be helpful here. -->
<!--       + Author contributions: Briefly outline what each team member contributed to the project. -->
<!-- Project reports should be uploaded by 11:59 PM on Sunday, October 29th. -->
<!-- # References -->
<div id="refs" class="references csl-bib-body hanging-indent">
<div id="ref-grogan2012origin" class="csl-entry">
Grogan, Eileen D, Richard Lund, and Emily Greenfest-Allen. 2012.
<span>“The Origin and Relationships of Early Chondrichthyans.”</span>
<em>Biology of Sharks and Their Relatives</em> 2: 3–29.
</div>
<div id="ref-lauder2015swimming" class="csl-entry">
Lauder, George V, and Valentina Di Santo. 2015. <span>“Swimming
Mechanics and Energetics of Elasmobranch Fishes.”</span> In <em>Fish
Physiology</em>, 34:219–53.
</div>
<div id="ref-stein2018global" class="csl-entry">
Stein, R William, Christopher G Mull, Tyler S Kuhn, Neil C Aschliman,
Lindsay NK Davidson, Jeffrey B Joy, Gordon J Smith, Nicholas K Dulvy,
and Arne O Mooers. 2018. <span>“Global Priorities for Conserving the
Evolutionary History of Sharks, Rays and Chimaeras.”</span> <em>Nature
Ecology &amp; Evolution</em> 2 (2): 288–98.
</div>
<div id="ref-sternes2020body" class="csl-entry">
Sternes, Phillip C, and Kenshu Shimada. 2020. <span>“Body Forms in
Sharks (Chondrichthyes: Elasmobranchii) and Their Functional,
Ecological, and Evolutionary Implications.”</span> <em>Zoology</em> 140:
125799.
</div>
<div id="ref-thomson1976heterocercal" class="csl-entry">
Thomson, Keith Stewart. 1976. <span>“On the Heterocercal Tail in
Sharks.”</span> <em>Paleobiology</em> 2 (1): 19–38.
</div>
<div id="ref-thomson1977body" class="csl-entry">
Thomson, Keith Stewart, and Dan E Simanek. 1977. <span>“Body Form and
Locomotion in Sharks.”</span> <em>American Zoologist</em> 17 (2):
343–54.
</div>
<div id="ref-wilga2004biomechanics" class="csl-entry">
Wilga, Cheryl AD, and George V Lauder. 2004. <span>“Biomechanics of
Locomotion in Sharks, Rays, and Chimeras.”</span> <em>Biology of Sharks
and Their Relatives</em> 5: 139–64.
</div>
</div>
</div>



</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->

<script>
$(document).ready(function ()  {

    // temporarily add toc-ignore selector to headers for the consistency with Pandoc
    $('.unlisted.unnumbered').addClass('toc-ignore')

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
