---
title: "Module 5 Project"
bibliography: class.bib
---
<style>
div.green { background-color:#c7d1c0; border-radius: 5px; padding: 20px;}
</style>
<!-- Add icon library -->
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.14.0/css/all.min.css">



```{r images, include=FALSE,eval=FALSE}
library(tidyverse)
library(ape)
library(treeio)
library(rgbif)
library(data.table)
library(phytools)
library(fishtree)
library(rgbif)
library(doParallel)
library(foreach)
library(parallel)
library(dataRetrieval)
library(geomorph)


### sort image files

f <- list.files("ignore/shark_photos",full.names = T,include.dirs = F,recursive = T)

gifs <- f[grepl(".gif$",f)]

img <- tibble(
  file=f,
  sp=basename(dirname(f)),
  is_gif=grepl(".gif$",f)
) 

sapply(img %>% filter(is_gif) %>% pull(file),function(x) file.copy(x,paste0("ignore/class_shark_images/",basename(x))))

#need lateral images of squatina sp.

img %>% 
  select(sp,is_gif) %>% 
  count() 
```


```{r data, include=F,message=F,eval=F}
shark_dat_ <- read_csv("data/shark_hab_added.csv") 
fb_res <- readRDS("data/shark_fb_res.RDS") %>% 
   add_row(sp=c("Squatina squatina","Squatina australis"),
          file_name=c("Sqsq1.jpg","Sqaus1.jpg"))

##sources
###https://bilderreich.de/1634/fact-sheet-angelshark-squatina-squatina.html
###https://fishesofaustralia.net.au/home/species/3283
 
f_student <- gsub("_.csv","",list.files("shark_points"))

shark_dat <- fb_res %>% 
  left_join(shark_dat_) %>% 
  #filter(file_name %in% f_student) %>% 
  mutate(habitat=ifelse(habitat=="demersal","benthic",habitat))

write_csv(shark_dat,"data/shark_data.csv")
write_csv(shark_dat,"shark_data.csv")

dir.create("ignore/class_shark_images/for_upload")

f_img <- list.files("ignore/class_shark_images",full.names = T)

# groups <- rep(paste0("group_",1:9),each=23)
# groups <- sample(groups,length(f_img))
# img_dat <- tibble(file_path=f_img,file_name=basename(f_img)) %>% 
#   left_join(fb_res) %>% 
#   left_join(shark_dat_) %>% 
#   mutate(group=groups)
# 
# sapply(groups %>% unique,function(x) dir.create(paste0("ignore/class_shark_images/for_upload/",x)))
#   
# for(i in 1:nrow(img_dat)){
# file.copy(img_dat[i,]$file_path,paste0("ignore/class_shark_images/for_upload/",img_dat[i,]$group,"/",img_dat[i,]$file_name))
# }



```



# Introduction

Sharks are an ancient group of cartilaginous fishes in the class Chondrichthyes, subclass Elasmobranchii, and subdivision Selachimorpha. The selachians include over 500 species that are further placed in in two superorders: Squalomorphii and Galeomorphii. Modern sharks originated nearly 200 million years ago (@grogan2012origin) and, since then, have diversified into nearly every major marine ecosystem. This evolutionary success has been attributed to the diversity of their locomotor modes and associated body shape (@lauder2015swimming,@thomson1977body). A modest body of work devoted to the [ecomorphology](https://en.wikipedia.org/wiki/Ecomorphology) of sharks has focused largely on caudal find shape as it relates to ecology and locomotor modes (@thomson1976heterocercal,@wilga2004biomechanics). Sharks have a heterocercal tails in which the caudal part of notochord flexes dorsally and supports a large dorsal lobe of the fin, while, with more than a few exceptions, the ventral lobe is small.

<br>
<br>

<center>
<img src="https://ars.els-cdn.com/content/image/1-s2.0-S0944200620300581-gr1.jpg" alt="bull shark" width="400"/>



<br>
<br>

(<b>A</b>) The phylogenetic relationships and systematic placement of major groups of shark and (<b>B</b>) @thomson1977body's ecomorphological groups. Figure from @sternes2020body.
</center>
<br>
<br>

In there seminal work on shark ecomorphology, @thomson1977body put forth that there were four shark morphotypes:

  1. A group characterized by having a deep body, large pectoral fins, a symmetrical caudal fin with a  narrow  peduncle (ie., base to the  fin) and high-aspect ratio. A white shark represents this group.
  2. A group characterized by a body less deep then Group 1 and a cadual fin that is swept (ie., large heterocercal angle--the angle between the upper and lower lobes of the fin). This group includes most carcharhinid sharks such as the bull shark.
  3.  A group with a large head, blunt snout, a body that has anteriorly placed pectoral fins and posteriorly placed dorsal and pelvic, and a caudal fin with a low heterocercal angle and small ventral lobe. Catsharks (family Scyliorhinidae) and dogfish (order Squaliformes) represent this group.
  4. A group characterized by a caudal fin with a higher aspect angle similar to that of Group 2 but lacking an anal fin, e.g. the dalatiids.
  
<center>
<table>
    <tr>
        <td>
        <center>
            <img src="https://earthlife.net/wp-content/uploads/What-Is-A-Elasmobranch-1280x718-1.webp" alt="bull shark" height="200"/>                   
            <br>
A white shark, <i>Carcharias carchardon</i>, typical of Group 1.
<br>
<br>
</center>
        </td>
         <td>
         <img style="height: 20px; width: 20px"/>
          </td>
         <td>
         <center>
<img src="https://biogeodb.stri.si.edu/sftep/resources/img/images/species/95_8407.jpg" alt="bull shark" height="200"/>                   
<br>
A bull shark, <i>Carcharhinus leucas</i>, typical of Group 2.
<br>
<br>
</center>
        </td>
    </tr>
    
<br>
<br>
    <tr>
        <td>
        <center>
            <img src="https://www.vichighmarine.ca/wp-content/uploads/2017/03/Spiny-Dogfishy-600x321.jpg" alt="bull shark" height="200"/>                   
            <br>
A dogfish of the genus <i>Squalus</i> typical of Group 3.
<br>
</center>
        </td>
         <td>
         <img style="height: 20px; width: 20px"/>
         </td>
         <td>
         <center>
<img src="https://biogeodb.stri.si.edu/caribbean/resources/img/images/species/4437_2257.jpg" alt="bull shark" height="200"/>                   
<br>
A kitefin shark, <i>Dalatias licha</i>, typical of Group 4.
<br>
</center>
        </td>
    </tr>
    
</table>
</center>
<br>
<br>


@sternes2020body used [geometric morphometrics](https://en.wikipedia.org/wiki/Morphometrics#Landmark-based_geometric_morphometrics) and a [multivariate  analysis](https://en.wikipedia.org/wiki/Principal_component_analysis) to evaluate the groups proposed by @thomson1977body and found that sharks fell into one of two morphotypes (A and B). Their Group A was characterized by an elongate body, posteriorly placed dorsal fin, and a swept caudal while Group B was characterized by a deeper body, anteriorly place dorsal fins, and a less swept caudal fin. @thomson1977body suggested that the shape of species in these two broad ecomorphs evolved due to differences in ecological niche and [locomotor behavior](https://en.wikipedia.org/wiki/Fish_locomotion). Specifically, they posited that the species of elongate Group A are typically [benthic](https://en.wikipedia.org/wiki/Demersal_fish) and swim like eels whereas the deeper bodied species of Group B are often [pelagic](https://en.wikipedia.org/wiki/Pelagic_fish) and swim like salmon or tunas.



<center>
<img src="https://ars.els-cdn.com/content/image/1-s2.0-S0944200620300581-ga1_lrg.jpg" alt="bull shark" width="400"/>



<br>
<br>

From @sternes2020body who found sharks could be assigned two morphotype groups: A, elongate body, posteriorly placed dorsal fin, and a swept caudal fin; B, deeper body, anteriorly place dorsal fins, and a less swept caudal fin.
</center>
<br>
<br>

Although the work of @thomson1977body and later @sternes2020body points to interesting patterns in shark body-shape, neither study explicitly studied the role habitat---that is, pelagic or benthic---had in influencing the evolution body-shape evolution. In particular, @sternes2020body merely observed that their two groups with different body shapes often varied in their habitat preference. In addition, @sternes2020body made no explicit, quantitative phylogenetic test of their patterns. This leaves open an important question: **Do pelagic and benthic sharks have significantly different body shape?** Subsequently, we might also ask, **Has the morphology of pelagic and benthic sharks evolved at the same rate?** These are the questions we will take on in this project.

These questions represent two an important thrust of evolutionary research: what macroevolutionary events have contributed to diversity, whether in species or phenotype. To answer questions like these requires that we essentially recapitulate the @sternes2020body body-shape data set using landmark-based [geometric morphometrics](https://en.wikipedia.org/wiki/Morphometrics#Landmark-based_geometric_morphometrics). This is a relatively straightforward, if tedious, method of capturing shape variation in a group of interest. For this, you'll digitize 14 homologous landmarks on images of some 220 species of sharks. Using the best practices in the field of geometric morphometrics to align our landlandmarks and extract shape information, we'll assess morphological disparity in a phylogenetic framework and assess the difference in rates of morphological evolution between the groups between pelagic and benthic sharks. 

# Methods

## Image acquisition

A few decades ago, a morphometric project like this would require visiting museums collections from around the world to take images of the specimens and species of interest. In fact, this was a major part of Prof. Kenaley's PhD thesis, which included trips to natural history collections in France, Denmark, Great Britain, Taiwan, Japan, Monaco, New Zealand, and Australia. Ahhhhhh, the good ol' days. Fortunately (or unfortunately), travel to such wonderful places may be avoided several resources exist that contain a trove of images of fish-like vertebrates. 

We'll make use of shark images in the [FishBase](https://www.fishbase.us/) catalog: a clearing house of all things fish that has amassed tons of data from the scientific literature. Prof. Kenaley used the `rfishbase` package to rip images of scientific drawings of sharks. This resulted in a cache of images for the ~220 species included in a recent phylogeny of the elasmobranchs (@stein2018global).

## Species list, phylogeny, and ecology

The 200+ species we'll analyze in the study were chosen because they had decent images on FishBase and cover the nooks and crannies of the @stein2018global phylogeny, giving a broad sample of the group. As we move on, we'll use [this table](data/shark_data.csv) to keep track of images we need to landmark, the species the images belongs to, and the habitat for each.The habitat data for our study come from @compagno1984sharks and @compagno2001sharks. Let's load these data now so we have it handy.




```{r sharkdat}
library(tidyverse)
shark_dat <- read_csv("shark_data.csv")
```


Because we'll be comparing body shape in sharks, this project is by definition a comparative one and we'll need a phylogeny for the group. Fortunately a relatviely recent phylogenomic analysis of over 500 species of elasmobranchs (sharks + batoids) has been published by @stein2018global. Unfortunately, there is some phylogenetic discordance in this study. Specifically, when researchers analyze phylogenetic relationships, they typically repeat their analysis in a process known as [bootstrapping](https://en.wikipedia.org/wiki/Computational_phylogenetics#Evaluating_tree_support). In phylogenetics, bootstrapping is conducted using the columns of the character matrix. Each pseudoreplicate contains the same number of species (rows) and characters (columns) randomly sampled from the original matrix, with replacement. A phylogeny is reconstructed from each pseudoreplicate, with the same methods used to reconstruct the phylogeny from the original data. This often results in a data set of trees that have conflicting relationships---that is, phylogenetic uncertainty---and this was the case for the @stein2018global study. We'll have to account for this in our analysis. But, more on that later . . .

In any case, we've used the taxa studied in this paper to find images in FishBase and we'll use [a tree file](data/shark_trees.nex) representing phylogeny for our comparative analysis. 

<center>
<br>
![](https://media.springernature.com/lw685/springer-static/image/art%3A10.1038%2Fs41559-017-0448-4/MediaObjects/41559_2017_448_Fig1_HTML.jpg?as=webp){width=50%}
<br>

The phylogenetic relationships of the elasmobranchs according the @stein2018global.
</center>
<br>

## Geometric morphometrics

As we learned in class and what you've been reading, researchers interested in comparing shape change between species often use landmark-based geometric morphometrics. A geometric morphometric study such as ours generally follow these steps:

1. Data collection: select landmarks of interest, usually through digitization of specimen images. 
2. Data standardization: Make landmarks comparable across all specimens, usually through superimposition.
3. Analysis: choose a statistical approach appropriate to the questions.
4. Interpretation: use the outcome of the statistical analysis to assess the original questions.



### Digitization and landmarking

Our shape analysis will be based on 14 landmarks placed on images of shark bodies of our ~220 species. To landmarks on the body, we'll follow this generalize workflow below in the image analysis program FIJI (FIJI is just imageJ). 

<br>
<center>
![](shark_points.png){width=70%}
<br>
</center>
<br>

You and your team/group should will be assigned about 20 of the ~200 species to outline. Images of these species can be accessed through [this directory](https://drive.google.com/drive/folders/1X8oHbgvaK_5-gzFtc2p2dS3nWXjFZeUc?usp=sharing). 

First gather your images to work on and follow the steps below for each. In FIJI:

1. Open am image.

2. Select the multipoint tool.

3. Digitize the 14 landmarks you see above.

4. Measure the XY positions of these points (`cmd/ctrl+m`). 

5. Select the macro tab and run the macro (`cmd/ctrl+r`).

6. Save the results to the appropriate directory.

7. Close the results and image tab (don't save the image).

This process is demonstrated here.
<br>
<br>
<br>
<center>
<iframe width="460" height="315" src="https://www.youtube.com/embed/-yZu3FkJLzU" frameborder="0" allowfullscreen></iframe>
</center>
<br>



**Note:** For the first image you outline, you will have to open the macro editor ("Plugins" $\rightarrow$ "New" $\Macro$). Paste the script command from below into the editor and save this new macro to an appropriate directory, perhaps your cloned repo. You'll want to access this later ("Plugins" $\rightarrow$ "Edit") so that you can automate the process of saving text files with the correct fielname.



```{}
f = getTitle();
dir=getDirectory("Choose where to save data ");
selectWindow("Results");
saveAs("Results", dir+f+"_.csv");
```


This script saves the results as a *tab-separated* text file with same filname as the image file name and appends "_.csv".  **This is super important!!** and how we'll link the landmarks to a species.

After your team has completed landmarking the assigned species, upload your text files to [this directory](https://drive.google.com/drive/folders/1j7sod1NB1jmEiPiduYvOOIlukkMikfL5?usp=sharing).  Once all the data appear here, Prof. Kenaley will evaluate the quality of the landmarks and do some post-processing if required (i.e., file-name checks, quality, etc.). You can then download [these evaluated data](data/class_out_data_f23.zip) for shape analysis.

### Superimposition

After digitizing landmarks, we must remove the distracting variations of size, orientation, and position that can vary between specimens and thus add noisey variables in analysis of shape. This can be mitigated by using a superimposition method. Generalized procrustes analysis (GPA) is the superimposition method of choice nowadays. GPA removes the variation of size, orientation, and position by superimposing the landmarks in a common coordinate system. The landmarks for all specimens are optimally translated, rotated, and scaled based on a least-squared estimation. The first step is translation and rotation to minimize the squared and summed differences (squared Procrustes distance) between landmarks on each specimen. Then the landmarks are individually scaled to the same units of centroid size. Centroid size is the square root of the sum of squared distances of the landmarks in configuration to their mean location. The translation, rotation, and scaling bring the landmark configurations for all specimens into a common coordinate system so that the only differing variables are based on shape alone. The new superimposed landmarks can now be analyzed in multivariate statistical analyses


![](gpa.png)

To perform a Procrustes superimposition on our landmark data---along with many other analyses---we will turn the the `geomorph` package. Please install and load `geomorph`. But, before we do, let's download and read in about 90 or so landmark files that Prof. Kenaley produced ahead of this project. You can find these example files [here](data/sharks/shark_points.zip). You'll have to download this compressed directory and unzip it to your project directory.  You'll also have to download and install the `abind` package as well. 

Note: **these are just examples**. You will need to download the full data set evaluated by Prof. K. form this location.

To get started with superimposition, let's load the packages `geomorph` and `abind`. When you unzip the `shark_points.zip` directory to you project directory, you have a folder named `shark_points`. Our first operation will be to list the files in that directory and then read them in, all at once.

```{r exdata}
library(geomorph)
library(abind)

f <- list.files("shark_points",full.names = T,pattern=".csv")

xy <- read_csv(f,id="file") %>% 
  select(file,X,Y) %>% 
  mutate(Y=abs(Y-max(Y))) %>% 
  mutate(file_name=gsub("*_.csv","",basename(file))) %>% 
  left_join(shark_dat) 
```

Here we used `list.files()` with `pattern=pattern=".csv"` to find all the files in the `shark_points` directory. We also set `full.names = T` so that the full file path is saved to the object `f`. 

```{r fewf}
head(f)
```


This made reading all the files and adding them to one tibble in a single operation possible with `read_csv()`. Notice we added a `file` column to the tibble with `id="file"` and then selected only the columns `file`, `X`, `Y`. This operation is saved to an object logically named `xy`. 

For landmarks to used in `geomorph`, they must be stored as a 3-dimensional array, that is, essentially a list of similarly sized 2D matrices, where the position the $i^{th}$ matrix is in the $i^{th}$ index of the third dimension, i.e., `array[,,i]`.  To massage our large tibble into such an array, we have to split the tibble into a bunch of lists for this, we can simply run a `for` loop, breaking the data up into a list of data frames according to species. Notice how we use `filter()` to split the large tibble into into small data.frames that will reside in `ldk_l`. This will leave us with a list of data.frames as long as all our files. We then use `abind`'s `abind()` function to bind the list of data.frames in `ldk_l` into an array, with each data.frame taking a position in the third dimension. This array is saved in the object `ldk`. Lastly, we'll specify that the names of this third dimension should be the unique values of `species` in the original tibble. **Note: this is how we'll retrieve species information.**




```{r ldk}
ldk_l <- list()

shark_sp <- xy$sp %>% unique

for(i in shark_sp){
  ldk_l[[i]] <- xy %>% 
    filter(sp==i) %>% 
    select(X,Y)
    data.frame
}


ldk <-  abind(ldk_l, along=3)

dimnames(ldk)[[3]] <-names(ldk_l)

ldk[,,1:3] #the first few 
```


With a proper array in place as `ldk`, we're now free to move on the superimposition with the `geomorph` function `gpagen()`. We'll do this saving the output to `ldk_al` (for landmark alignment) and plot the the 14 aligned and superimposed landmarks for all the example specimens.

```{r, gpa}
ldk_al <- gpagen(ldk)
plot(ldk_al)
```


### Analysis Part I: A Morphospace

In geometric morphometric analysis, researchers first turn to describing the axes of shape variation.Indeed, this is what @sternes2020body did to produce the figure above. This and what we're after is often termed a [morphospace](https://en.wiktionary.org/wiki/morphospace), a representation of the shape the study group does and does not have. 

To develop such a thing, biologist usually turn to Principal Components Analysis (PCA) whereby a very complicated suite of variables is distilled into a set of fewer variables that describe the important variance among the original variables. The coordinates contained in our `ldk_al` object describe dozens of variables: scaled distances between all the landmarks. So we have many variable to consider and this becomes, very quickly, a serious [multivariate problem](https://en.wikipedia.org/wiki/Multivariate_statistics). PCA is suited to multivariate analysis because it reduces the complexity of the data, taking many variables and distilling them into just a few. At its core, PCA produces a series of regressions, or vectors, that pass through the values of the original variables. This operation is performed iteratively, with the first vector (the first principal component, PC1), drawn through the variable space that accounts for the most variance in the data set. After the combination of variables forming PC1 is removed, a subsequent vector (PC2) is drawn through another unique variable space that describes the second most amount of variance. This continues until most of the variance is captured, resulting in scores of PCs. The first two or three, however, are usually enough to describe a health amount of variance, say 90% or more.

To perform PCA, we'll use `geomorph`s `gm.prcomp()` function like so. Plotting reveals our first result.


```{r pca,message=F,eval=T}
pca <- gm.prcomp(ldk_al$coords)
plot(pca)
```

Wow, now we're getting close to the @sternes2020body results and answering, at least in part, one of our questions: **Does habitat result in different body shapes**. But let's break down what this plot means first. By default, when we plot a PCA object, the first two PCs are plotted, PC1 on the x axis, PC2 on the y. Each point represents the PC scores for a specimen in the data set. Notice that the amount of variance explained by our PCs is given and, as expected, PC1 contains most of the variance (54%) and PC2 considerably less less (19%). If we plot again, but specify a different combination of PCs, say 1 and 3, we get a different plot, of course. Notice the variance explained by PC3 is predictably less than that of PC2.

```{r, pc2}
plot(pca,1,3)
```

OK, so based on our PC plots, it looks as if we have two groupings, just like @sternes2020body. But, do these groups correspond to largely pelagic vs. benthic species? It's hard to tell. We don't really know because we don't know which species belong to each point in our plots, nor their habitat.

Notice, however, that the names of the coordinates in `ldk` were pegged in a previous operation to the species names. 

```{r names}
head(dimnames(ldk)[[3]])
```

These values were maintained throughout the superimpositon step and in the PCA operation. The coordinates plotted in the PCA plots are contained in a matrix named `x` in our `pca` object. Notice the row names:

```{r rownmaes}
pca$x %>% data.frame %>% select(Comp1,Comp2) %>% head
```

So, we have the species names from whence the data, both land mark and PCA, came. Fortunately, Professor Kenaley assembled a table containing of all the the species  for landmarking, and the species and habitat for each. You can download [this file](data/shark_data.csv) and read in the data.


What's left to do in exploring if benthic and pelagic sharks occupy different morphospace is to combine this information with the PC scores in `pca$x`. This can be done thusly:

```{r combdat}
PCA <- pca$x %>% 
  data.frame %>% 
  select(Comp1:Comp4) %>% 
  mutate(sp=rownames(pca$x)) %>% 
  left_join(shark_dat)

head(PCA)
```

With that over with, we can dispence with the ugly PCA plots above and use `ggplot` to visualize our results with repsect to habitat.

```{r ggpca}
PCA %>% 
  ggplot(aes(Comp1,Comp2,col=habitat))+geom_point()

PCA %>% 
  ggplot(aes(Comp1,Comp3,col=habitat))+geom_point()
```

Voila! For morphospace described by PCs 1, 2, and 3, it appeears that pelagic species inhabit a high PC1 score and never a low PC1 score. 

This is wonderful, but you may be asking, "What is PC1 (or any other) describing?" Our PCs points are abstractions of shape, but they don't represent explicitly what shape that is. To get a sense of what our PCs represent in terms of shape values, we can turn to thin-plate spline analysis.

**Note**, you're not asked to do anything like this for the report. Prof. K just wants you show you have you can derive shape changes represented by the PC axes.

```{r rw}

#find min and max of each PC
max_PC1 <- which.max(pca$x[,1])
min_PC1 <- which.min(pca$x[,1])
max_PC2 <- which.max(pca$x[,2])
min_PC2 <- which.min(pca$x[,2])

#get the concesus, the mean shape of all data.
M <- ldk_al$consensus

#plot
par(mfrow=c(2,2))

#plot the change form the concensus to min of PC1
plotRefToTarget(M, ldk_al$coords[,,min_PC1], mag = 1,method="vector",label = T)
title("min PC1 warp",cex=0.6)

#plot the change form the concensus to max of PC1
plotRefToTarget(M, ldk_al$coords[,,max_PC1], mag = 1,method="vector",label = T)
title("max PC1 warp",cex=0.6)

#plot the change form the concensus to min of PC2
plotRefToTarget(M, ldk_al$coords[,,min_PC2], mag = 1,method="vector",label = T)
title("min PC2 warp",cex=0.6)

#plot the change form the concensus to max of PC2
plotRefToTarget(M, ldk_al$coords[,,max_PC2], mag = 1,method="vector",label = T)
title("max PC3 warp",cex=0.6)

```

So, it looks like 

### Analysis: Part II

Our PCA analysis reveals an interesting pattern. This, however, is not quantitative assessment *per se*, merely a visual and qualitative result and not dissimilar to the qualitative results reported by @sternes2020body. Several quantitative frameworks for assessing differences in shape exist. The `geomorph` package contains several, but we'll use just one: morphological disparity. The function `morphol.disparity()` will do this for us by making pairwise comparisons of residuals of a linear fit to identify differences among groups. It's essentially an ANOVA for shape. The statistical significance of the calculated Procrustes variances between the different growth stages is assessed using a randomized residual permutation, that is random set residuals from all groups are chosen in each iteration.

`morphol.disparity()` requires a special data structure, a `geomorph` `data.frame` and this can be constructed using the `geomorph.data.frame()` function. It takes the landmark data and other variables of interest (i.e., in our case, habitat). First, we'll establish the special data frame using our aligned landmarks and habitat data from the `PCA` tibble.

```{r morphdis,eval=T}

gdf <- geomorph.data.frame(ldk_al,
                             habitat=PCA$habitat
)
```

To you use the `morphol.disparity()` function, we then need to establish a formula describing the  model to use and tell the function which groups the model should include. This is encoded with `coords ~ 1` which indicates we'll look at the differences in means and  `groups= ~ habitat`, indicating that we'll evaluate means between the habitat groups. `data = gdf` indicates we'll use the `gdf` data frame we established above, while `iter = 10000` indicates we'll be perform 10,000 radom permutations. Lastly, we just summarize the `md` object to return the results

```{r morphdis2}
md <- morphol.disparity(coords ~ 1, groups= ~ habitat, data = gdf, iter = 10000, print.progress = FALSE)

summary(md)

```

Notice the the pairwise differences are returned both for the Procrustes shape variance and the absolute differences along with the p values. As it turns out, at least for this limited data set of ~90 species, there's no significant morphological disparity between groups.

We have a preliminary answer to our first question, "Does habitat use predict different body shape?". But this of course could change with a larger sample of species. 

```{r phy,message=FALSE,eval=F,include=F}
library(phytools)
phy <- readNexus("data/shark_trees.nex")

phy2 <- keep.tip(phy,PCA$sp)

  gdf <- geomorph.data.frame(ldk_al,
                             species = PCA$sp,
                             habitat=PCA$habitat,
                             phy=phy2[[1]]
  )
  

library(parallel)  
shark_pgls <- procD.pgls(coords ~ habitat, phy = phy, data = gdf, 
iter = 10000,print.progress = F)


summary(shark_pgls)
```

### Analysis Part III: Evolution in Parallel

Our secondary question was, if you remember, was, "**Has the body shape of pelagic and benthic sharks evolved at the same rate?**". This is a much different question than the first. We want to know, overall, if the shape of lineages that invaded a particular habitat has evolved faster or slower than those invading another habitat. For this we'll turn to the `geomorph` package once again, in particular, its `compare.evol.rates()` function. This function calculates net rates of shape evolution for groups of species on a phylogeny from a set of aligned landmarks. The approach is based on between-species differences in morphospace after phylogenetic transformation. In other words, after correcting for phylogenetic relatedness, it compares relative rates of shape change.

Because we're correcting for phylogeny in this analysis, we need the [tree from @stein2018global mentioned above](data/shark_trees.nex). Once this tree file is downloaded, let's load, read it, in and store it to an object named `phy`. Notice you'll need the `phytools` package and the function `readNexus()`.


```{r readphy}
library(phytools)
phy <- readNexus("data/shark_trees.nex")
length(phy)
```

Next let's, restrict our tree file to those species in our data set. For this, we'll used `ape`'s `keep.tip()` which keeps only the tips in a vector, in this case, the species in our PCA and landmark analysis.

```{r phydrop}
phy2 <- keep.tip(phy,PCA$sp)
```


To undertake the analysis with `compare.evol.rates()`, we must specify `A`, the aligned coordinates in a lankmark array (`ldk_al$coord`), a tree (more in just a bit), and a named vector of groups descriping to which group each species in the array belongs.

Let's establish the grouping vector by passing the `habitat` values in our `PCA` tibble to an object named `gp_hab` and then give the names of this vector the `species` values. Then we run the function.


```{r evolrate}
gp_hab <- PCA$habitat
names(gp_hab) <- PCA$sp

er <- compare.evol.rates(A=ldk_al$coords, phy=phy2[[1]], method="simulation",gp=gp_hab,print.progress = F)

print(er)
```

So it appears the benthic group body shape has evolved 1.6 times faster than the pelagic group and this is significant (*p=0.002*). But, we have a problem. You may have noticed that when we loaded our tree file, the length of this list of trees is 500. These are 500 trees produced by @stein2018global using the bootstrapping method mentioned earlier. Many of these trees are incongruent, that is, they represent different relationships for the sharks and this, in turn, represents some serious phylogenetic uncertainty. 

Also notice that in our rate analysis above, we used only one tree in this list (`phy=phy2[[1]]`). There are 499 other trees we could have used, perhaps each representing a different phylogenetic hypothesis. The results of a comparative analysis like this could, and likely will, vary according to the phylogenetic hypothesis used in it. So what to do? 

This is a common problem faced by comparative biologists and, fortunately, there seems to be a consensus among us to repeat our comparative analysis using a random sample or, ambitiouslu, all of the trees in that represent the phylogenetic uncertainty. By repeating an analysis, we produce a series of results based on new parameters (i.e., phylogenetic distance) to our model. In this case, the new parameters would include by using any or all of the trees. This establishes what is akin to a [posterior sample](https://en.wikipedia.org/wiki/Posterior_probability) and from this we can extract a posterior distribution of estimates, in our case, a distribution of evolutionary rates.

Therefore, we have before us the proposition of repeating out analysis, let's say 10 times. To do this, we could randomly sample 10 of the and build a posterior distribution of evolutionary rates using a `for` loop. 

```{r loop,echo=T,message=F}

set.seed(1234)
phy_sample <- sample(phy2,10)

er_l <- list()


t <- system.time(for(i in 1:length(phy_sample)){
 er_l[[i]] <-compare.evol.rates(A=ldk_al$coords, phy=phy_sample[[i]], method="simulation",gp=gp_hab,print.progress = F) %>% summary
}
)


```

Here we used `seet.seed()` to make the `sample()` function randomly sample the same set of 10 trees. This is important so that we can all perform the same exact analyses, that is, our analyses are repeatable. You'll notice that the loop uses each random tree to repeat the analysis and store the summary of the model (i.e., the rate comparisons) to the list `er_l`. Notice also that the entire loop is wrapped in `system.time()`. This is done so that, in this preliminary analysis, we can record how much time the loop took. Printing `t` reveals that it took 7 seconds. 
Because our results are stored in a list, we can use `lapply()` to extract the rate comparisons. `lapply()` takes a list and applies a function to. In our case, we need to encoded a custom function that extracts `sigma.d.gp` and `sigma.d.ratio`, the rate for each group and rate ratio between them, respectively. For this, we ask the function to work on every value of `er_l`, read it as `x`, then combine `sigma.d.gp` and `sigma.d.ratio` and store it in `er_` temporarily. We then apply a name to the third value "ratio", and return the vector. `lapply()` returns a list, so we combine all elements of it using `do.call()`. We apply the row-binding function `rbind()` to the list using the pipe and `.` and lastly, make the output a data frame.


```{r erlist}


er <- lapply(er_l,function(x) {
  er_ <- c(x$sigma.d.gp,x$sigma.d.ratio)
  names(er_)[3] <-"ratio"
  return(er_)
}) %>% do.call(rbind,.) %>% data.frame

head(er)
```

With a little massaging with `pivot_longer()`, we can plot the posterior distribution of rates for each habitat like so.

```{r ploter}
er %>% 
  pivot_longer(benthic:pelagic) %>% 
  ggplot(aes(name,value)) + geom_boxplot()

```

Indeed it appears that when we estimate evolutionary rates for 10 random trees, accounting for uncertainty in species relationships, pelagic species evolve slower than benthic ones.

Now, this is all well and good. But, most comparative biologists would argue 10 random trees does not a posterior make. We should probably sample many more or, perhaps use the entire set of trees. 

What's to stop us? Well, if we shoot for the stars and use all 500 trees in 500 separate analyses, we'd be looking at running a for loop 50 times longer than 10 trees took (roughly 350 s or 6 minutes, see figure below). In the world of comparative methods, this isn't such a long time, but remember, this is a preliminary analysis of only 90 or so of the 220 we hope to include. The figure below indicates that including all 220 species and using all 500 trees would take ~2500 s or about 40 minutes!!!

<br>

```{r timescale,echo=F,include=F}
library(wesanderson)
library(ggsci)

if(F){
args <- expand_grid(sp=seq(10,90,10),ntrees=c(10,50,200,500))

res <- list()

for(i in 1:nrow(args)){

  sp_i <- sample(PCA$sp,args[i,]$sp)
  hab_i <- PCA %>% filter(sp%in% sp_i) %>% pull(habitat)
  names(hab_i) <- PCA %>% filter(sp%in% sp_i) %>% pull(sp)
  phy_i <- sample(keep.tip(phy2,sp_i),args[i,]$ntrees)
  
  ldk_i <- xy %>% 
  filter(sp %in% sp_i) %>%  
  select(sp,X,Y) %>% 
  group_split(sp,.keep = F) %>% 
  lapply(.,as.data.frame) %>% 
  abind(., along=3)

dimnames(ldk_i)[[3]] <-sp_i

gpa_i <- gpagen(ldk_i,print.progress = F)

  res[[i]] <-  system.time(
    for(p in 1:length(phy_i))
      {compare.evol.rates(A=gpa_i$coords, phy=phy_i[[p]], method="simulation",gp=hab_i,print.progress = F) 
        
    }
    
  )
       cat("\r",paste0(i, " of ", nrow(args)))                    
}
 



args$time <- lapply(res,function(x) x[3]) %>% unlist

saveRDS(args,"er_time.RDS")

}


args <- readRDS("er_time.RDS")
args2 <- args %>% mutate(pred="no") %>% 
  rbind(
 args %>%
  group_by(ntrees) %>% 
  dplyr::reframe(time=predict(lm(time~poly(sp,3)),newdata=data.frame(sp=c(100,150,220))),sp=c(100,150,220)) %>% 
  mutate(pred="yes") %>% 
  select(sp,ntrees,time,pred)
)%>% 
  mutate(al=ifelse(pred=='yes',1,0.7)) 

args2 %>% 
  ggplot(aes(sp,time,col=as.factor(ntrees),group=ntrees,shape=pred,alpha=al))+geom_point()+geom_line()+theme_classic(15)+guides(col=guide_legend(title="# of trees"))+scale_alpha(guide="none")+scale_shape(guide="none")+theme(legend.position = c(0.2,0.8))+xlab("# of species") +ylab("time (s)")+geom_text(aes(50,500,label="computed"),inherit.aes = F)+
  geom_text(aes(175,2200,label="predicted"),inherit.aes = F)+
    scale_color_aaas()
```
<br>
<center>
The polynomial relationship between time and number of taxa in comparative analysis, in this case computing evolutionary rates of shape change with `geomorph`'s `compare.evol.rates()`. 
</center
<br>
<br>


We should, therefore, be prepared for the long haul. But, hold on. I don't know about you, but on most days, I don't have 40 minutes to spare, even for very interesting comparative analyses. Fortunately, to avoid 100s of iterations of the same operations on different data, we can make use of a very common work-around in scientific computing: [parallelization](https://en.wikipedia.org/wiki/Parallel_computing), the simultaneous execution of different pieces of a larger computation across multiple computing processors or cores. If fact, this sort of problem is considered [embarrassingly parallel](https://en.wikipedia.org/wiki/Embarrassingly_parallel), that is, it is so very easy to split this task into many parallel subtasks. Specifically, we have 500 identical tasks, we need only to break these up into subtasks that run in parallel. That's what we'll do.

`R` has several packages for parallel computing. Perhaps the most used is `parallel`, so cleverly named. Let's work on an example of how a task could be parallelized using `parallel`'s `mclappy()` function.

`mclapply` is simply an extension of `lapply()` that takes the tasks performed on each item in a list and spreads the tasks across multiple cores on a computer. This is perfect for embarrassingly parallel problems.

For example, say we wanted to do something that's simple, but takes just a little bit of time, like generation 10 million random values from a normal distribution (`mean=0`, `sd=1`) any number of times. Let's encode a function that does this and allows one to specify how many times to do it. 

```{r parafun}

my_fun <- function(x){for(i in x) rnorm(1e7)}

```

For comparison, let's first use the non-parallel `lapply()` to run the function 30 times and then use `mclapply()` to do the same. Notice that we've again used `system.time()` to capture the duration of the operations. Also notice that we used `parallel`'s `detectCores()` to detect the number of cores on the machine and we subtracted 2 from this value. Must modern laptops and desktops have $>$6 cores, but we don't want all of them to be used by the operation. We should leave a few cores that the computer can use for operations running in the background.

```{r para,cache=T}
t_nomc <- system.time(lapply(1:30,my_fun))

t_mc <- system.time(mclapply(1:30,my_fun,mc.cores = detectCores()-2)) #8 cores

print(t_nomc)
print(t_mc)
```


As you can see, the parallelized operation was about 6 times faster than `lapply()`. This framework looks like it could really speed up our estimation body shape evolutionary rates. You are tasked with assessing evolutionary rates based on habitat using all species in the data set and all 500 of the phylogenetic trees from @stein2018global using a paralleled approach.


# Project Report

Please submit your report and all associated data to your team GitHub repository as an .Rmd document with HTML output that addresses the following:


  * Establish an array suitable for analysis using the `geomorph` functions outlined above. Construct this from all specimens digitized in [this directory](data/class_landmark_data.zip) and Procrustes align them (**note:** this will be available soon after all landmark data is collected).
  * Does habitat result in the occupation of a different morphospace? Please answer this question using a `ggplot`-rendered morphospace based on the first two PCs from a principal components analysis of your alignment.
  * Is there morphological disparity between habitats?
  * Is the rate of morphological evolution significantly different between different habitat types?
    1. Answer this question based on a posterior distribution of rates based on all 500 phylogenetic trees.
    2. Make sure the distribution of rate estimates is create through the use of a parallelization of your analysis.
    3. Please assess the statistical significance of habitat on the distribution using a simple two-sided t-test (i.e. `t.test()`).
    
  * What inferences with regards to ecology and behavior can you make concerning the relationship between habitat and both shape and rates of morphological evolution? 



In answering your questions, your .Rmd should include the following components:

   * A YAML header that specifies HTML output, the authors, and a bibliography named "BIOL3140.bib". **Submit this bibliography as well!**
   * Sections including an introduction, methods, results, discussion, author contributions, and references. Make sure that each, aside from the references, includes one to two short paragraphs. Specifially:
      + Introduction: Frame the questions, indicating why they are important, what background work has been done in this realm, and how you will answer them. **Please include at least one reference to support the summary of previous work.** Note: this can be done easily by refiguring the introduction to this project report.
      + Methods: Explicitly state how you answered the questions, including a narrative of all the analyses both qualitative and quantitative.
      + Results: Include any appropriate figures or tables and a narrative of the main results that are important to answering the questions.
      + Discussion: Succinctly declare how the results relate to the question and how they compare to previous work devoted to the topic. In addition, be sure to comment on the importance of your findings to the broader topic at hand. **Please include at least one reference to another relevant study.** Note: circling back to the introductions, both to this project description's and yours, will be helpful here.
      + Author contributions: Briefly outline what each team member contributed to the project.

Project reports should be uploaded by 11:59 PM on Sunday, October 20th.

# References

