<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />




<title>Module 7 Project</title>

<script src="site_libs/header-attrs-2.28/header-attrs.js"></script>
<script src="site_libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/flatly.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/jqueryui-1.13.2/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/default.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>
<script src="site_libs/htmlwidgets-1.5.4/htmlwidgets.js"></script>
<link href="site_libs/leaflet-1.3.1/leaflet.css" rel="stylesheet" />
<script src="site_libs/leaflet-1.3.1/leaflet.js"></script>
<link href="site_libs/leafletfix-1.0.0/leafletfix.css" rel="stylesheet" />
<script src="site_libs/proj4-2.6.2/proj4.min.js"></script>
<script src="site_libs/Proj4Leaflet-1.0.1/proj4leaflet.js"></script>
<link href="site_libs/rstudio_leaflet-1.3.1/rstudio_leaflet.css" rel="stylesheet" />
<script src="site_libs/leaflet-binding-2.2.2/leaflet.js"></script>
<script src="site_libs/leaflet-providers-2.0.0/leaflet-providers_2.0.0.js"></script>
<script src="site_libs/leaflet-providers-plugin-2.2.2/leaflet-providers-plugin.js"></script>
<link href="site_libs/HomeButton-0.0.1/home-button.css" rel="stylesheet" />
<script src="site_libs/HomeButton-0.0.1/home-button.js"></script>
<script src="site_libs/HomeButton-0.0.1/easy-button-src.min.js"></script>
<script src="site_libs/clipboard-0.0.1/setClipboardText.js"></script>
<link href="site_libs/mapviewCSS-0.0.1/mapview-popup.css" rel="stylesheet" />
<link href="site_libs/mapviewCSS-0.0.1/mapview.css" rel="stylesheet" />

<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>






<link rel="stylesheet" href="style.css" type="text/css" />



<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
details > summary > p:only-child {
  display: inline;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark the anchor link active (and if it's in a dropdown, also mark that active)
  var dropdown = menuAnchor.closest('li.dropdown');
  if (window.bootstrap) { // Bootstrap 4+
    menuAnchor.addClass('active');
    dropdown.find('> .dropdown-toggle').addClass('active');
  } else { // Bootstrap 3
    menuAnchor.parent().addClass('active');
    dropdown.addClass('active');
  }

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before, .tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "\e259";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "\e258";
  font-family: 'Glyphicons Halflings';
  border: none;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-bs-toggle="collapse" data-target="#navbar" data-bs-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">BIOL 3140</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="schedule.html">Schedule</a>
</li>
<li>
  <a href="at_a_glance.html">Calendar</a>
</li>
<li>
  <a href="WCRs.html">WCRs</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    Projects
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li class="dropdown-header">Phase I</li>
    <li>
      <a href="phaseI_rubric.html">Phase I report guidlines</a>
    </li>
    <li>
      <a href="mod1_proj.html">Module 1 Project</a>
    </li>
    <li>
      <a href="mod2_proj.html">Module 2 Project</a>
    </li>
    <li>
      <a href="mod3_proj.html">Module 3 Project</a>
    </li>
    <li>
      <a href="mod4_proj.html">Module 4 Project</a>
    </li>
    <li class="dropdown-header">Phase II</li>
    <li>
      <a href="phaseII_rubric.html">Phase II report guidlines</a>
    </li>
    <li>
      <a href="mod5_proj.html">Module 5 Project</a>
    </li>
    <li>
      <a href="mod6_proj.html">Module 6 Project</a>
    </li>
    <li>
      <a href="mod7_proj.html">Module 7 Project</a>
    </li>
    <li class="dropdown-header">Phase III</li>
    <li>
      <a href="phaseIII_rubric.html">Phase III report guidlines</a>
    </li>
    <li>
      <a href="final_proj.html">Final Project</a>
    </li>
  </ul>
</li>
<li>
  <a href="https://github.com/orgs/bcorgbio/discussions">BIOL 3140 Discussion</a>
</li>
<li>
  <a href="https://bostoncollege.instructure.com/courses/1659476/gradebook">Grades on Canvas</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    Peer Contribution Evaluations
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="peer_contributions.html">Peer Contribution Evaluation Guidelines</a>
    </li>
    <li>
      <a href="https://forms.gle/ATeFa9wsV1t4g27g6">Author Contribution Evaluation Form</a>
    </li>
  </ul>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">



<h1 class="title toc-ignore">Module 7 Project</h1>

</div>


<style>
div.green { background-color:#c7d1c0; border-radius: 5px; padding: 20px;} 
</style>
<!-- Add icon library -->
<p><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.14.0/css/all.min.css"></p>
<div id="introduction" class="section level1">
<h1>Introduction</h1>
<p><img
src="https://images.ctfassets.net/pujs1b1v0165/sTnbs2M49c9AgPZxp7rDZ/15d77d1064ae1a311ce2f04298e6928b/still_hunting_deer.jpg?w=2400" /></p>
<p>White-tailed deer (WTD, <em>Odocoileus virginianus</em>) are native
to much of North America and represent an ecologically important species
across a diverse array of habitats, from dense mixed forest to open
farm- and grasslands. The currently estimated WTD population of 30
million in the United States is about the same as pre-colonization. Due
to a combination of over hunting and poor resource management practices,
the US population dipped to fewer than 2 million by about 1900, but the
abundance of this species has been on a steady rebound for over the last
100 years.</p>
<p>However, the historical abundance of WTD in New England is largely
different from the continental trend. New England marks the northern
edge of the species’ historical range and WTD were rare in the area,
especially in northern New England. What’s more, due to post-colonial
land-use practices—particarly, near total deforestation for agricultural
purposes–WTD were largely extirpated from the area by the late 18th
century. With changes associated with a move away from an agrarian
economy, reforestation, and the succession of mixed habitats, WTD
abundance slowly increased through the last half of the 20th century.
However, it’s only recently, say within the last few decades, that New
England WTD abundance has in increased sharply, particularly in suburban
areas.</p>
<p>The question of what’s driven this recent sharp increase in WTD
abundance is the focus of this project. In particualar, we’ll ask if WTD
harvest (number taken during hunting season, a proxy for abundance) is
associated with land-use and population (a proxy for suburbanization).
For this we’ll focus on the great state of New Hampshire,
Prof. Kenaley’s birthplace.</p>
<p>To answer such a question, one could, of course, assemble harvest,
land-use, and population data for the Granite State as a whole and
commence with the typical statistical operations (e.g., linear modeling,
a la <code>harvest~population*land-use</code>). However, NH is a rather
diverse state in the two former, potentially predictive variables. In
the north of the state, there are lots of forests. In the southern part
of the state, there a lots more people, and a mix of developed,
forested, and agricultural land. In between, we have a diversity
variably populated scattered amongst forested, farmed, and forgotten
(i.e., successional) areas. That is, each of these attributes vary
spatially across the landscape and focussing on this great state as a
whole wouldn’t account for the important variation across it. We
therefore need to enter the realm of spatial analysis, specifically <a
href="https://en.wikipedia.org/wiki/Spatial_analysis#Spatial_association">spatial
associate</a>. By this, we mean the degree to which things are similarly
arranged in space.</p>
<p><img
src="https://dfzljdn9uc3pi.cloudfront.net/2020/8262/1/fig-1-2x.jpg" /></p>
<p><br> The 1982 WTD densities (A) and 2001–2005 deer densities (B) in
the continental United States.</p>
<p><img
src="https://milliontrees.me/wp-content/uploads/2021/07/deer-population.jpg" /></p>
<p>This species is also the most popular large game animal on the
continent</p>
<p>Just a few decades ago, to assess such a spatial question would
require buying extremely expensive pieces of <a
href="https://en.wikipedia.org/wiki/ArcGIS">software</a>.</p>
<p>Let’s take a momentary break for the icy climes of the granite state
and consider how spatial analysis might be undertaken in R. We’ll
consider examples from a much more beachy place, Bermuda, where, as far
as I know, there are no WTD.</p>
<div id="spatial-analysis-in-r" class="section level2">
<h2>Spatial Analysis in R</h2>
<p>Many R packages have been developed for spatial analysis. Many are
idiosyncratic, requiring one to work with particular object classes that
don’t work well with other data science frameworks like
<code>tidyverse</code>. There are a few, namely <code>sf</code> and
<code>stars</code> that do work well with the <code>tidyverse</code>.
<code>sf</code> provides a table format for simple features, a spatial
object class where feature geometries are stored in a list-column (i.e.,
each row in contains a column as a list of shapes). <code>stars</code>
was written to support raster data, amongst other things (e.g., time
series). The development of both packages came from the bottom up, with
lots of input from the data science community that relies so much on
<code>tidyverse</code>. The packages are designed to work together.
Functions or methods operating on <code>sf</code> or <code>stars</code>
objects start with <code>st_</code>, making it easy to recognize them as
appropriate for spatial tasks in the <code>sf</code> and
<code>stars</code> workflow.</p>
</div>
<div id="the-spatial-analysis-framework" class="section level2">
<h2>The Spatial Analysis Framework</h2>
<p>Spatial analysis is a sprawling field in data science. Although
spatial data and its analysis is often simply treated as longitude and
latitude and some attribute or variable that varies with it, the field
is so much more. Yes, coordinates are important, but so are the patterns
that emerge when we consider how attributes vary within and between
spatial areas and grids.</p>
<p>This representation of data in shape or grid format marks an
important framework within spatial analysis: in addition to point
(coordinate) data, attributes are often aggregated within either shapes
(e.g., lines, polygons, etc.) or grids. The former is considered shape
information, the later raster information, an allusion to pixels of an
image. We can obtain shape or raster data from myriad sources: state and
federal agencies, universities and non-profit organizations, and even
<code>R</code> packages that facilitate data download.</p>
<div id="working-with-shape-data" class="section level3">
<h3>Working with shape data</h3>
<p>Shape data often contain geopolitical information (e.g.,city, state,
or country boundaries) and even topological data (e.g., the boundaries
of rivers, lakes, and oceans, and elevation contours). For instance,
let’s load a shape of the beautiful country of Bermuda using the
<code>rnauturalearth</code> package and kick up <code>sf</code>, the
package of choice that deals in shape data, to produce a simple map:</p>
<pre class="r"><code>library(rnaturalearth)
library(sf)</code></pre>
<pre><code>## Linking to GEOS 3.11.0, GDAL 3.5.3, PROJ 9.1.0; sf_use_s2() is TRUE</code></pre>
<pre class="r"><code>library(tidyverse)</code></pre>
<pre><code>## ── Attaching core tidyverse packages ──────────────────────── tidyverse 2.0.0 ──
## ✔ dplyr     1.1.1     ✔ readr     2.1.4
## ✔ forcats   1.0.0     ✔ stringr   1.5.0
## ✔ ggplot2   3.4.1     ✔ tibble    3.2.1
## ✔ lubridate 1.9.2     ✔ tidyr     1.3.0
## ✔ purrr     1.0.1</code></pre>
<pre><code>## ── Conflicts ────────────────────────────────────────── tidyverse_conflicts() ──
## ✖ dplyr::filter() masks stats::filter()
## ✖ dplyr::lag()    masks stats::lag()
## ℹ Use the conflicted package (&lt;http://conflicted.r-lib.org/&gt;) to force all conflicts to become errors</code></pre>
<pre class="r"><code>bermuda &lt;- ne_states(country=&quot;Bermuda&quot;) %&gt;% 
   st_as_sf

bermuda %&gt;% 
  ggplot()+
  geom_sf()</code></pre>
<p><img src="mod6_proj_files/figure-html/sfintro-1.png" width="672" /></p>
<p>Notice here, that we have to convert the shape retireved from the
<code>rnaturalearth</code> package (an ojbect of class
“SpatialPolygonsDataFrame”) to an <code>sf</code> object. This sort of
conversion is perhaps the most vexing part of spatial analysis in R.
That is, many spatial object classes exist depending on the
<code>R</code> package used and to work with objects across functions of
different origins, conversion needs to happen. Yet, we’re often unsure
of which conversion function to use. We’ll try to keep conversion to a
minimum in this project, but be ware. Fortunately, <code>sf</code> has
the handy <code>st_as_sf()</code> that allows conversion of a spatial
polygon data frame to an <code>sf</code> object.</p>
<p>Because <code>sf</code> plays nice with the <code>tidyverse</code>,
to plot and <code>sf</code> object, we can use the <code>ggplot()</code>
framework that’s become so much of the class. For this we use
<code>sf</code>’s <code>geom_sf()</code>.</p>
</div>
<div id="working-with-raster-data" class="section level3">
<h3>Working with raster data</h3>
<p>Raster data is presented in essentially pixel format. That is, each
pixel in a spatial field represents 3 dimensions: some area of a
particular 2D resolution (e.g., 30 x 30 m, 1 x 1 km) and some value of
attribute, the color.</p>
<pre class="r"><code>library(stars)</code></pre>
<pre><code>## Loading required package: abind</code></pre>
<pre class="r"><code>bermuda_pop &lt;- read_stars(&quot;https://data.worldpop.org/GIS/Population_Density/Global_2000_2020_1km/2020/BMU/bmu_pd_2020_1km.tif&quot;) %&gt;% st_crop(bermuda)


  ggplot()+
  geom_stars(data=bermuda_pop)</code></pre>
<p><img src="mod6_proj_files/figure-html/rastintro-1.png" width="672" /></p>
</div>
<div id="combining-shape-and-raster-data" class="section level3">
<h3>Combining shape and raster data</h3>
<pre class="r"><code>bermuda2 &lt;-  bermuda_pop %&gt;% 
  setNames(&quot;population&quot;) %&gt;% 
  st_as_sf() %&gt;% 
  st_join( bermuda) %&gt;% 
  group_by(name) %&gt;% 
  summarise(population=sum(population))
    


bermuda2 %&gt;% 
  ggplot(aes(fill=population)) + 
  geom_sf()</code></pre>
<p><img src="mod6_proj_files/figure-html/comb-1.png" width="672" /></p>
</div>
<div id="plotting-prety-maps-in-r" class="section level3">
<h3>Plotting prety maps in R</h3>
<pre class="r"><code>library(mapview)

mapview(bermuda2, zcol = &quot;population&quot;)</code></pre>
<div id="htmlwidget-771d8f79400e117cd6fe" style="width:672px;height:480px;" class="leaflet html-widget"></div>
<script type="application/json" data-for="htmlwidget-771d8f79400e117cd6fe">{"x":{"options":{"minZoom":1,"maxZoom":52,"crs":{"crsClass":"L.CRS.EPSG3857","code":null,"proj4def":null,"projectedBounds":null,"options":{}},"preferCanvas":false,"bounceAtZoomLimits":false,"maxBounds":[[[-90,-370]],[[90,370]]]},"calls":[{"method":"addProviderTiles","args":["CartoDB.Positron","CartoDB.Positron","CartoDB.Positron",{"errorTileUrl":"","noWrap":false,"detectRetina":false,"pane":"tilePane"}]},{"method":"addProviderTiles","args":["CartoDB.DarkMatter","CartoDB.DarkMatter","CartoDB.DarkMatter",{"errorTileUrl":"","noWrap":false,"detectRetina":false,"pane":"tilePane"}]},{"method":"addProviderTiles","args":["OpenStreetMap","OpenStreetMap","OpenStreetMap",{"errorTileUrl":"","noWrap":false,"detectRetina":false,"pane":"tilePane"}]},{"method":"addProviderTiles","args":["Esri.WorldImagery","Esri.WorldImagery","Esri.WorldImagery",{"errorTileUrl":"","noWrap":false,"detectRetina":false,"pane":"tilePane"}]},{"method":"addProviderTiles","args":["OpenTopoMap","OpenTopoMap","OpenTopoMap",{"errorTileUrl":"","noWrap":false,"detectRetina":false,"pane":"tilePane"}]},{"method":"createMapPane","args":["polygon",420]},{"method":"addPolygons","args":[[[[{"lng":[-64.7929164147102,-64.8012497480102,-64.8012497480102,-64.7929164147102,-64.7845830814102,-64.7845830814102,-64.7762497481102,-64.7762497481102,-64.7762497481102,-64.7845830814102,-64.7929164147102],"lat":[32.2995833986019,32.2995833986019,32.2912500653019,32.2912500653019,32.2912500653019,32.2829167320019,32.2829167320019,32.2912500653019,32.2995833986019,32.2995833986019,32.2995833986019]}]],[[{"lng":[-64.6762497485102,-64.6845830818102,-64.6845830818102,-64.6845830818102,-64.6762497485102,-64.6679164152102,-64.6679164152102,-64.6679164152102,-64.6762497485102],"lat":[32.3829167316019,32.3829167316019,32.3745833983019,32.3662500650019,32.3662500650019,32.3662500650019,32.3745833983019,32.3829167316019,32.3829167316019]}]],[[{"lng":[-64.7345830816102,-64.7345830816102,-64.7429164149102,-64.7512497482102,-64.7595830815102,-64.7595830815102,-64.7679164148102,-64.7762497481102,-64.7762497481102,-64.7845830814102,-64.7845830814102,-64.7845830814102,-64.7762497481102,-64.7762497481102,-64.7679164148102,-64.7679164148102,-64.7595830815102,-64.7595830815102,-64.7512497482102,-64.7512497482102,-64.7429164149102,-64.7345830816102,-64.7345830816102],"lat":[32.3079167319019,32.3162500652019,32.3162500652019,32.3162500652019,32.3162500652019,32.3079167319019,32.3079167319019,32.3079167319019,32.2995833986019,32.2995833986019,32.2912500653019,32.2829167320019,32.2829167320019,32.2912500653019,32.2912500653019,32.2829167320019,32.2829167320019,32.2912500653019,32.2912500653019,32.2995833986019,32.2995833986019,32.2995833986019,32.3079167319019]}]],[[{"lng":[-64.7095830817102,-64.7179164150102,-64.7179164150102,-64.7179164150102,-64.7095830817102,-64.7095830817102,-64.7012497484102,-64.6929164151102,-64.6929164151102,-64.6929164151102,-64.7012497484102,-64.7095830817102,-64.7095830817102],"lat":[32.3329167318019,32.3329167318019,32.3245833985019,32.3162500652019,32.3162500652019,32.3245833985019,32.3245833985019,32.3245833985019,32.3329167318019,32.3412500651019,32.3412500651019,32.3412500651019,32.3329167318019]}],[{"lng":[-64.7429164149102,-64.7429164149102,-64.7429164149102,-64.7345830816102,-64.7262497483102,-64.7262497483102,-64.7345830816102,-64.7345830816102,-64.7262497483102,-64.7262497483102,-64.7179164150102,-64.7095830817102,-64.7095830817102,-64.7012497484102,-64.7012497484102,-64.7095830817102,-64.7179164150102,-64.7179164150102,-64.7262497483102,-64.7345830816102,-64.7345830816102,-64.7429164149102,-64.7429164149102],"lat":[32.3329167318019,32.3245833985019,32.3162500652019,32.3162500652019,32.3162500652019,32.3245833985019,32.3245833985019,32.3329167318019,32.3329167318019,32.3412500651019,32.3412500651019,32.3412500651019,32.3495833984019,32.3495833984019,32.3579167317019,32.3579167317019,32.3579167317019,32.3495833984019,32.3495833984019,32.3495833984019,32.3412500651019,32.3412500651019,32.3329167318019]}]],[[{"lng":[-64.8012497480102,-64.8012497480102,-64.7929164147102,-64.7845830814102,-64.7762497481102,-64.7762497481102,-64.7679164148102,-64.7679164148102,-64.7595830815102,-64.7595830815102,-64.7595830815102,-64.7679164148102,-64.7762497481102,-64.7845830814102,-64.7845830814102,-64.7845830814102,-64.7929164147102,-64.8012497480102,-64.8012497480102],"lat":[32.2745833987019,32.2662500654019,32.2662500654019,32.2662500654019,32.2662500654019,32.2745833987019,32.2745833987019,32.2829167320019,32.2829167320019,32.2912500653019,32.2995833986019,32.2995833986019,32.2995833986019,32.2995833986019,32.2912500653019,32.2829167320019,32.2829167320019,32.2829167320019,32.2745833987019]}],[{"lng":[-64.8179164146102,-64.8095830813102,-64.8095830813102,-64.8179164146102,-64.8179164146102],"lat":[32.2829167320019,32.2829167320019,32.2912500653019,32.2912500653019,32.2829167320019]}]],[[{"lng":[-64.8095830813102,-64.8179164146102,-64.8179164146102,-64.8095830813102,-64.8095830813102,-64.8012497480102,-64.7929164147102,-64.7845830814102,-64.7845830814102,-64.7762497481102,-64.7762497481102,-64.7762497481102,-64.7845830814102,-64.7845830814102,-64.7929164147102,-64.8012497480102,-64.8095830813102],"lat":[32.3079167319019,32.3079167319019,32.2995833986019,32.2995833986019,32.2912500653019,32.2912500653019,32.2912500653019,32.2912500653019,32.2829167320019,32.2829167320019,32.2912500653019,32.2995833986019,32.2995833986019,32.3079167319019,32.3079167319019,32.3079167319019,32.3079167319019]}]],[[{"lng":[-64.6512497486102,-64.6512497486102,-64.6595830819102,-64.6595830819102,-64.6679164152102,-64.6762497485102,-64.6845830818102,-64.6929164151102,-64.6929164151102,-64.7012497484102,-64.7012497484102,-64.7095830817102,-64.7179164150102,-64.7179164150102,-64.7179164150102,-64.7095830817102,-64.7012497484102,-64.7012497484102,-64.6929164151102,-64.6845830818102,-64.6762497485102,-64.6679164152102,-64.6595830819102,-64.6512497486102,-64.6512497486102],"lat":[32.3662500650019,32.3745833983019,32.3745833983019,32.3829167316019,32.3829167316019,32.3829167316019,32.3829167316019,32.3829167316019,32.3745833983019,32.3745833983019,32.3662500650019,32.3662500650019,32.3662500650019,32.3579167317019,32.3495833984019,32.3495833984019,32.3495833984019,32.3579167317019,32.3579167317019,32.3579167317019,32.3579167317019,32.3579167317019,32.3579167317019,32.3579167317019,32.3662500650019]}],[{"lng":[-64.7095830817102,-64.7012497484102,-64.6929164151102,-64.6929164151102,-64.6845830818102,-64.6845830818102,-64.6929164151102,-64.7012497484102,-64.7095830817102,-64.7095830817102,-64.7095830817102],"lat":[32.3245833985019,32.3245833985019,32.3245833985019,32.3329167318019,32.3329167318019,32.3412500651019,32.3412500651019,32.3412500651019,32.3412500651019,32.3329167318019,32.3245833985019]}]],[[{"lng":[-64.8679164144102,-64.8762497477102,-64.8762497477102,-64.8762497477102,-64.8845830810102,-64.8845830810102,-64.8845830810102,-64.8762497477102,-64.8679164144102,-64.8595830811102,-64.8512497478102,-64.8512497478102,-64.8595830811102,-64.8679164144102,-64.8679164144102,-64.8679164144102,-64.8595830811102,-64.8595830811102,-64.8679164144102],"lat":[32.2995833986019,32.2995833986019,32.2912500653019,32.2829167320019,32.2829167320019,32.2745833987019,32.2662500654019,32.2662500654019,32.2662500654019,32.2662500654019,32.2662500654019,32.2745833987019,32.2745833987019,32.2745833987019,32.2829167320019,32.2912500653019,32.2912500653019,32.2995833986019,32.2995833986019]}]],[[{"lng":[-64.7429164149102,-64.7512497482102,-64.7512497482102,-64.7429164149102,-64.7429164149102,-64.7345830816102,-64.7345830816102,-64.7262497483102,-64.7262497483102,-64.7179164150102,-64.7095830817102,-64.7095830817102,-64.7095830817102,-64.7179164150102,-64.7179164150102,-64.7262497483102,-64.7345830816102,-64.7429164149102,-64.7429164149102],"lat":[32.3162500652019,32.3162500652019,32.3079167319019,32.3079167319019,32.2995833986019,32.2995833986019,32.3079167319019,32.3079167319019,32.3162500652019,32.3162500652019,32.3162500652019,32.3245833985019,32.3329167318019,32.3329167318019,32.3245833985019,32.3245833985019,32.3245833985019,32.3245833985019,32.3162500652019]}]],[[{"lng":[-64.8762497477102,-64.8679164144102,-64.8679164144102,-64.8595830811102,-64.8595830811102,-64.8679164144102,-64.8762497477102,-64.8845830810102,-64.8845830810102,-64.8762497477102,-64.8762497477102],"lat":[32.2579167321019,32.2579167321019,32.2662500654019,32.2662500654019,32.2745833987019,32.2745833987019,32.2745833987019,32.2745833987019,32.2662500654019,32.2662500654019,32.2579167321019]}],[{"lng":[-64.8679164144102,-64.8595830811102,-64.8512497478102,-64.8429164145102,-64.8345830812102,-64.8262497479102,-64.8179164146102,-64.8179164146102,-64.8095830813102,-64.8095830813102,-64.8179164146102,-64.8262497479102,-64.8345830812102,-64.8345830812102,-64.8429164145102,-64.8512497478102,-64.8595830811102,-64.8679164144102,-64.8679164144102],"lat":[32.2495833988019,32.2495833988019,32.2495833988019,32.2495833988019,32.2495833988019,32.2495833988019,32.2495833988019,32.2579167321019,32.2579167321019,32.2662500654019,32.2662500654019,32.2662500654019,32.2662500654019,32.2579167321019,32.2579167321019,32.2579167321019,32.2579167321019,32.2579167321019,32.2495833988019]}]],[[{"lng":[-64.8179164146102,-64.8262497479102,-64.8345830812102,-64.8345830812102,-64.8262497479102,-64.8179164146102,-64.8095830813102,-64.8095830813102,-64.8179164146102],"lat":[32.2912500653019,32.2912500653019,32.2912500653019,32.2829167320019,32.2829167320019,32.2829167320019,32.2829167320019,32.2912500653019,32.2912500653019]}],[{"lng":[-64.8262497479102,-64.8345830812102,-64.8345830812102,-64.8262497479102,-64.8179164146102,-64.8095830813102,-64.8012497480102,-64.7929164147102,-64.7929164147102,-64.7845830814102,-64.7845830814102,-64.7929164147102,-64.7929164147102,-64.8012497480102,-64.8012497480102,-64.8095830813102,-64.8179164146102,-64.8179164146102,-64.8262497479102],"lat":[32.2662500654019,32.2662500654019,32.2579167321019,32.2579167321019,32.2579167321019,32.2579167321019,32.2579167321019,32.2579167321019,32.2662500654019,32.2662500654019,32.2745833987019,32.2745833987019,32.2829167320019,32.2829167320019,32.2745833987019,32.2745833987019,32.2745833987019,32.2662500654019,32.2662500654019]}]]],null,"bermuda2 - population",{"crs":{"crsClass":"L.CRS.EPSG3857","code":null,"proj4def":null,"projectedBounds":null,"options":{}},"pane":"polygon","stroke":true,"color":"#333333","weight":0.5,"opacity":[0.9,0.9,0.9,0.9,0.9,0.9,0.9,0.9,0.9,0.9,0.9],"fill":true,"fillColor":["#006791","#4B0055","#FDE333","#00C475","#7ED357","#00C475","#7ED357","#006791","#008298","#00C475","#7ED357"],"fillOpacity":[0.6,0.6,0.6,0.6,0.6,0.6,0.6,0.6,0.6,0.6,0.6],"smoothFactor":1,"noClip":false},["<div class='scrollableContainer'><table class=mapview-popup id='popup'><tr class='coord'><td><\/td><th><b>Feature ID&emsp;<\/b><\/th><td>1&emsp;<\/td><\/tr><tr><td>1<\/td><th>name&emsp;<\/th><td>City of Hamilton&emsp;<\/td><\/tr><tr><td>2<\/td><th>population&emsp;<\/th><td> 6340.716&emsp;<\/td><\/tr><tr><td>3<\/td><th>geometry&emsp;<\/th><td>sfc_MULTIPOLYGON&emsp;<\/td><\/tr><\/table><\/div>","<div class='scrollableContainer'><table class=mapview-popup id='popup'><tr class='coord'><td><\/td><th><b>Feature ID&emsp;<\/b><\/th><td>2&emsp;<\/td><\/tr><tr><td>1<\/td><th>name&emsp;<\/th><td>City of Saint George&emsp;<\/td><\/tr><tr><td>2<\/td><th>population&emsp;<\/th><td> 2112.074&emsp;<\/td><\/tr><tr><td>3<\/td><th>geometry&emsp;<\/th><td>sfc_MULTIPOLYGON&emsp;<\/td><\/tr><\/table><\/div>","<div class='scrollableContainer'><table class=mapview-popup id='popup'><tr class='coord'><td><\/td><th><b>Feature ID&emsp;<\/b><\/th><td>3&emsp;<\/td><\/tr><tr><td>1<\/td><th>name&emsp;<\/th><td>Devonshire&emsp;<\/td><\/tr><tr><td>2<\/td><th>population&emsp;<\/th><td>17478.636&emsp;<\/td><\/tr><tr><td>3<\/td><th>geometry&emsp;<\/th><td>sfc_MULTIPOLYGON&emsp;<\/td><\/tr><\/table><\/div>","<div class='scrollableContainer'><table class=mapview-popup id='popup'><tr class='coord'><td><\/td><th><b>Feature ID&emsp;<\/b><\/th><td>4&emsp;<\/td><\/tr><tr><td>1<\/td><th>name&emsp;<\/th><td>Hamilton&emsp;<\/td><\/tr><tr><td>2<\/td><th>population&emsp;<\/th><td>12606.292&emsp;<\/td><\/tr><tr><td>3<\/td><th>geometry&emsp;<\/th><td>sfc_MULTIPOLYGON&emsp;<\/td><\/tr><\/table><\/div>","<div class='scrollableContainer'><table class=mapview-popup id='popup'><tr class='coord'><td><\/td><th><b>Feature ID&emsp;<\/b><\/th><td>5&emsp;<\/td><\/tr><tr><td>1<\/td><th>name&emsp;<\/th><td>Paget&emsp;<\/td><\/tr><tr><td>2<\/td><th>population&emsp;<\/th><td>14804.220&emsp;<\/td><\/tr><tr><td>3<\/td><th>geometry&emsp;<\/th><td>sfc_MULTIPOLYGON&emsp;<\/td><\/tr><\/table><\/div>","<div class='scrollableContainer'><table class=mapview-popup id='popup'><tr class='coord'><td><\/td><th><b>Feature ID&emsp;<\/b><\/th><td>6&emsp;<\/td><\/tr><tr><td>1<\/td><th>name&emsp;<\/th><td>Pembroke&emsp;<\/td><\/tr><tr><td>2<\/td><th>population&emsp;<\/th><td>12243.552&emsp;<\/td><\/tr><tr><td>3<\/td><th>geometry&emsp;<\/th><td>sfc_MULTIPOLYGON&emsp;<\/td><\/tr><\/table><\/div>","<div class='scrollableContainer'><table class=mapview-popup id='popup'><tr class='coord'><td><\/td><th><b>Feature ID&emsp;<\/b><\/th><td>7&emsp;<\/td><\/tr><tr><td>1<\/td><th>name&emsp;<\/th><td>Saint George's&emsp;<\/td><\/tr><tr><td>2<\/td><th>population&emsp;<\/th><td>14198.692&emsp;<\/td><\/tr><tr><td>3<\/td><th>geometry&emsp;<\/th><td>sfc_MULTIPOLYGON&emsp;<\/td><\/tr><\/table><\/div>","<div class='scrollableContainer'><table class=mapview-popup id='popup'><tr class='coord'><td><\/td><th><b>Feature ID&emsp;<\/b><\/th><td>8&emsp;<\/td><\/tr><tr><td>1<\/td><th>name&emsp;<\/th><td>Sandys&emsp;<\/td><\/tr><tr><td>2<\/td><th>population&emsp;<\/th><td> 6111.398&emsp;<\/td><\/tr><tr><td>3<\/td><th>geometry&emsp;<\/th><td>sfc_MULTIPOLYGON&emsp;<\/td><\/tr><\/table><\/div>","<div class='scrollableContainer'><table class=mapview-popup id='popup'><tr class='coord'><td><\/td><th><b>Feature ID&emsp;<\/b><\/th><td>9&emsp;<\/td><\/tr><tr><td>1<\/td><th>name&emsp;<\/th><td>Smith's&emsp;<\/td><\/tr><tr><td>2<\/td><th>population&emsp;<\/th><td> 8230.446&emsp;<\/td><\/tr><tr><td>3<\/td><th>geometry&emsp;<\/th><td>sfc_MULTIPOLYGON&emsp;<\/td><\/tr><\/table><\/div>","<div class='scrollableContainer'><table class=mapview-popup id='popup'><tr class='coord'><td><\/td><th><b>Feature ID&emsp;<\/b><\/th><td>10&emsp;<\/td><\/tr><tr><td>1<\/td><th>name&emsp;<\/th><td>Southampton&emsp;<\/td><\/tr><tr><td>2<\/td><th>population&emsp;<\/th><td>13538.276&emsp;<\/td><\/tr><tr><td>3<\/td><th>geometry&emsp;<\/th><td>sfc_MULTIPOLYGON&emsp;<\/td><\/tr><\/table><\/div>","<div class='scrollableContainer'><table class=mapview-popup id='popup'><tr class='coord'><td><\/td><th><b>Feature ID&emsp;<\/b><\/th><td>11&emsp;<\/td><\/tr><tr><td>1<\/td><th>name&emsp;<\/th><td>Warwick&emsp;<\/td><\/tr><tr><td>2<\/td><th>population&emsp;<\/th><td>14461.775&emsp;<\/td><\/tr><tr><td>3<\/td><th>geometry&emsp;<\/th><td>sfc_MULTIPOLYGON&emsp;<\/td><\/tr><\/table><\/div>"],{"maxWidth":800,"minWidth":50,"autoPan":true,"keepInView":false,"closeButton":true,"closeOnClick":true,"className":""},["6340.71606445312","2112.07443237305","17478.6361694336","12606.2922363281","14804.2196350098","12243.551574707","14198.6921081543","6111.39785766602","8230.44647216797","13538.2760009766","14461.7752380371"],{"interactive":false,"permanent":false,"direction":"auto","opacity":1,"offset":[0,0],"textsize":"10px","textOnly":false,"className":"","sticky":true},{"stroke":true,"weight":1,"opacity":0.9,"fillOpacity":0.84,"bringToFront":false,"sendToBack":false}]},{"method":"addScaleBar","args":[{"maxWidth":100,"metric":true,"imperial":true,"updateWhenIdle":true,"position":"bottomleft"}]},{"method":"addHomeButton","args":[-64.8845830810102,32.2495833988019,-64.6512497486102,32.3829167316019,true,"bermuda2 - population","Zoom to bermuda2 - population","<strong> bermuda2 - population <\/strong>","bottomright"]},{"method":"addLayersControl","args":[["CartoDB.Positron","CartoDB.DarkMatter","OpenStreetMap","Esri.WorldImagery","OpenTopoMap"],"bermuda2 - population",{"collapsed":true,"autoZIndex":true,"position":"topleft"}]},{"method":"addLegend","args":[{"colors":["#4B0055 , #403073 12.285933574026%, #1C598A 25.3012068291776%, #017D97 38.3164800843292%, #029E94 51.3317533394808%, #05BA81 64.3470265946324%, #6BCF60 77.362299849784%, #C4DE34 90.3775731049356%, #FDE333 "],"labels":["4,000","6,000","8,000","10,000","12,000","14,000","16,000"],"na_color":null,"na_label":"NA","opacity":1,"position":"topright","type":"numeric","title":"bermuda2 - population","extra":{"p_1":0.12285933574026,"p_n":0.903775731049356},"layerId":null,"className":"info legend","group":"bermuda2 - population"}]}],"limits":{"lat":[32.2495833988019,32.3829167316019],"lng":[-64.8845830810102,-64.6512497486102]},"fitBounds":[32.2495833988019,-64.8845830810102,32.3829167316019,-64.6512497486102,[]]},"evals":[],"jsHooks":{"render":[{"code":"function(el, x, data) {\n  return (\n      function(el, x, data) {\n      // get the leaflet map\n      var map = this; //HTMLWidgets.find('#' + el.id);\n      // we need a new div element because we have to handle\n      // the mouseover output separately\n      // debugger;\n      function addElement () {\n      // generate new div Element\n      var newDiv = $(document.createElement('div'));\n      // append at end of leaflet htmlwidget container\n      $(el).append(newDiv);\n      //provide ID and style\n      newDiv.addClass('lnlt');\n      newDiv.css({\n      'position': 'relative',\n      'bottomleft':  '0px',\n      'background-color': 'rgba(255, 255, 255, 0.7)',\n      'box-shadow': '0 0 2px #bbb',\n      'background-clip': 'padding-box',\n      'margin': '0',\n      'padding-left': '5px',\n      'color': '#333',\n      'font': '9px/1.5 \"Helvetica Neue\", Arial, Helvetica, sans-serif',\n      'z-index': '700',\n      });\n      return newDiv;\n      }\n\n\n      // check for already existing lnlt class to not duplicate\n      var lnlt = $(el).find('.lnlt');\n\n      if(!lnlt.length) {\n      lnlt = addElement();\n\n      // grab the special div we generated in the beginning\n      // and put the mousmove output there\n\n      map.on('mousemove', function (e) {\n      if (e.originalEvent.ctrlKey) {\n      if (document.querySelector('.lnlt') === null) lnlt = addElement();\n      lnlt.text(\n                           ' lon: ' + (e.latlng.lng).toFixed(5) +\n                           ' | lat: ' + (e.latlng.lat).toFixed(5) +\n                           ' | zoom: ' + map.getZoom() +\n                           ' | x: ' + L.CRS.EPSG3857.project(e.latlng).x.toFixed(0) +\n                           ' | y: ' + L.CRS.EPSG3857.project(e.latlng).y.toFixed(0) +\n                           ' | epsg: 3857 ' +\n                           ' | proj4: +proj=merc +a=6378137 +b=6378137 +lat_ts=0.0 +lon_0=0.0 +x_0=0.0 +y_0=0 +k=1.0 +units=m +nadgrids=@null +no_defs ');\n      } else {\n      if (document.querySelector('.lnlt') === null) lnlt = addElement();\n      lnlt.text(\n                      ' lon: ' + (e.latlng.lng).toFixed(5) +\n                      ' | lat: ' + (e.latlng.lat).toFixed(5) +\n                      ' | zoom: ' + map.getZoom() + ' ');\n      }\n      });\n\n      // remove the lnlt div when mouse leaves map\n      map.on('mouseout', function (e) {\n      var strip = document.querySelector('.lnlt');\n      if( strip !==null) strip.remove();\n      });\n\n      };\n\n      //$(el).keypress(67, function(e) {\n      map.on('preclick', function(e) {\n      if (e.originalEvent.ctrlKey) {\n      if (document.querySelector('.lnlt') === null) lnlt = addElement();\n      lnlt.text(\n                      ' lon: ' + (e.latlng.lng).toFixed(5) +\n                      ' | lat: ' + (e.latlng.lat).toFixed(5) +\n                      ' | zoom: ' + map.getZoom() + ' ');\n      var txt = document.querySelector('.lnlt').textContent;\n      console.log(txt);\n      //txt.innerText.focus();\n      //txt.select();\n      setClipboardText('\"' + txt + '\"');\n      }\n      });\n\n      }\n      ).call(this.getMap(), el, x, data);\n}","data":null},{"code":"function(el, x, data) {\n  return (function(el,x,data){\n           var map = this;\n\n           map.on('keypress', function(e) {\n               console.log(e.originalEvent.code);\n               var key = e.originalEvent.code;\n               if (key === 'KeyE') {\n                   var bb = this.getBounds();\n                   var txt = JSON.stringify(bb);\n                   console.log(txt);\n\n                   setClipboardText('\\'' + txt + '\\'');\n               }\n           })\n        }).call(this.getMap(), el, x, data);\n}","data":null}]}}</script>
</div>
</div>
<div id="methods" class="section level2">
<h2>Methods</h2>
<p>Back to the frozen enclace of NH.</p>
<div id="data-source" class="section level3">
<h3>Data source</h3>
<p>OK, let’s first address the obvious. We will be using deer harvest
data as a proxy for WTD abundance. Understandably, for some this may be
a difficult thing with which to grapple. After all, we’re considering
how many times a majestic animal has been killed, to put in plainly.
There are many ethical considerations here and I’m happy to discuss them
all in private. But, if we can, let’s just take the data at face value:
a measure of abundance.</p>
<p>Our deer harvest data comes from New Hampshire Department of Fish and
Wildlife’s last three <a
href="https://www.wildlife.nh.gov/hunting-nh/hunting-reports-harvest-summaries">annual
harvest summaries</a> from 2021-2023. These</p>
<pre class="r"><code>deer &lt;- read_csv(&quot;data/NH_deer.txt&quot;) %&gt;% 
  mutate(pbpNAME=
           str_replace(TOWN,&quot;_&quot;,&quot; &quot;) %&gt;% str_to_title(),
         HSM=as.numeric(HSM),
         TOTAL=as.numeric(TOTAL),
         ) </code></pre>
<pre><code>## Warning: One or more parsing issues, call `problems()` on your data frame for details,
## e.g.:
##   dat &lt;- vroom(...)
##   problems(dat)</code></pre>
<pre><code>## Rows: 730 Columns: 7
## ── Column specification ────────────────────────────────────────────────────────
## Delimiter: &quot;,&quot;
## chr (3): TOWN, WMU, MALE
## dbl (3): FEMALE, TOTAL, HSM
## num (1): year
## 
## ℹ Use `spec()` to retrieve the full column specification for this data.
## ℹ Specify the column types or set `show_col_types = FALSE` to quiet this message.</code></pre>
</div>
<div id="shape-data" class="section level3">
<h3>Shape data</h3>
<pre class="r"><code>#intro:
#https://r-spatial.org/book/07-Int/rosf.html

#https://www.paulamoraga.com/book-spatial/spatial-autocorrelation.html

library(rnaturalearth)

library(sf)
library(tidyverse)
library(mapview)


#change to URL
nh &lt;- read_sf(&quot;data/NH/New_Hampshire_Political_Boundaries.shp&quot;)

ggplot() +
  geom_sf(data = nh)+theme_void()</code></pre>
<p><img src="mod6_proj_files/figure-html/nhpol-1.png" width="672" /></p>
</div>
<div id="raster-data" class="section level3">
<h3>Raster data</h3>
<pre class="r"><code>library(raster)</code></pre>
<pre><code>## Loading required package: sp</code></pre>
<pre><code>## 
## Attaching package: &#39;raster&#39;</code></pre>
<pre><code>## The following object is masked from &#39;package:dplyr&#39;:
## 
##     select</code></pre>
<pre class="r"><code>#large data set, 30m resolution
lu &lt;- stars::read_stars(&quot;ignore/data/land_use/NLCD_2021_Land_Cover_L48_20230630_qtPZ6zMaTAibmibqfd37.tiff&quot;)

nh_crs&lt;- crs(nh)
new_crs &lt;-  st_crs(nh_crs)

#low res for plotting
lu2 &lt;- lu %&gt;% 
 setNames(&quot;land_use&quot;) %&gt;% 
     st_transform_proj( new_crs) %&gt;% 
st_downsample(30,mean) %&gt;% 
  st_crop(nh)</code></pre>
<pre><code>## Warning in st_crop.stars(., nh): crop only crops regular grids</code></pre>
<pre class="r"><code>plot(lu2)</code></pre>
<p><img src="mod6_proj_files/figure-html/landuse-1.png" width="672" /></p>
</div>
<div id="adding-custom-data-to-shape-data" class="section level3">
<h3>Adding custom data to shape data</h3>
<pre class="r"><code>nh_deer &lt;- nh %&gt;% left_join(deer) 
nh_deer %&gt;%filter(HSM==is.na(HSM)) %&gt;%  dplyr::select(pbpNAME,year,HSM) 
 
ggplot() +
  geom_sf(data=nh_deer %&gt;% na.omit,aes(fill=HSM))+theme_void()+
  facet_wrap(.~year)+
  scale_fill_viridis_c(option = &quot;magma&quot;,na.value = 0)

nh_deer %&gt;% 
  ungroup %&gt;% 
  mutate(pop=PB_TOWN_16 %&gt;% as.numeric()) %&gt;% 
  group_by(pbpNAME) %&gt;% 
  dplyr::summarise(pop=mean(pop) %&gt;% log,KSM=mean(KSM,na.rm=T) %&gt;% log) %&gt;%
  ggplot(aes(pop,KSM))+geom_point(alpha=0.3)+
  geom_smooth(method=&quot;polyx&quot;)+ 
  scale_x_log10()

nh_deer %&gt;% 
  ggplot(aes(PB_TOWN_16 %&gt;% as.numeric,KSM))+geom_point(alpha=0.3)+
  geom_smooth(method=&quot;lm&quot;,formula=y ~ poly(x, 4, raw=TRUE)) + 
  scale_x_log10()



mapview(nh_deer, zcol = &quot;HSM&quot;)</code></pre>
<pre class="r"><code>#https://www.paulamoraga.com/book-spatial/spatial-autocorrelation.html
# https://r-spatial.org/book/07-Introsf.html
#javascript:alert(&#39;Sign-in to download data&#39;)

#stars way
lu &lt;- stars::read_stars(&quot;ignore/data/land_use/NLCD_2021_Land_Cover_L48_20230630_qtPZ6zMaTAibmibqfd37.tiff&quot;)

nh_crs&lt;- crs(nh)
new_crs &lt;-  st_crs(nh_crs)






plot(lu2)


lu3 &lt;- lu2 %&gt;% 
   st_crop(nh) %&gt;% 
  as.data.frame(xy=T)  %&gt;% 
  na.omit() 




 deer_lu &lt;-  lu2 %&gt;% 
   st_crop(nh) %&gt;%
  st_as_sf() %&gt;% 
     st_join(nh_deer)
 
 library(units)
 deer_lu %&gt;% 
   mutate(area=st_area(deer_lu)) %&gt;% 
   as.data.frame() %&gt;% 
   group_by(pbpNAME) %&gt;% 
   mutate(area_tot=sum(area)) %&gt;% 
     group_by(pbpNAME,land_use) %&gt;% 
   summarize(area_lu=sum(area)/area_tot) %&gt;% 
   ungroup %&gt;% 
   filter(pbpNAME==&quot;Rochester&quot;) %&gt;% 
   ggplot(aes(x=land_use,area_lu,fill=land_use)) +geom_bar(stat=&quot;identity&quot;) +facet_wrap(.~pbpNAME)+coord_flip()
   
   
wlu2 %&gt;% 
  st_crop(nh) %&gt;% 
  as.data.frame(xy=T)  %&gt;% 
  na.omit() 
  

ggplot()+
  geom_stars(data=lu2 %&gt;% st_crop(nh))+
    theme_void()+scale_fill_viridis_d(option = &quot;C&quot;)




class(nh)</code></pre>
<p>gpw &lt;- st_as_sf(gpw_raw) plot(gpw_raw)</p>
<p>lu &lt;-
terra::rast(“ignore/data/land_use/NLCD_2021_Land_Cover_L48_20230630_qtPZ6zMaTAibmibqfd37.tiff”)</p>
<p>lu2 %&gt;% filter(<code>NLCD Land Cover Class</code>!=“Unclassified”)
%&gt;% plot()</p>
<p>lu2 &lt;- lu res(lu2) &lt;-600 lu2 &lt;- resample(lu, lu2)
plot(lu2)</p>
<p>lu2 &lt;- st_as_stars(lu2)</p>
<p>coords &lt;- crds(lu, na.rm=FALSE) geo_coords &lt;- project(coords,
crs(lu2), “+proj=longlat”)</p>
<p>library(rgdal) library(raster) projectRaster(lu2,
crs=CRS(“+proj=longlat”))</p>
<p>lu[ 1:50, 1:50]</p>
<p>lu_poly &lt;- st_as_sf(lu2, as_points = FALSE, merge = T)</p>
<p>lu_poly %&gt;%
filter(<code>NLCD Land Cover Class</code>!=“Unclassified”) %&gt;%
ggplot(aes(fill=<code>NLCD Land Cover Class</code>))+geom_sf()</p>
<p>#<a href="https://doi90.github.io/EE_Intro_Spatial_Workshop/"
class="uri">https://doi90.github.io/EE_Intro_Spatial_Workshop/</a></p>
<p>lu2 &lt;- st_transform(lu2,crs(nh_))</p>
<p>st_as_sf(lu2, as_points = TRUE, merge = FALSE) lu_poly &lt;-
st_as_sf(lu2, as_points = FALSE, merge = FALSE) lu_poly %&gt;% plot(
key.pos = 1, pal = sf.colors, lwd = 2, key.length = 0.8)
stars::st_as_sf(lu2, as_points = FALSE, merge = TRUE) lu_poly &lt;-
as.polygons(lu2) p = st_as_sf(lu, as_points = FALSE, merge = TRUE)
lu_poly %&gt;%
filter(<code>NLCD Land Cover Class</code>!=“Unclassified”) %&gt;% plot
plot(p)</p>
<pre><code>




  select(starts_with(&quot;Canis&quot;)|starts_with(&quot;snp&quot;)) %&gt;% 
  pivot_longer(cols = contains(&quot;Canis&quot;))
mds &lt;- snps %&gt;%
  dist() %&gt;%          
  cmdscale() %&gt;%
  as_tibble()

colnames(mds) &lt;- c(&quot;Dim1&quot;, &quot;Dim2&quot;)

wolf &lt;- cl_mds %&gt;% 
  filter(species==&quot;Canis lupus&quot;)%&gt;%
  select()
  summarize(Dim1=mean(Dim1),Dim2=mean(Dim2))

cl_mds &lt;- mds %&gt;% mutate(id=rownames(snps)) %&gt;% left_join(cl_local)
cl_mds %&gt;% 
  group_by(state,species,id) %&gt;%
  mutate(dist=sum(c(abs(Dim1-wolf$Dim1),abs(Dim2-wolf$Dim2)))) %&gt;% 
  ggplot(aes(state,dist))+geom_boxplot()
  
  ggplot(aes(Dim1,Dim2,col=species,shape=Region))+geom_point()

</code></pre>
</div>
</div>
</div>
<div id="section" class="section level1">
<h1></h1>
<p><a
href="https://royalsocietypublishing.org/doi/abs/10.1098/rsbl.2009.0575"
class="uri">https://royalsocietypublishing.org/doi/abs/10.1098/rsbl.2009.0575</a></p>
</div>
<div id="methods-1" class="section level1">
<h1>Methods</h1>
<div id="experimental-design" class="section level2">
<h2>Experimental Design</h2>
<p>To assess the affect of eccentric fatigue of elbow flexor muscles,
we’ll be measuring the isometric force over the range of motion of the
forearm produced by MVC. To capture MVC data, you must design and
construct a portable Arduino-based force-data acquisition system (DAQ).
Your device should record force data over the course of 30s.</p>
</div>
<div id="daq-design-components" class="section level2">
<h2>DAQ Design Components</h2>
<ol style="list-style-type: decimal">
<li>100-kg S-type Load Cell with HX711 amplifier</li>
<li>Arduino Uno microcontroller</li>
</ol>
</div>
<div id="daq-assembly-and-programming" class="section level2">
<h2>DAQ Assembly and Programming</h2>
<p>Assembly for this force logger is almost entirely captured in <a
href="https://www.instructables.com/Arduino-Tension-Scale-With-40-Kg-Luggage-Load-Cell/">this
tutorial</a>. And, Prof. K has wired the load cell to the amplifier, so
y’all just need to wire up the ampflifier to the Arduino</p>
<p><del>You’ll see that the sketch in the tutorial requires a
calibration step to output appropriate force or mass units.</del>
<strong>Don’t worry about calibration.</strong> We’ll be looking at
normalized force, so units don’t matter.</p>
<!-- As for your SD storage integration, make sure your sketch writes one column data to the SD card with the header "force". -->
<p>For each experiment, you’ll be recording 12 bouts of MVC force, each
corresponding to a different elbow angle, for 30 s. For each arm angle,
one only needs to unplug the Arduino and hit the red restart button on
the Arduino to begin taking force data. So as not to confuse from where
your data came, it’s a good idea to change the name of the text file
being saved to for each experiment, e.g., “Team_subject1_control.csv”
and “Team_subject1_fatigue.csv”.</p>
</div>
<div id="data-collection" class="section level2">
<h2>Data Collection</h2>
<div id="elbow-angle" class="section level3">
<h3>Elbow Angle</h3>
<p>You’ll first need to construct a ganiometer, a simple device that
measures the angle between the upper and lower arm (i.e., the elbow
angle). One approach is to use the outline of a small plate to cut a
disc out of cardboard and then bisect the diameter and then continue to
bisect this semicircle into equal segments (Fig. 4.)</p>
<center>
<img src="gan.jpg" style="width:50.0%" /> <br> <strong>Fig. 4</strong> A
simple ganiometer made of cardboard.
</center>
<p><br></p>
<p>To use the ganiometer, find the center of the elbow joint in lateral
view and mark this position with a marker. From this point, mark a line
along the lateral margin of the radius. (Fig. 5). Then, poke a hole
through the center of the ganiometer and affix it to the upper arm so
its center aligns with your mark at the elbow joint and the 0-180<span
class="math inline">\(^o\)</span> axis aligns with the long axis of the
humerus (Fig. 6).You can affix the ganiometer to the upper arm by
passing a strip of surgical tape through two tabs cut out on either side
of the 0-180<span class="math inline">\(^o\)</span> axis (Fig. 6).</p>
<center>
<img src="joint.jpg" style="width:50.0%" /> <br> <strong>Fig. 4</strong>
The elbow joint on a tiny human and the lateral margin of the radius
marked with a dark line. Thanks, Zo!
</center>
<p><br></p>
<center>
<img src="90deg.jpg" style="width:50.0%" /> <br> <strong>Fig. 5</strong>
The ganiometer affixed to the upper arm with surgical tape with its
center placed over the elbow joint. Note how the marked line on the
radius indicates an elbow angle of about 90<span
class="math inline">\(^o\)</span>.
</center>
<p><br></p>
<p>As the elbow flexes or extends, you should be able to peak at the
ganiometer to assess elbow angle.</p>
</div>
<div id="isometric-mvcs" class="section level3">
<h3>Isometric MVCs</h3>
<p>We will collect isometric force data from MVCs over 12 arm angles,
45-157.5<span class="math inline">\(^o\)</span> at 11.25<span
class="math inline">\(^o\)</span> (call it 11<span
class="math inline">\(^o\)</span>) intervals. Your load-cell is affixed
to a wooden or plastic dowel with a zip tie at one end and has a looped
zip tie attached to the other. Pass the looped zip tie around the
provided C-clamp and fasten the clamp to something that won’t move under
the torque of your forearm. This could be table, desk, ect. Hold the
dowel in your hand as shown in Fig. 6. Make sure your elbow is supported
by something rigid (i.e. the top of a chair back, etc.).</p>
<center>
<img src="grip.jpg" style="width:50.0%" /> <br> <strong>Fig. 6</strong>
How to grip the the load-cell.
</center>
<p><br></p>
<p>First collect your control (i.e., non-fatigued) data by starting at
the largest arm angle, 157.5<span class="math inline">\(^o\)</span>. You
can move through the 12 arm angles by rotating your body about your
hips. That is, at the high elbow angles, you’ll be more upright, at
lower angles, more bent over. At each angle, flex your elbow as hard as
you can to produce a MVC, resetting your DAQ system each time. Make sure
the load-cell forms a right angle with reference to your hand (Fig. 6).
You might want to practice a few times to get a sense of how this will
proceed.</p>
</div>
<div id="mvcs-after-eccentric-fatigue" class="section level3">
<h3>MVCs after Eccentric Fatigue</h3>
<p>To eccentrically fatigue your forearm flexors, you’ll perform
isokinetic controlled drops of a heavy object weighing 1/3 your maximum
isometric force recorded in the control experiments. A book bag or back
pack full of heavy items (ahhh. . . books) should do just fine.</p>
<p>To perform your isokinetic eccentric drops, grab the book bag with
the hand not involved in the experiment and pass it to the other as its
forearm is fully flexed. With the bag now in your study hand, lower the
bag slowly and at a constant speed. When the arm is fully extended, grab
the book bag with the free hand and then flex the study forearm. Repeat
by passing the bag to your study hand. Continue repeating this exercise
for 3 min. Keep the cycles uniformly frequent by monitoring the duration
(probably 3-4 s each cycle). At the end of the 3 mins, your forearm
flexors should feel tired.</p>
<p>After you fatigue your flexors, wait 2 min. Then repeat the
experiment exactly as you did for the control data.</p>
<p>Once you complete your experiments, upload your 24 data files to <a
href="https://drive.google.com/drive/folders/1ncojVtzbQaZWKNdFgWtkK5CeJ_PmAbq0?usp=sharing">this
Google directory</a>. Be sure to name your files according to the
following convention: “team_subject_angle_control/fatigue.csv”.
<strong>Please stick to all lower case.</strong></p>
</div>
</div>
<div id="software-and-coding-the-arduino" class="section level2">
<h2>Software and Coding the Arduino</h2>
<p>The sketch needed for this experiment could be quite simple. It does,
however, require you to download the HX711 library.</p>
<p>As a primer, this would work:</p>
<pre class="c"><code>
#include &quot;HX711.h&quot; // include the appropriate library for `scale` functions


HX711 scale; //invoke scale functions
    
void setup() {

 Serial.begin(115200); //set baud rate
   scale.begin(6, 7); //set digital pins 6 and 7 as the data and clock pins
  scale.tare(); //zero the measurements

}


void loop() {

  float newReading= scale.get_units(10); //establish new variable from load cel

   Serial.println(newReading); //print load cell value

}
</code></pre>
<p><strong>NOTE:</strong>* You may find it handy to work with a <a
href="https://www.arduino.cc/en/software/OldSoftwareReleases">previous
release of Arduino IDE</a> because the current version does not allow
copying data from the serial monitory</p>
</div>
<div id="summary-of-experimental-protocol" class="section level2">
<h2>Summary of Experimental Protocol</h2>
<ul>
<li>Fabricate and program a load-cell data acquisition system.</li>
<li>Make a ganiometer.</li>
<li>Record isometric MVCs at 12 distinct elbow angles.</li>
<li>Eccentrically fatigue your flexors.</li>
<li>Wait 2 minutes.</li>
<li>Repeat recording isometric MVCs at the same 12 distinct elbow
angles.</li>
<li>Upload data to <a
href="https://drive.google.com/drive/folders/1ncojVtzbQaZWKNdFgWtkK5CeJ_PmAbq0?usp=sharing">this
directory</a>.</li>
</ul>
<p>Pleas upload your data by Monday, November 13th.</p>
</div>
<div id="data-analysis" class="section level2">
<h2>Data Analysis</h2>
<p>For each experiment, we’ll stick to a convention within the muscle
physiology world by considering maximum normalized force. That is:</p>
<ol style="list-style-type: decimal">
<li>Find maximum force for each recording (don’t worry about the
specific values or units because you’ll. . .).</li>
<li>Find maximum force (<span class="math inline">\(F_{max}\)</span>)
across all angles for each experiment (control or fatigued) for each
individual and then divide all maximum values for that experiment by
<span class="math inline">\(F_{max}\)</span>. Your values will now be
normalize, i.e., scaled 0-1 with a value of 1 representing the highest
force recorded during an experiment.</li>
</ol>
<p>Using the normalized force data from the control and fatigued
experiments, we’ll consider shifts at which angle <span
class="math inline">\(F_{max}\)</span> occurred, the difference in <span
class="math inline">\(\theta_{max}\)</span>. To assess any shift in
<span class="math inline">\(\theta_{max}\)</span> with our limited elbow
angle samples, we could simply look at the angle at which the flexors
produce the most force. For instance, for the these data:</p>
<pre class="r"><code>library(ggplot2)
ang &lt;- seq(45,168.75,length.out = 12) #elbow angle

normF &lt;- c(0.71, 0.77, 0.83, 0.91, 0.97, 1.00, 0.96, 0.94, 0.89, 0.84, 0.77, 0.74) #you&#39;ll get this by reading in data and calculating normalized values


qplot(ang,normF)+geom_point(aes(x=ang[which.max(normF)],y=normF[which.max(normF)]),col=&quot;red&quot;,size=4)</code></pre>
<pre><code>## Warning: `qplot()` was deprecated in ggplot2 3.4.0.
## This warning is displayed once every 8 hours.
## Call `lifecycle::last_lifecycle_warnings()` to see where this warning was
## generated.</code></pre>
<p><img src="mod6_proj_files/figure-html/unnamed-chunk-4-1.png" width="672" /></p>
<pre class="r"><code>ang[which.max(normF)] #Find angle for Fmax, theta_max</code></pre>
<pre><code>## [1] 101.25</code></pre>
<p>So, of the 12 angles that we sampled (45–157.5<span
class="math inline">\(^o\)</span>), the maximum isometric force occurred
at angle (<span class="math inline">\(\theta_{max}\)</span>=)
101.25<span class="math inline">\(^o\)</span>. Simply doing this
overlooks the fact that the maximum force could have occurred between
any two of the sampled angles. Given the trend in the figure above,
<span class="math inline">\(\theta_{max}\)</span> could have been an
angle either greater and less than 101.25<span
class="math inline">\(^o\)</span>. To estimate the trend and find a
perhaps more refined estimation of <span
class="math inline">\(F_{max}\)</span> we’ll resort to (wait for it) . .
. modeling. In this case, a series of <a
href="https://en.wikipedia.org/wiki/Polynomial_regression">polynomial
models</a> should work. Polynomial models are great for fitting
parabolic data like those we have here, but are tricky in that one can
keep adding monomial terms very quickly to achieve a better fit. This
requires the practitioner to make sure that they are not overfitting the
model. Fortunately, we have the tools to prevent such a thing using AIC,
as we have learned.</p>
<p>Therefore, let’s fit some polynomial models to our example data with
a different number of monomial terms (i.e., orders) and tease apart the
fit and penalty for including more terms using AICc. Fortunately, we can
use the <code>lm()</code> and <code>poly()</code> functions from the
base package to fit the models and use <code>AICc()</code> from the
<code>MuMIn</code> package for AICc fitting.</p>
<pre class="r"><code>library(MuMIn)

poly.m2 &lt;- lm(normF~poly(ang,2)) #second order
poly.m3 &lt;- lm(normF~poly(ang,3)) #third order
poly.m4 &lt;- lm(normF~poly(ang,4)) #fourth order

AICc(poly.m2,poly.m3,poly.m4) #the second order model fits best</code></pre>
<pre><code>##         df      AICc
## poly.m2  4 -43.90316
## poly.m3  5 -39.60496
## poly.m4  6 -49.47748</code></pre>
<pre class="r"><code>x.pred &lt;- seq(45,157.5,length.out = 1000) #define 1000 angles from our range

normF.pred &lt;- predict(poly.m4,newdata = data.frame(ang=x.pred)) #predict the force using 1000 angles

qplot(ang,normF)+geom_point(aes(x=x.pred,y=normF.pred),col=&quot;red&quot;)+geom_point(aes(x=x.pred[which.max(normF.pred)],y=normF.pred[which.max(normF.pred)]),size=5,col=&quot;blue&quot;)</code></pre>
<p><img src="mod6_proj_files/figure-html/unnamed-chunk-5-1.png" width="672" /></p>
<pre class="r"><code>x.pred[which.max(normF.pred)] #theta_max</code></pre>
<pre><code>## [1] 104.9099</code></pre>
<p>In this example we constructed 3 polynomial models of the 2nd, 3rd,
4th variety and then used AICc to find the lowest AICc score. The 4th
order model fits best, so we then produced 1000 discrete values within
our range with <code>seq()</code> to pass to<code>predict()</code> to
predict force over this range using our best fitting model (producing
<code>normF.pred</code>). We then found the max of
<code>normF.pred</code> (the <span
class="math inline">\(F_{max}\)</span>) and plotted that, along with our
predictions. Lastly, we found <span
class="math inline">\(\theta_{max}\)</span>, a value of 104.9.</p>
<p>A goal of this project is to detect shifts in <span
class="math inline">\(F_{max}\)</span> with respect to angle. If we had
two data sets, one control and one produced after fatigue, we merely
need to compare the two <span
class="math inline">\(\theta_{max}\)</span>, that is, the shift in angle
for <span class="math inline">\(F_{max}\)</span> predicted by two
polynomial models. This would look something like this:</p>
<pre class="r"><code>normF.fat &lt;- c(0.65, 0.71, 0.81, 0.89,0.92, 0.96, 0.99, 1, 0.96, 0.92, 0.84, 0.77) #fatigue data, you&#39;ll get this by reading in data and calculating normalized values


qplot(ang,normF.fat)+geom_point(aes(x=ang[which.max(normF.fat)],y=normF.fat[which.max(normF.fat)]),col=&quot;red&quot;,size=4)</code></pre>
<p><img src="mod6_proj_files/figure-html/unnamed-chunk-6-1.png" width="672" /></p>
<pre class="r"><code>ang[which.max(normF.fat)] #Find angle for Fmax for fatigue data</code></pre>
<pre><code>## [1] 123.75</code></pre>
<pre class="r"><code>poly.m2.fat &lt;- lm(normF.fat~poly(ang,2)) #second order
poly.m3.fat &lt;- lm(normF.fat~poly(ang,3)) #third order
poly.m4.fat &lt;- lm(normF.fat~poly(ang,4)) #fourth order

AICc(poly.m2.fat,poly.m3.fat,poly.m4.fat) #the second order model fits best</code></pre>
<pre><code>##             df      AICc
## poly.m2.fat  4 -54.91075
## poly.m3.fat  5 -55.05750
## poly.m4.fat  6 -49.10034</code></pre>
<pre class="r"><code>normF.pred.fat &lt;- predict(poly.m2.fat,newdata = data.frame(ang=x.pred)) #predict the force using 1000 angles

qplot(ang,normF)+geom_point(aes(x=x.pred,y=normF.pred.fat),col=&quot;red&quot;)+geom_point(aes(x=x.pred[which.max(normF.pred.fat)],y=normF.pred.fat[which.max(normF.pred.fat)]),size=5,col=&quot;blue&quot;)</code></pre>
<p><img src="mod6_proj_files/figure-html/unnamed-chunk-6-2.png" width="672" /></p>
<pre class="r"><code>x.pred[which.max(normF.pred.fat)]-x.pred[which.max(normF.pred)] #shift in angle for Fmax, theta_max</code></pre>
<pre><code>## [1] 10.24775</code></pre>
<p>Looks as if there’s a 10.3<span class="math inline">\(^o\)</span>
shift in <span class="math inline">\(\theta_{max}\)</span>.</p>
<p>Say we had a bunch of data (from only 10 subjects) and you wanted to
compare the control against fatigued data graphically, you could do the
following. First we’ll simulate some control and fatigue data, adding
random noise drawn form a normal distribution and plot the data with a
color according to experiment to quickly assess the pattern.</p>
<pre class="r"><code>library(tidyverse)

set.seed(1234) #to keep random noise the same

sub &lt;- 1:10 #ten subjects
dat.l &lt;- list() #empty list

#loop to run through subjects and simulate data based on noise added to data above
for(i in sub){
  noise &lt;- rnorm(length(ang),mean = 0.02,sd = 0.01)
  noise2 &lt;- rnorm(length(ang),mean = 0.02,sd = 0.01)
  dat.l[[i]] &lt;- tibble(
    rbind(
      data.frame(ang,force=normF+noise,subject=i,exp=&quot;control&quot;),
      data.frame(ang,force=normF.fat+noise2,subject=i,exp=&quot;fatigue&quot;)
  )  
  )%&gt;%
    group_by(subject,exp)%&gt;%
    mutate(force=force/max(force))
  
}

dat &lt;- do.call(rbind,dat.l) #get data into one tibble

dat%&gt;%
  ggplot(aes(ang,force,col=exp))+geom_point()</code></pre>
<p><img src="mod6_proj_files/figure-html/unnamed-chunk-7-1.png" width="672" /></p>
<p>Now let’s take our data and get AICc scores for our three polynomial
models. We’ll group the data by subject and experiment and retrieve AICc
from each model (<code>m2</code>, <code>m3</code>, and <code>m4</code>).
Then, we’ll pivot the tibble to a longer format so that we have a single
column for AICc values.</p>
<pre class="r"><code>AICs &lt;- dat%&gt;%
  group_by(subject,exp)%&gt;%
  summarize(
    m2=AICc(lm(force~poly(ang,2))), #second order
    m3=AICc(lm(force~poly(ang,3))), #third order
    m4=AICc(lm(force~poly(ang,4))) #fourth order
  )%&gt;%
  pivot_longer(m2:m4,names_to=&quot;model&quot;,values_to=&quot;AICc&quot;)%&gt;%
  print()</code></pre>
<pre><code>## # A tibble: 60 × 4
## # Groups:   subject [10]
##    subject exp     model  AICc
##      &lt;int&gt; &lt;chr&gt;   &lt;chr&gt; &lt;dbl&gt;
##  1       1 control m2    -43.2
##  2       1 control m3    -39.3
##  3       1 control m4    -42.5
##  4       1 fatigue m2    -49.3
##  5       1 fatigue m3    -46.0
##  6       1 fatigue m4    -37.7
##  7       2 control m2    -43.8
##  8       2 control m3    -39.4
##  9       2 control m4    -55.2
## 10       2 fatigue m2    -57.6
## # ℹ 50 more rows</code></pre>
<p>Now we’ll do something very similar, predicting values for 1000
discrete angles across our range of elbow positions. Notice that we’ve
summarized the data by subject, experiment and model to find the angle
(<code>x.pred</code>) which matches the max force value.</p>
<pre class="r"><code>fits &lt;- dat%&gt;%
  group_by(subject,exp)%&gt;%
  summarize(
    m2=predict(lm(force~poly(ang,2)),newdata=data.frame(ang=x.pred)), #second order
    m3=predict(lm(force~poly(ang,3)),newdata=data.frame(ang=x.pred)), #third order
    m4=predict(lm(force~poly(ang,4)),newdata=data.frame(ang=x.pred)) #fourth order
  )%&gt;%
   pivot_longer(m2:m4,names_to=&quot;model&quot;)%&gt;%
   group_by(subject,exp,model)%&gt;%
   summarize(theta_max=x.pred[which.max(value)])%&gt;%
   print()</code></pre>
<pre><code>## Warning: Returning more (or less) than 1 row per `summarise()` group was deprecated in
## dplyr 1.1.0.
## ℹ Please use `reframe()` instead.
## ℹ When switching from `summarise()` to `reframe()`, remember that `reframe()`
##   always returns an ungrouped data frame and adjust accordingly.
## Call `lifecycle::last_lifecycle_warnings()` to see where this warning was
## generated.</code></pre>
<pre><code>## # A tibble: 60 × 4
## # Groups:   subject, exp [20]
##    subject exp     model theta_max
##      &lt;int&gt; &lt;chr&gt;   &lt;chr&gt;     &lt;dbl&gt;
##  1       1 control m2         107.
##  2       1 control m3         103.
##  3       1 control m4         104.
##  4       1 fatigue m2         115.
##  5       1 fatigue m3         118.
##  6       1 fatigue m4         117.
##  7       2 control m2         107.
##  8       2 control m3         103.
##  9       2 control m4         105.
## 10       2 fatigue m2         115.
## # ℹ 50 more rows</code></pre>
<p>Now let’s join these the <code>AICs</code> and <code>fits</code>
tables so that we can filter the model predictions by which model fits
best. Note <code>filter(best==TRUE)</code> after constructing a
<code>best</code> column according to the minimum AICc score using
<code>mutate()</code>.</p>
<pre class="r"><code>best.models &lt;- fits%&gt;%
  left_join(AICs)%&gt;%
  group_by(subject,exp)%&gt;%
  mutate(best=AICc==min(AICc))%&gt;%
  filter(best==TRUE)%&gt;%
  dplyr::select(-best)%&gt;%
  print()</code></pre>
<pre><code>## Joining with `by = join_by(subject, exp, model)`</code></pre>
<pre><code>## # A tibble: 20 × 5
## # Groups:   subject, exp [20]
##    subject exp     model theta_max  AICc
##      &lt;int&gt; &lt;chr&gt;   &lt;chr&gt;     &lt;dbl&gt; &lt;dbl&gt;
##  1       1 control m2         107. -43.2
##  2       1 fatigue m2         115. -49.3
##  3       2 control m4         105. -55.2
##  4       2 fatigue m2         115. -57.6
##  5       3 control m4         107. -47.7
##  6       3 fatigue m3         119. -57.2
##  7       4 control m4         104. -55.4
##  8       4 fatigue m3         120. -51.9
##  9       5 control m4         103. -48.9
## 10       5 fatigue m2         115. -51.3
## 11       6 control m4         103. -41.7
## 12       6 fatigue m3         120. -53.7
## 13       7 control m4         104. -48.0
## 14       7 fatigue m3         120. -55.9
## 15       8 control m2         107. -48.5
## 16       8 fatigue m3         116. -65.9
## 17       9 control m4         103. -48.3
## 18       9 fatigue m2         115. -52.3
## 19      10 control m4         105. -49.3
## 20      10 fatigue m2         115. -55.0</code></pre>
<p>To investigate whether a shift if <span
class="math inline">\(\theta_{max}\)</span> is different between the
control and fatigue experiments, we can run the following ANOVA.</p>
<pre class="r"><code>anova(lm(theta_max~exp,best.models))</code></pre>
<pre><code>## Analysis of Variance Table
## 
## Response: theta_max
##           Df Sum Sq Mean Sq F value    Pr(&gt;F)    
## exp        1 749.21  749.21  180.18 8.144e-11 ***
## Residuals 18  74.85    4.16                      
## ---
## Signif. codes:  0 &#39;***&#39; 0.001 &#39;**&#39; 0.01 &#39;*&#39; 0.05 &#39;.&#39; 0.1 &#39; &#39; 1</code></pre>
<p>Lastly, we should calculate the mean shift with SEM. To find the
shift, we can pivot the <code>best.models</code> tibble to a wider
format so we can have <span class="math inline">\(\theta_{max}\)</span>
for each experiment in separate columns and find the difference with
<code>mutate()</code>. Then, <code>summarise()</code> can be used to
find the mean and SEM.</p>
<pre class="r"><code>best.models%&gt;%
  pivot_wider(id_cols=subject,names_from = exp,values_from=theta_max)%&gt;%
  mutate(shift=fatigue-control)%&gt;%
  ungroup()%&gt;%
  summarise(mean.shift=mean(shift),se.shift=sd(shift)/sqrt(length(shift)))</code></pre>
<pre><code>## # A tibble: 1 × 2
##   mean.shift se.shift
##        &lt;dbl&gt;    &lt;dbl&gt;
## 1       12.2    0.960</code></pre>
</div>
</div>
<div id="project-report" class="section level1">
<h1>Project Report</h1>
<p>Please submit your report to your team GitHub repository as an .Rmd
document with HTML output that addresses the following questions:</p>
<ul>
<li>Do class-wide force-angle data for isometric MVC accord to a typical
FL relationship both under control and fatigued conditions?</li>
<li>Is there a significant shift between <span
class="math inline">\(\theta_{max}\)</span> between the control and
fatigue FL relationships?</li>
</ul>
<p>In answering these questions, be sure to use the visualization,
modeling, and model-assessments tools we’ve used in the course so
far.</p>
<p>The answers and narrative in your .Rmd should include the following
components:</p>
<ul>
<li>A YAML header that specifies HTML output, the authors, and a
bibliography named “BIOL3140.bib”. <strong>Submit this bibliography as
well!</strong></li>
<li>Sections including an introduction, methods, results, discussion,
author contributions, and references. Make sure that each, aside from
the references, includes one to two short paragraphs. Specifically:
<ul>
<li>Introduction: Frame the questions, indicating why they are
important, what background work has been done in this realm, and how you
will answer them. <strong>Please include at least one reference to
support the summary of previous work.</strong> Note: this can be done
easily by refiguring the introduction to this project report.</li>
<li>Methods: Explicitly state how you answered the questions, including
a narrative of all the analyses both qualitative and quantitative.</li>
<li>Results: Include any appropriate figures or tables and a narrative
of the main results that are important to answering the questions.</li>
<li>Discussion: Succinctly declare how the results relate to the
question and how they compare to previous work devoted to the topic. In
addition, be sure to comment on the importance of your findings to the
broader topic at hand. <strong>Please include at least one reference to
another relevant study.</strong> Note: circling back to the
introductions, both to this project description and yours, will be
helpful here.</li>
<li>Author contributions: Briefly outline what each team member
contributed to the project.</li>
</ul></li>
</ul>
<p>Project reports should be uploaded by 11:59 PM on Thursday, November
16th.</p>
<div class="green">
<div class="row">
<div class="col-sm-2">
<a class="far fa-sticky-note fa-5x" style="color:#939681"></a>
<!-- regular style -->
</div>
<div class="col-sm-9">
<p>Please have a look at out <a href="phaseII_rubric.html">Phase II
report rubric</a> to get a sense of how this and other Phase II reports
will be grade.</p>
</div>
</div>
</div>
<p><br></p>
</div>
<div id="references" class="section level1">
<h1>References</h1>
</div>



</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->

<script>
$(document).ready(function ()  {

    // temporarily add toc-ignore selector to headers for the consistency with Pandoc
    $('.unlisted.unnumbered').addClass('toc-ignore')

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
