<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />




<title>Module 6 Project</title>

<script src="site_libs/header-attrs-2.16/header-attrs.js"></script>
<script src="site_libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/sandstone.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>

<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>



<style type="text/css">
  code {
    white-space: pre;
  }
  .sourceCode {
    overflow: visible;
  }
</style>
<style type="text/css" data-origin="pandoc">
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
div.sourceCode { margin: 1em 0; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {  background-color: #f8f8f8; }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ef2929; } /* Alert */
code span.an { color: #8f5902; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #c4a000; } /* Attribute */
code span.bn { color: #0000cf; } /* BaseN */
code span.cf { color: #204a87; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4e9a06; } /* Char */
code span.cn { color: #000000; } /* Constant */
code span.co { color: #8f5902; font-style: italic; } /* Comment */
code span.cv { color: #8f5902; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #8f5902; font-weight: bold; font-style: italic; } /* Documentation */
code span.dt { color: #204a87; } /* DataType */
code span.dv { color: #0000cf; } /* DecVal */
code span.er { color: #a40000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #0000cf; } /* Float */
code span.fu { color: #000000; } /* Function */
code span.im { } /* Import */
code span.in { color: #8f5902; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #204a87; font-weight: bold; } /* Keyword */
code span.op { color: #ce5c00; font-weight: bold; } /* Operator */
code span.ot { color: #8f5902; } /* Other */
code span.pp { color: #8f5902; font-style: italic; } /* Preprocessor */
code span.sc { color: #000000; } /* SpecialChar */
code span.ss { color: #4e9a06; } /* SpecialString */
code span.st { color: #4e9a06; } /* String */
code span.va { color: #000000; } /* Variable */
code span.vs { color: #4e9a06; } /* VerbatimString */
code span.wa { color: #8f5902; font-weight: bold; font-style: italic; } /* Warning */

</style>
<script>
// apply pandoc div.sourceCode style to pre.sourceCode instead
(function() {
  var sheets = document.styleSheets;
  for (var i = 0; i < sheets.length; i++) {
    if (sheets[i].ownerNode.dataset["origin"] !== "pandoc") continue;
    try { var rules = sheets[i].cssRules; } catch (e) { continue; }
    var j = 0;
    while (j < rules.length) {
      var rule = rules[j];
      // check if there is a div.sourceCode rule
      if (rule.type !== rule.STYLE_RULE || rule.selectorText !== "div.sourceCode") {
        j++;
        continue;
      }
      var style = rule.style.cssText;
      // check if color or background-color is set
      if (rule.style.color === '' && rule.style.backgroundColor === '') {
        j++;
        continue;
      }
      // replace div.sourceCode by a pre.sourceCode rule
      sheets[i].deleteRule(j);
      sheets[i].insertRule('pre.sourceCode{' + style + '}', j);
    }
  }
})();
</script>



<style type="text/css">
/* for pandoc --citeproc since 2.11 */
div.csl-bib-body { }
div.csl-entry {
  clear: both;
}
.hanging div.csl-entry {
  margin-left:2em;
  text-indent:-2em;
}
div.csl-left-margin {
  min-width:2em;
  float:left;
}
div.csl-right-inline {
  margin-left:2em;
  padding-left:1em;
}
div.csl-indent {
  margin-left: 2em;
}
</style>

<link rel="stylesheet" href="style.css" type="text/css" />



<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
details > summary > p:only-child {
  display: inline;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark the anchor link active (and if it's in a dropdown, also mark that active)
  var dropdown = menuAnchor.closest('li.dropdown');
  if (window.bootstrap) { // Bootstrap 4+
    menuAnchor.addClass('active');
    dropdown.find('> .dropdown-toggle').addClass('active');
  } else { // Bootstrap 3
    menuAnchor.parent().addClass('active');
    dropdown.addClass('active');
  }

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "&#xe258;";
  border: none;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-bs-toggle="collapse" data-target="#navbar" data-bs-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">BIOL 3140</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="schedule.html">Schedule</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    Projects
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li class="dropdown-header">Phase I</li>
    <li>
      <a href="phaseI_rubric.html">Phase I report rubric</a>
    </li>
    <li>
      <a href="mod1_proj.html">Module 1 Project</a>
    </li>
    <li>
      <a href="mod2_proj.html">Module 2 Project</a>
    </li>
    <li>
      <a href="mod3_proj.html">Module 3 Project</a>
    </li>
    <li>
      <a href="mod4_proj.html">Module 4 Project</a>
    </li>
    <li>
      <a href="mod5_proj.html">Module 5 Project</a>
    </li>
    <li class="dropdown-header">Phase II</li>
    <li>
      <a href="phaseII_rubric.html">Phase II report rubric</a>
    </li>
    <li>
      <a href="mod6_proj.html">Module 6 Project</a>
    </li>
    <li>
      <a href="mod7_proj.html">Module 7 Project</a>
    </li>
    <li>
      <a href="mod8_proj.html">Module 8 Project</a>
    </li>
    <li>
      <a href="mod9_proj.html">Module 9 Project</a>
    </li>
    <li class="dropdown-header">Phase III</li>
    <li>
      <a href="phaseIII_rubric.html">Phase III report rubric</a>
    </li>
    <li>
      <a href="final_proj.html">Final Project</a>
    </li>
    <li class="dropdown-header">DaqPacks</li>
    <li>
      <a href="daqpack.html">DaqPack Item List</a>
    </li>
  </ul>
</li>
<li>
  <a href="https://github.com/orgs/bcorgbio/teams/biol3140_f22">BIOL 3140 Discussion</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">



<h1 class="title toc-ignore">Module 6 Project</h1>

</div>


<style>
div.green { background-color:#c7d1c0; border-radius: 5px; padding: 20px;}
</style>
<!-- Add icon library -->
<p><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.14.0/css/all.min.css"></p>
<div id="introduction" class="section level1">
<h1>Introduction</h1>
<p>Butterflies and moths (the Lepidoptera) are an exceptional <a
href="https://en.wikipedia.org/wiki/Evolutionary_radiation">radiation</a>
of insects, comprising nearly 160,000 described extant species. (For
context, the vertebrates comprise just 70,000 species or so.) The
Lepidoptera play a fundamental role in almost every terrestrial
ecosystem. These ecologically important species use paired wings to
power flight in search of places to lay eggs, feed on nectar and other
resources, and find mates. The airy environment in which moths and
butterflies fly undoubtedly exerts natural selective pressure on wing
size and shape. However, it appears as if the hind- and forewings have
different roles to play in powered flight and the ecology of the
Lepidoptera. If hindwings are removed from butterflies, they are able to
fly, but at the cost of speed and maneuverability (Jantzen and Eisner
2008). Therefore, it has been proposed that forewing shape and size may
be the result of stabilizing selection imposed by the physical and fluid
requirements of flight and hindwing shape and size may be determined
more by neutral selection, sexual selection, or selection due to
predator avoidance (Sourakov 2013, Barber et al. 2015, Willmott et
al. 2017, Rubin et al. 2018, Chazot et al. 2016).</p>
<center>
<p><img
src="https://3.bp.blogspot.com/-0ZJtpbcAlWo/W9R6ek_sNPI/AAAAAAAAUj0/ncYRT0nN9MctC-x60-boMe8bodGXNM2LQCLcBGAs/s450/ButterflyWing3a.jpg"
style="width:50.0%" /></p>
<br> <em>The anotomy of a butterfly or moth.</em>
</center>
<p><br> <br></p>
<p>Studies based on experimental manipulations like those of Jantzen and
Eisner (2008), important as they are, are limited in their ability to
characterize shape variables and the ecological roles of fore- and hind
wings over evolutionary time scales and across lepidoptera lineages.
Comparative analysis of shape data from natural history collections has
helped bridge the gap between experimental studies and macroevolutionary
patterns. If there is in fact different constraints on the evolution of
fore- and hindwing morphology, we would expect to find a consistent
patterns of diffent fore- and hindwing shape across the lepidopter.
Indeed, a number of studies have investigated whether for- and hindwing
shape and size differs in several isolated groups of butterflies.
Strauss (1990) found the shape of hindwings were much more variable than
forewings in heliconiine and ithomiine butterflies. In contrast, Chazot
et al. (2016) found a strong correlation between forewing and hindwing
shapes in the famous morpho group of butterflies. Therefore, the limited
research on this topic does not resolve whether fore- and hindwing shape
differ consistently across the Lepidoptera. Each of these studies was
limited in their phylogenetic scope, encompassing very few species
spanning few families of butterflies. Addressing whether fore- and
hindwing shape varies independently, a much broader phylogenetic sample
should be analyzed, one that includes not just representatives of the
butterfly lineage, but also moths.</p>
<center>
<p><img
src="https://d3i71xaburhd42.cloudfront.net/bb0cd297955c4ec58bd60777dc345f7c1d5d93bd/3-Figure1-1.png"
style="width:50.0%" /></p>
<br> <em>Several species of Heliconiinae butterflies, demonstrating the
variability in wingshape. From <span class="citation">Penz and Peggie
(2003)</span>.</em>
</center>
<p><br> <br></p>
<p>The goal of this project is to study wing-shape evolution across the
lepidoptera, including hundreds of species from both the moths (a
paraphyletic group) and butterflies. To do this, we’ll take a
morphometric approach, characterizing the shape outline of fore- and
hindwings from about 200 species of Lepidoptera.</p>
</div>
<div id="methods" class="section level1">
<h1>Methods</h1>
<div id="image-acquisition" class="section level2">
<h2>Image acquisition</h2>
<p>A few decades ago, a morphometric project like this would require
visiting museums collections from around the world to take images of the
specimens and species of interest. In fact, this was a major part of
Prof. Kenaley’s PhD thesis, which included trips to natural history
collections in France, Denmark, Great Britain, Taiwan, Japan, Monaco,
New Zealand, and Australia. Ahhhhhh, the gool ol’ days. Fortunately (or
unfortunately), travel to such wonderful places may be avoided as
collections have made major efforts to photograph their specimens. Now
that most museums around the world have their holdings cataloged in
databases, researchers can search their collections for records that
have images.</p>
<p>Until very recently, amassing a data set of images for morphometric
work would require querying individual databases, one at a time.
However, most natural history museums share their data with the Global
Biodiversity Information Facility (GBIF), a clearing house of
collections and species information. In the context of our project, one
could search the GBIF database online for moth and butterfly records
that have images. Fortunately for use, the folks at GBIF have created
and R package, <a
href="https://www.gbif.org/tool/81747/rgbif"><code>rgif</code></a>, that
permits access to GBIF’s application programming interface (API),
rendering their records searchable from the command line.</p>
<p>Thus, the images we’ll be studying were found by querying GBIFs
holdings and downloading the images from URLs that are provided with the
records. Prof. Kenaley has screened these images to make sure they are
suitable for digitizing and morphometric analysis.</p>
<p>The species list</p>
<p>BMNH image collection: <a
href="https://data.nhm.ac.uk/dataset/56e711e6-c847-4f99-915a-6894bb5c5dea/resource/05ff2255-c38a-40c9-b657-4ccb55ab2feb?q=lepidoptera&amp;_gl=1*z3b8cj*_ga*NDMwNjI0MzMyLjE2NjQ4MjMzMzg"
class="uri">https://data.nhm.ac.uk/dataset/56e711e6-c847-4f99-915a-6894bb5c5dea/resource/05ff2255-c38a-40c9-b657-4ccb55ab2feb?q=lepidoptera&amp;_gl=1*z3b8cj*_ga*NDMwNjI0MzMyLjE2NjQ4MjMzMzg</a>.*_ga_PYMKGK73C4*MTY2NDgyMzMzOC4xLjAuMTY2NDgyMzMzOC4wLjAuMA..&amp;filters=_has_image%3Atrue</p>
<p>Phylogeny: <a
href="https://www.pnas.org/doi/pdf/10.1073/pnas.1907847116"
class="uri">https://www.pnas.org/doi/pdf/10.1073/pnas.1907847116</a></p>
<p>Tree from ^ <a
href="https://www.pnas.org/cms/10.1073/pnas.1907847116/asset/075c8ff7-6363-462d-a528-f2b13710804d/assets/graphic/pnas.1907847116fig01.jpeg"
class="uri">https://www.pnas.org/cms/10.1073/pnas.1907847116/asset/075c8ff7-6363-462d-a528-f2b13710804d/assets/graphic/pnas.1907847116fig01.jpeg</a></p>
<p>I think this is tree file for figure S12 (based on <a
href="https://datadryad.org/stash/dataset/doi:10.5061/dryad.j477b40">READme</a>
file and <a
href="https://www.pnas.org/doi/suppl/10.1073/pnas.1907847116/suppl_file/pnas.1907847116.sapp.pdf">Appendix</a>)</p>
<p>papilio paper: <a
href="https://academic.oup.com/sysbio/article/69/5/813/5817324"
class="uri">https://academic.oup.com/sysbio/article/69/5/813/5817324</a>
To capture shape data, we’ll produce outlines for fore- and hindwing
shapes from photographs of moths and butterflies</p>
<p>evolution paper: <a
href="https://onlinelibrary.wiley.com/doi/full/10.1111/evo.12114"
class="uri">https://onlinelibrary.wiley.com/doi/full/10.1111/evo.12114</a></p>
</div>
<div id="species-list-and-phylogeny" class="section level2">
<h2>Species list and phylogeny</h2>
<p>Because we’ll be comparing wing shape across species of the
Lepidoptera, this project is by definition a comparative one and we’ll
need a phylogeny for the group. Fortunately a recent phylogenomic
analysis for 200 species of Lepidotera by <span
class="citation">Kawahara et al. (2019)</span> resolved the
relationships of the majore groups of this order. We’ve used the taxa
studied in this paper to find images in GBIF and we’ll use <a
href="lep_tree2.tre">a tree file</a> representing phylogeny for our
comparative analsyis.</p>
<center>
<p><br> <img
src="https://www.pnas.org/cms/10.1073/pnas.1907847116/asset/075c8ff7-6363-462d-a528-f2b13710804d/assets/graphic/pnas.1907847116fig01.jpeg"
style="width:50.0%" /> <br></p>
The phylogenetic relationships of the Lepidoptera according the <span
class="citation">Kawahara et al. (2019)</span>.
</center>
<p><br></p>
</div>
<div id="digitization" class="section level2">
<h2>Digitization</h2>
<p>Our shape analysis will be based on the outline of the hind- and
fore-wings of ~200 species of Lepidoptera. To outline the wings, we’ll
follow this generalize workflow below in the image analysis program FIJI
(FIJI is just imageJ).</p>
<br>
<center>
<img src="lep_workflow.jpeg" style="width:50.0%" /> <br>
</center>
<p><br></p>
<p>By now, you and your team should have been assigned about 20 of the
~200 species to outline and have accessed <a
href="https://drive.google.com/drive/folders/1NEdQ0h5KXum3DzbhCzl857R_5YxZEQq3?usp=sharing">the
directories containing their photos</a>. First choose one of your
species to work on. For each species, start with what you think is the
best image, that is, it has a scale bar, the wings are not damaged, and
the hindwing is most exposed. Open that best image in Fiji.</p>
<p>After opening the photo, follow the steps above.</p>
<ol style="list-style-type: decimal">
<li><p>If there’s a scale bar, <a
href="https://imagejdocu.list.lu/gui/analyze/set_scale">set the
scale</a>.</p></li>
<li><p>If the right wing pair is damaged, flip the image horizontally.
Select “Image” <span class="math inline">\(\rightarrow\)</span>
“Transform” <span class="math inline">\(\rightarrow\)</span> “Flip
horizontally”. <strong>Note:</strong> you may have to do this only for
the fore- or hindwing.</p></li>
<li><p>Select the <a
href="https://imagej.nih.gov/ij/docs/guide/146-19.html">polygon
selection tool</a>.</p></li>
<li><p>Trace the wing by placing the first point on the polygon on the
anterior margin of the wing where it meets the thorax. Like so .
.</p></li>
</ol>
<center>
<img src="firstpoint.png" style="width:40.0%" /> <br>
</center>
<p><br></p>
<p>Continue to trace the outline of the wing being sure to include
enough points to capture more complex areas. For instance, in this
swallowtail hindwing, it may require 30 or more points to outline the
interesting projections along the posterior margin.</p>
<br>
<center>
<img src="hindwing.png" style="width:40.0%" /> <br>
</center>
<p><br></p>
<p>You may also find it easier to get an acurate outline if you zoom in
(<code>crtl+</code> or <code>command+</code>).</p>
<ol start="5" style="list-style-type: decimal">
<li>Run the following macro in imageJ.</li>
</ol>
<pre><code>run(&quot;Fit Spline&quot;);
run(&quot;Properties... &quot;, &quot;list&quot;);</code></pre>
<p>This script <a
href="https://en.wikipedia.org/wiki/Spline_interpolation">fits a
spline</a> through the selected points and produces a list of the x,y
coordinates.</p>
<p><strong>Note:</strong> For the first image you outline, you will have
to open the macro editor (“Plugins” <span
class="math inline">\(\rightarrow\)</span> “New” <span
class="math inline">\(\Macro\)</span>). Paste the script command from
above into the editor and save this new macro to an appropriate
directory, perhaps you cloned repo. You’ll want to access this later
(“Plugins” <span class="math inline">\(\rightarrow\)</span> “Edit”) so
that you automate this process.</p>
<ol start="6" style="list-style-type: decimal">
<li>Save the results table that pops up and be sure to append either
“_hindwing” or “_forewing” to the file name and make sure this is before
the “.txt” file extension. <code>command+s</code> or <code>ctrl+s</code>
works.</li>
</ol>
<p>For some species there may be only one image. For others there may be
10 or more. If you can, aim to outline the wing of three specimens per
species.</p>
<p>After you team has completed outlining the assigned species, upload
your text files to <a
href="https://drive.google.com/drive/folders/1M7kb_6EooVdXZlMvkpOPSbwaRu2TfCDJ?usp=sharing">this
directory</a>. Once all the data appear here, your team can download all
the outline data.</p>
</div>
<div id="shape-analysis" class="section level2">
<h2>Shape analysis</h2>
<p>As we learned in class and you’ve been reading, researchers
interested in comparing shape change between species can use one of two
methods: landmark- or outline-base shape analysis. We’ll use the
latter.</p>
<p>Outline analysis involves breaking the shapes down into a
mathematical description of the points that constitute the shapes. The
most popular approach for fitting functions is elliptical Fourier
analysis (EFA) in which the shapes in a data set are described as a
series of ellipses. EFA attempts to align a number of ellipses to the
outline of shapes and the scale of the major and minor axes of the
ellipses are then used as the descriptor of the shapes. Of course, one
could fit a lot of ellipses and describe the shape perfectly, i.e., with
100% power; however practitioners usually stop short of this and choose
some lower lever of lower, say 99%. The number of ellipses is referred
to as the number of harmonics. In the figure below, you see the a higher
number of harmonics describe the shape in question more perfectly.</p>
<br>
<center>
<img
src="https://www.researchgate.net/profile/Ryan-Felice/publication/260448082/figure/fig4/AS:297032695730198@1447829521359/Outline-reconstruction-using-Elliptical-Fourier-Descriptors-Black-contours-represent-the.png"
style="width:60.0%" /> <br> Elliptical Fourier analysis of the
pygostyle, the fused plate of caudal vertebrae that support tail
feathers in birds.
</center>
<p><br></p>
<p>Once EFA is used to describe the shapes in a data set, one is left
with many variables to consider including all the descriptors of all the
ellipses that form the outlines. This could be many hundreds of
variables in a data set like ours. This is a serious <a
href="https://en.wikipedia.org/wiki/Multivariate_statistics">multivariate
problem</a>.</p>
<p>For such a problem, researchers usually turn to Principal Components
analysis (PCA) whereby the very complicated suite of variables is
distilled into a set of fewer variables that describe the important
variance among the original variables. In a sense, PCA produces a series
of regressions, or vectors, that pass through the values of the original
variables. This operation is performed iteratively, with the first
vector (the first principal component, PC1), drawn through the variable
space that accounts for the most variance in the data set. After the
combination of variables forming PC1 is removed, a subsequent vector
(PC2) is drawn through another unique variable space that describes the
second most amount of variance. This continues until most of the
varaince is captured, resulting in scores of PCs. The first two or
three, however, are usually enough to describe a health amount of
variance, say 90% or more.</p>
<p>To perform EFA and then PCA, we need only to turn to one R package,
<code>Momocs</code>. Let’s load this package and begin with an example
analysis that we can scale up on our full data set.</p>
<p>Let’s begin by reading in one of the outline text files,
Prof. Kenaley has already produced for 10 species of butterflies and
moths. Please <a href="lep_examples.zip">download these</a> to follow
along. You can unzip this directory and place it your project
directory.</p>
<div class="sourceCode" id="cb2"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb2-1"><a href="#cb2-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(tidyverse)</span>
<span id="cb2-2"><a href="#cb2-2" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(Momocs)</span>
<span id="cb2-3"><a href="#cb2-3" aria-hidden="true" tabindex="-1"></a>f <span class="ot">&lt;-</span> <span class="fu">list.files</span>(<span class="st">&quot;lep_examples&quot;</span>,<span class="at">pattern=</span><span class="st">&quot;.txt&quot;</span>,<span class="at">full.names =</span> <span class="cn">TRUE</span>)</span>
<span id="cb2-4"><a href="#cb2-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb2-5"><a href="#cb2-5" aria-hidden="true" tabindex="-1"></a>out <span class="ot">&lt;-</span> <span class="fu">read_delim</span>(f[<span class="dv">1</span>],<span class="at">delim=</span><span class="st">&quot;</span><span class="sc">\t</span><span class="st">&quot;</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb2-6"><a href="#cb2-6" aria-hidden="true" tabindex="-1"></a>  <span class="fu">as.matrix</span>()</span></code></pre></div>
<pre><code>## Rows: 361 Columns: 2
## ── Column specification ─────────────────────────────────────────────
## Delimiter: &quot;\t&quot;
## dbl (2): X, Y
## 
## ℹ Use `spec()` to retrieve the full column specification for this data.
## ℹ Specify the column types or set `show_col_types = FALSE` to quiet this message.</code></pre>
<div class="sourceCode" id="cb4"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb4-1"><a href="#cb4-1" aria-hidden="true" tabindex="-1"></a>out <span class="sc">%&gt;%</span> </span>
<span id="cb4-2"><a href="#cb4-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">list</span>() <span class="sc">%&gt;%</span> </span>
<span id="cb4-3"><a href="#cb4-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">Out</span>() <span class="sc">%&gt;%</span> </span>
<span id="cb4-4"><a href="#cb4-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">coo_flipx</span>() <span class="sc">%&gt;%</span> </span>
<span id="cb4-5"><a href="#cb4-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">stack</span>()</span></code></pre></div>
<p><img src="mod6_proj_files/figure-html/momocexp1-1.png" width="672" /></p>
<p>Notice that we’ve found all the files in the “lep_examples” directory
with <code>list.files()</code>, searching for the patter “.txt” and
using the full filepath to make a loading the data in the directory
possible. Next, we use <code>read_delim()</code> with
<code>delim=\t</code> rather than <code>read_csc()</code> because FIJI
outputs tab-delimited data. We save this outline as <code>out</code>
after tranforming it to a matrix, the prefer format for
<code>Momocs</code>.</p>
<p>In the next operation, we transform this matrix into an
<code>Momocs</code> outline using <code>Out()</code>; however,
<code>Out()</code> must take a list of matrices and therefore we pipe in
the <code>list()</code> function first. Next, <code>coo_flipx()</code>
is invoked to flip the outline about the x axis–remember that the origin
of the image field is the upper left, not the lower left in the usual
graphical representation. Lastly, <code>stack()</code> is invoked to
visualize the outline. And here we have a forewing visualized.</p>
<p>Now, we will of course have hundreds of files to load and analyze
with PCA. For this challenge we will rely on the <code>for</code> loop
convention we’ve learned in projects past. We have the file pathes store
in <code>f</code>, so let’s develop this loop and store our outline data
in a list.</p>
<div class="sourceCode" id="cb5"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb5-1"><a href="#cb5-1" aria-hidden="true" tabindex="-1"></a>outs.l <span class="ot">&lt;-</span> <span class="fu">list</span>()</span>
<span id="cb5-2"><a href="#cb5-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-3"><a href="#cb5-3" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span>(i <span class="cf">in</span> f){</span>
<span id="cb5-4"><a href="#cb5-4" aria-hidden="true" tabindex="-1"></a>  outs.l[[i]] <span class="ot">&lt;-</span> out <span class="ot">&lt;-</span> <span class="fu">read_delim</span>(i,<span class="at">delim=</span><span class="st">&quot;</span><span class="sc">\t</span><span class="st">&quot;</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb5-5"><a href="#cb5-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">as.matrix</span>()</span>
<span id="cb5-6"><a href="#cb5-6" aria-hidden="true" tabindex="-1"></a>}</span></code></pre></div>
<p>As was mentioned above, the <code>Out()</code> function prefers a
list, so we can work on the list established here to visualize our
outlines.</p>
<div class="sourceCode" id="cb6"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb6-1"><a href="#cb6-1" aria-hidden="true" tabindex="-1"></a>outs.l <span class="sc">%&gt;%</span> </span>
<span id="cb6-2"><a href="#cb6-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">Out</span>() <span class="sc">%&gt;%</span> </span>
<span id="cb6-3"><a href="#cb6-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">coo_flipx</span>() <span class="sc">%&gt;%</span> </span>
<span id="cb6-4"><a href="#cb6-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">stack</span>()</span></code></pre></div>
<p><img src="mod6_proj_files/figure-html/out-1.png" width="672" /></p>
<p>Wonderful! Fore- and hindwings loaded for the 10 examples species.
However, notice we have a few problems. First, both fore- and hindwing
are included in the</p>
</div>
</div>
<div id="project-report" class="section level1">
<h1>Project Report</h1>
<p>Please submit your report to your team GitHub repository as an .Rmd
document with HTML output that addresses the following:</p>
<ul>
<li>lorem ipsum</li>
<li>lorem ispum</li>
</ul>
<p>In answering your question, your .Rmd should include the following
components:</p>
<ul>
<li>A YAML header that specifies HTML output, the authors, and a
bibliograph named “BIOL3140.bib”. <strong>Submit this bibliography as
well!</strong></li>
<li>Sections including an introduction, methods, results, discussion,
author contributions, and references. Make sure that each, aside from
the references, includes one to two short paragraphs. Specifially:
<ul>
<li>Introduction: Frame the questions, indicating why they are
important, what background work has been done in this realm, and how you
will answer them. <strong>Please include at least one reference to
support the summary of previous work.</strong> Note: this can be done
easily by refiguring the introduction to this project report.</li>
<li>Methods: Explicitly state how you answered the questions, including
a narrative of all the anlyses both qualitative and quantitative.</li>
<li>Results: Include any appropriate figures or tables and a narrative
of the main results that are important to answering the questions.</li>
<li>Discussion: Succinctly declare how the results relate to the
question and how they compare to previous work devoted to the topic. In
addition, be sure to comment on the importance of your findings to the
broader topic at hand. <strong>Please include at least one reference to
another relevant study.</strong> Note: circling back to the
introductions, both to this project description and yours, will be
helpful here.</li>
<li>Author contributions: Breifly outline what each team member
contributed to the project.</li>
</ul></li>
</ul>
<p>Project reports should be uploaded by 11:59 PM on Sunday, October
24th.</p>
<div class="sourceCode" id="cb7"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb7-1"><a href="#cb7-1" aria-hidden="true" tabindex="-1"></a>f <span class="ot">&lt;-</span> <span class="fu">list.files</span>(<span class="st">&quot;./proj6data&quot;</span>,<span class="at">full.names =</span> T)</span>
<span id="cb7-2"><a href="#cb7-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb7-3"><a href="#cb7-3" aria-hidden="true" tabindex="-1"></a><span class="fu">gsub</span>(<span class="st">&quot;BIOL3140(.+)&quot;</span>,<span class="st">&quot;BIOL314020</span><span class="sc">\\</span><span class="st">1&quot;</span>,<span class="st">&quot;BIOL3140_subject21&quot;</span>)</span>
<span id="cb7-4"><a href="#cb7-4" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span>(i <span class="cf">in</span> f ){</span>
<span id="cb7-5"><a href="#cb7-5" aria-hidden="true" tabindex="-1"></a> i2 <span class="ot">&lt;-</span> <span class="fu">gsub</span>(<span class="st">&quot;(.+)BIOL3140(.+)&quot;</span>,<span class="st">&quot;</span><span class="sc">\\</span><span class="st">1BIOL314020</span><span class="sc">\\</span><span class="st">2&quot;</span>,i)</span>
<span id="cb7-6"><a href="#cb7-6" aria-hidden="true" tabindex="-1"></a>  <span class="fu">file.copy</span>(i,i2)</span>
<span id="cb7-7"><a href="#cb7-7" aria-hidden="true" tabindex="-1"></a>}</span></code></pre></div>
</div>
<div id="references" class="section level1 unnumbered">
<h1 class="unnumbered">References</h1>
<div id="refs" class="references csl-bib-body hanging-indent">
<div id="ref-kawahara2019phylogenomics" class="csl-entry">
Kawahara, Akito Y, David Plotkin, Marianne Espeland, Karen Meusemann,
Emmanuel FA Toussaint, Alexander Donath, France Gimnich, et al. 2019.
<span>“Phylogenomics Reveals the Evolutionary Timing and Pattern of
Butterflies and Moths.”</span> <em>Proceedings of the National Academy
of Sciences</em> 116 (45): 22657–63.
</div>
<div id="ref-penz2003phylogenetic" class="csl-entry">
Penz, Carla M, and Djunijanti Peggie. 2003. <span>“Phylogenetic
Relationships Among Heliconiinae Genera Based on Morphology
(Lepidoptera: Nymphalidae).”</span> <em>Systematic Entomology</em> 28
(4): 451–79.
</div>
</div>
</div>



</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->

<script>
$(document).ready(function ()  {

    // temporarily add toc-ignore selector to headers for the consistency with Pandoc
    $('.unlisted.unnumbered').addClass('toc-ignore')

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
